<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Setup &mdash; Vivarium E. coli 1.0.0 documentation</title>
      <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/nbsphinx-code-cells.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../_static/documentation_options.js?v=8d563738"></script>
        <script src="../../_static/doctools.js?v=888ff710"></script>
        <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
        <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../index.html" class="icon icon-home">
            Vivarium E. coli
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../../reference/processes.html">Processes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../reference/composites.html">Composites</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../reference/experiments.html">Experiments</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../tutorials.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../docs.html">Working with Documentation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../reference/stores.html">Stores</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../reference/api_ref.html">API Reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Vivarium E. coli</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Setup</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/notebooks/cofactors/cofactors_data_processing.ipynb.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="Setup">
<h1>Setup<a class="headerlink" href="#Setup" title="Link to this heading"></a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>

<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pprint</span>
<span class="kn">import</span> <span class="nn">ast</span>
<span class="kn">import</span> <span class="nn">re</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">dill</span>
<span class="kn">import</span> <span class="nn">requests</span>
<span class="kn">import</span> <span class="nn">xmltodict</span>
<span class="kn">import</span> <span class="nn">json</span>

<span class="n">pp</span> <span class="o">=</span> <span class="n">pprint</span><span class="o">.</span><span class="n">PrettyPrinter</span><span class="p">(</span><span class="n">depth</span><span class="o">=</span><span class="mi">6</span><span class="p">)</span>

<span class="n">os</span><span class="o">.</span><span class="n">chdir</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">expanduser</span><span class="p">(</span><span class="s1">&#39;~/vivarium-ecoli&#39;</span><span class="p">))</span>

<span class="n">ALLOWED_METAL_NAMES</span> <span class="o">=</span>   <span class="p">{</span><span class="s1">&#39;Iron&#39;</span><span class="p">:</span> <span class="s1">&#39;FE+2&#39;</span><span class="p">,</span> <span class="s1">&#39;Cobalt&#39;</span><span class="p">:</span> <span class="s1">&#39;CO+2&#39;</span><span class="p">,</span> <span class="s1">&#39;Copper&#39;</span><span class="p">:</span> <span class="s1">&#39;CU+2&#39;</span><span class="p">,</span> <span class="s1">&#39;Manganese&#39;</span><span class="p">:</span> <span class="s1">&#39;MN+2&#39;</span><span class="p">,</span> <span class="s1">&#39;Molybdenum&#39;</span><span class="p">:</span> <span class="s1">&#39;CPD-8123&#39;</span><span class="p">,</span> <span class="s1">&#39;Nickel&#39;</span><span class="p">:</span> <span class="s1">&#39;NI+2&#39;</span><span class="p">,</span> <span class="s1">&#39;Zinc&#39;</span><span class="p">:</span> <span class="s1">&#39;ZN+2&#39;</span><span class="p">,</span>
                        <span class="s1">&#39;Calcium&#39;</span><span class="p">:</span> <span class="s1">&#39;CA+2&#39;</span><span class="p">,</span> <span class="s1">&#39;Magnesium&#39;</span><span class="p">:</span> <span class="s1">&#39;MG+2&#39;</span><span class="p">,</span> <span class="s1">&#39;Sodium&#39;</span><span class="p">:</span> <span class="s1">&#39;NA+&#39;</span><span class="p">,</span> <span class="s1">&#39;Potassium&#39;</span><span class="p">:</span> <span class="s1">&#39;K+&#39;</span><span class="p">,</span>
                        <span class="s1">&#39;Iron-sulfur \(4Fe-4S\)&#39;</span><span class="p">:</span> <span class="s1">&#39;CPD-7&#39;</span><span class="p">,</span> <span class="s1">&#39;Iron-sulfur \(2Fe-2S\)&#39;</span><span class="p">:</span> <span class="s1">&#39;CPD-6&#39;</span><span class="p">,</span>
                         <span class="s1">&#39;Iron-sulfur \(4Fe-4S-S-AdoMet\)&#39;</span><span class="p">:</span> <span class="s1">&#39;CPD-7&#39;</span><span class="p">,</span> <span class="s1">&#39;Iron-sulfur \(3Fe-4S\)&#39;</span><span class="p">:</span> <span class="s1">&#39;3FE-4S&#39;</span><span class="p">,</span> <span class="s1">&#39;Iron-oxo-sulfur \(4Fe-2O-2S\)&#39;</span><span class="p">:</span> <span class="s1">&#39;CPD-7&#39;</span><span class="p">,</span>
                         <span class="s1">&#39;Iron-sulfur&#39;</span><span class="p">:</span> <span class="s1">&#39;CPD-7&#39;</span><span class="p">,</span> <span class="c1"># has to be after others since it is a substring of others</span>
                        <span class="s1">&#39;heme&#39;</span><span class="p">:</span> <span class="s1">&#39;Heme-b&#39;</span><span class="p">,</span> <span class="s1">&#39;Molybdate&#39;</span><span class="p">:</span> <span class="s1">&#39;CPD-3&#39;</span><span class="p">,</span> <span class="s1">&#39;heme B&#39;</span><span class="p">:</span> <span class="s1">&#39;Heme-b&#39;</span><span class="p">,</span> <span class="s1">&#39;Cobalamin&#39;</span><span class="p">:</span> <span class="s1">&#39;COB-I-ALAMIN&#39;</span><span class="p">,</span>
                         <span class="s1">&#39;Selenocysteine&#39;</span><span class="p">:</span> <span class="s1">&#39;L-SELENOCYSTEINE&#39;</span><span class="p">,</span>
                        <span class="s1">&#39;Divalent metal cation&#39;</span><span class="p">:</span> <span class="s1">&#39;Any+2&#39;</span><span class="p">}</span>


<span class="n">ACCEPTED_OTHER_FEATURES</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;PYRIDOXAL_PHOSPHATE&#39;</span><span class="p">,</span> <span class="s1">&#39;THIAMINE-PYROPHOSPHATE&#39;</span><span class="p">,</span> <span class="s1">&#39;FMN&#39;</span><span class="p">,</span> <span class="s1">&#39;FAD&#39;</span><span class="p">,</span> <span class="s1">&#39;LIPOIC-ACID&#39;</span><span class="p">,</span> <span class="s1">&#39;BIOTIN&#39;</span><span class="p">}</span>

<span class="n">AMINO_ACID_MAP</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;A&#39;</span><span class="p">:</span> <span class="s1">&#39;ALA&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">:</span> <span class="s1">&#39;CYS&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">:</span> <span class="s1">&#39;ASP&#39;</span><span class="p">,</span> <span class="s1">&#39;E&#39;</span><span class="p">:</span> <span class="s1">&#39;GLU&#39;</span><span class="p">,</span> <span class="s1">&#39;F&#39;</span><span class="p">:</span> <span class="s1">&#39;PHE&#39;</span><span class="p">,</span> <span class="s1">&#39;G&#39;</span><span class="p">:</span> <span class="s1">&#39;GLY&#39;</span><span class="p">,</span> <span class="s1">&#39;H&#39;</span><span class="p">:</span> <span class="s1">&#39;HIS&#39;</span><span class="p">,</span> <span class="s1">&#39;I&#39;</span><span class="p">:</span> <span class="s1">&#39;ILE&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;K&#39;</span><span class="p">:</span> <span class="s1">&#39;LYS&#39;</span><span class="p">,</span> <span class="s1">&#39;L&#39;</span><span class="p">:</span> <span class="s1">&#39;LEU&#39;</span><span class="p">,</span> <span class="s1">&#39;M&#39;</span><span class="p">:</span> <span class="s1">&#39;MET&#39;</span><span class="p">,</span> <span class="s1">&#39;N&#39;</span><span class="p">:</span> <span class="s1">&#39;ASN&#39;</span><span class="p">,</span> <span class="s1">&#39;P&#39;</span><span class="p">:</span> <span class="s1">&#39;PRO&#39;</span><span class="p">,</span> <span class="s1">&#39;Q&#39;</span><span class="p">:</span> <span class="s1">&#39;GLN&#39;</span><span class="p">,</span> <span class="s1">&#39;R&#39;</span><span class="p">:</span> <span class="s1">&#39;ARG&#39;</span><span class="p">,</span> <span class="s1">&#39;S&#39;</span><span class="p">:</span> <span class="s1">&#39;SER&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;T&#39;</span><span class="p">:</span> <span class="s1">&#39;THR&#39;</span><span class="p">,</span> <span class="s1">&#39;V&#39;</span><span class="p">:</span> <span class="s1">&#39;VAL&#39;</span><span class="p">,</span> <span class="s1">&#39;W&#39;</span><span class="p">:</span> <span class="s1">&#39;TRP&#39;</span><span class="p">,</span> <span class="s1">&#39;Y&#39;</span><span class="p">:</span> <span class="s1">&#39;TYR&#39;</span><span class="p">,</span> <span class="s1">&#39;U&#39;</span><span class="p">:</span> <span class="s1">&#39;SEL&#39;</span><span class="p">,</span> <span class="s1">&#39;*&#39;</span><span class="p">:</span> <span class="s1">&#39;TER&#39;</span><span class="p">}</span>

<span class="c1"># Create a list of temporarily allowed Gene Ontology terms to fix gaps in pathway annotations. Usually non-metabolic</span>
<span class="c1"># Currently: Tx Reg, Transcription, translation, DNA replication, Cell division, iron-sulfur cluster assembly, proteolysis, dna repair, copper response</span>
<span class="n">TEMP_GO_TERMS</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;GO:0006355&#39;</span><span class="p">:</span> <span class="s1">&#39;Regulation of transcription&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0010468&#39;</span><span class="p">:</span> <span class="s1">&#39;Regulation of transcription&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;GO:0006351&#39;</span><span class="p">:</span> <span class="s1">&#39;Transcription&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0006350&#39;</span><span class="p">:</span> <span class="s1">&#39;Transcription&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;GO:0006412&#39;</span><span class="p">:</span> <span class="s1">&#39;Translation&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0006260&#39;</span><span class="p">:</span> <span class="s1">&#39;DNA replication&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0006457&#39;</span><span class="p">:</span> <span class="s1">&#39;Protein folding&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0045454&#39;</span><span class="p">:</span> <span class="s1">&#39;Redox homeostasis&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;GO:0051301&#39;</span><span class="p">:</span> <span class="s1">&#39;Cell division&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0015288&#39;</span><span class="p">:</span> <span class="s1">&#39;Porin&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;GO:0016226&#39;</span><span class="p">:</span> <span class="s1">&#39;Iron-sulfur cluster assembly&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0006508&#39;</span><span class="p">:</span> <span class="s1">&#39;Proteolysis&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0006281&#39;</span><span class="p">:</span> <span class="s1">&#39;DNA repair&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;GO:0046688&#39;</span><span class="p">:</span> <span class="s1">&#39;Response to copper ion&#39;</span><span class="p">}</span>


<span class="c1"># as residues</span>
<span class="n">AMINO_ACID_RESIDUE_MASSES</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;ALA&#39;</span><span class="p">:</span> <span class="mf">71.03711</span><span class="p">,</span> <span class="s1">&#39;ARG&#39;</span><span class="p">:</span> <span class="mf">156.10111</span><span class="p">,</span> <span class="s1">&#39;ASN&#39;</span><span class="p">:</span> <span class="mf">114.04293</span><span class="p">,</span> <span class="s1">&#39;ASP&#39;</span><span class="p">:</span> <span class="mf">115.02694</span><span class="p">,</span>
    <span class="s1">&#39;CYS&#39;</span><span class="p">:</span> <span class="mf">103.00919</span><span class="p">,</span> <span class="s1">&#39;GLU&#39;</span><span class="p">:</span> <span class="mf">129.04259</span><span class="p">,</span> <span class="s1">&#39;GLN&#39;</span><span class="p">:</span> <span class="mf">128.05858</span><span class="p">,</span> <span class="s1">&#39;GLY&#39;</span><span class="p">:</span> <span class="mf">57.02146</span><span class="p">,</span>
    <span class="s1">&#39;HIS&#39;</span><span class="p">:</span> <span class="mf">137.05891</span><span class="p">,</span> <span class="s1">&#39;ILE&#39;</span><span class="p">:</span> <span class="mf">113.08406</span><span class="p">,</span> <span class="s1">&#39;LEU&#39;</span><span class="p">:</span> <span class="mf">113.08406</span><span class="p">,</span> <span class="s1">&#39;LYS&#39;</span><span class="p">:</span> <span class="mf">128.09496</span><span class="p">,</span>
    <span class="s1">&#39;MET&#39;</span><span class="p">:</span> <span class="mf">131.04049</span><span class="p">,</span> <span class="s1">&#39;PHE&#39;</span><span class="p">:</span> <span class="mf">147.06841</span><span class="p">,</span> <span class="s1">&#39;PRO&#39;</span><span class="p">:</span> <span class="mf">97.05276</span><span class="p">,</span> <span class="s1">&#39;SER&#39;</span><span class="p">:</span> <span class="mf">87.03203</span><span class="p">,</span>
    <span class="s1">&#39;THR&#39;</span><span class="p">:</span> <span class="mf">101.04768</span><span class="p">,</span> <span class="s1">&#39;TRP&#39;</span><span class="p">:</span> <span class="mf">186.07931</span><span class="p">,</span> <span class="s1">&#39;TYR&#39;</span><span class="p">:</span> <span class="mf">163.06333</span><span class="p">,</span> <span class="s1">&#39;VAL&#39;</span><span class="p">:</span> <span class="mf">99.06841</span><span class="p">,</span>
    <span class="s1">&#39;SEL&#39;</span><span class="p">:</span> <span class="mf">150.0379</span><span class="p">,</span> <span class="s1">&#39;TER&#39;</span><span class="p">:</span> <span class="mf">0.0</span>
<span class="p">}</span>


<span class="n">erroneous_monomer_metal_interactions</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;EG11415-MONOMER&quot;</span><span class="p">,</span> <span class="s2">&quot;EG12132-MONOMER&quot;</span><span class="p">,</span>
                                        <span class="s2">&quot;EG12332-MONOMER&quot;</span><span class="p">,</span> <span class="s2">&quot;EG11378-MONOMER&quot;</span><span class="p">,</span> <span class="s2">&quot;G7748-MONOMER&quot;</span><span class="p">,</span>
                                        <span class="s2">&quot;CRR-MONOMER&quot;</span><span class="p">,</span> <span class="s2">&quot;EG11663-MONOMER&quot;</span><span class="p">,</span> <span class="s2">&quot;GLYOXI-MONOMER&quot;</span><span class="p">,</span>
                                        <span class="s2">&quot;EG10697-MONOMER&quot;</span><span class="p">,</span> <span class="s2">&quot;EG12310-MONOMER&quot;</span><span class="p">,</span> <span class="s2">&quot;EG10694-MONOMER&quot;</span><span class="p">,</span> <span class="s2">&quot;EG10698-MONOMER&quot;</span><span class="p">,</span> <span class="c1">#Mn</span>
                                        <span class="s2">&quot;PPENTOMUT-MONOMER&quot;</span><span class="p">,</span> <span class="p">]</span> <span class="c1">#Mn</span>

<span class="k">def</span> <span class="nf">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">cur_pathway_idx</span><span class="p">,</span> <span class="n">pathway_matrix</span><span class="p">,</span> <span class="n">name_list</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">parent_dict</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>

    <span class="k">if</span> <span class="n">parent_dict</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">parent_dict</span> <span class="o">=</span> <span class="p">{}</span>

    <span class="n">cur_pathway_level</span> <span class="o">=</span> <span class="n">level_vector</span><span class="p">[</span><span class="n">cur_pathway_idx</span><span class="p">]</span>

    <span class="k">if</span> <span class="n">cur_pathway_level</span> <span class="o">==</span> <span class="n">level</span><span class="p">:</span>
        <span class="n">parent_dict</span><span class="p">[</span><span class="n">name_list</span><span class="p">[</span><span class="n">cur_pathway_idx</span><span class="p">]]</span> <span class="o">=</span> <span class="n">cur_pathway_level</span>

    <span class="n">parent_slice</span> <span class="o">=</span> <span class="n">pathway_matrix</span><span class="p">[:,</span> <span class="n">cur_pathway_idx</span><span class="p">]</span>
    <span class="n">parent_idxs</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">parent_slice</span> <span class="o">!=</span> <span class="mi">0</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>


    <span class="k">for</span> <span class="n">idx</span> <span class="ow">in</span> <span class="n">parent_idxs</span><span class="p">:</span>

        <span class="n">_</span> <span class="o">=</span> <span class="n">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">idx</span><span class="p">,</span> <span class="n">pathway_matrix</span><span class="p">,</span> <span class="n">name_list</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="p">,</span> <span class="n">parent_dict</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">parent_dict</span>
</pre></div>
</div>
</div>
<section id="Reload-data">
<h2>Reload data<a class="headerlink" href="#Reload-data" title="Link to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_complex_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/raw_complexes.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="c1"># read stoichiometry, cofactors and enzyme_reaction as literal sets</span>
<span class="k">for</span> <span class="n">column</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;stoichiometry&#39;</span><span class="p">,</span> <span class="s1">&#39;cofactors&#39;</span><span class="p">,</span> <span class="s1">&#39;enzyme_reaction&#39;</span><span class="p">]:</span>
    <span class="n">parsed_complex_df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">=</span> <span class="n">parsed_complex_df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">ast</span><span class="o">.</span><span class="n">literal_eval</span><span class="p">)</span>

<span class="n">parsed_protein_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/raw_proteins.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="k">for</span> <span class="n">column</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;cofactors&#39;</span><span class="p">,</span> <span class="s1">&#39;enzyme_reaction&#39;</span><span class="p">,</span> <span class="s1">&#39;metal_features&#39;</span><span class="p">,</span> <span class="s1">&#39;other_features&#39;</span><span class="p">,</span> <span class="s1">&#39;direct_annotations&#39;</span><span class="p">,</span> <span class="s1">&#39;go_annotations&#39;</span><span class="p">]:</span>
    <span class="n">parsed_protein_df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">ast</span><span class="o">.</span><span class="n">literal_eval</span><span class="p">)</span>


<span class="n">parsed_cofactor_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/raw_cofactors.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="k">for</span> <span class="n">column</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;elemental_composition&#39;</span><span class="p">]:</span>
    <span class="n">parsed_cofactor_df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">=</span> <span class="n">parsed_cofactor_df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">ast</span><span class="o">.</span><span class="n">literal_eval</span><span class="p">)</span>

<span class="n">parsed_pathway_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/raw_pathways.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="k">for</span> <span class="n">column</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;parents&#39;</span><span class="p">,</span> <span class="s1">&#39;children&#39;</span><span class="p">]:</span>
    <span class="n">parsed_pathway_df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">=</span> <span class="n">parsed_pathway_df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">ast</span><span class="o">.</span><span class="n">literal_eval</span><span class="p">)</span>

<span class="c1"># add GO &quot;pathways&quot; to pathway df</span>
<span class="n">go_collection</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">already_seen</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">go_collection</span><span class="o">.</span><span class="n">append</span><span class="p">({</span><span class="s1">&#39;id&#39;</span><span class="p">:</span> <span class="s1">&#39;Other functions&#39;</span><span class="p">,</span> <span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;Other functions&#39;</span><span class="p">,</span> <span class="s1">&#39;level&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s1">&#39;parents&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="p">[]})</span>

<span class="k">for</span> <span class="n">go_term</span><span class="p">,</span> <span class="n">go_name</span> <span class="ow">in</span> <span class="n">TEMP_GO_TERMS</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">go_name</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">already_seen</span><span class="p">:</span>
        <span class="n">already_seen</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">go_name</span><span class="p">)</span>
        <span class="n">go_collection</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="s1">&#39;children&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">go_name</span><span class="p">)</span>

        <span class="n">go_name_child</span> <span class="o">=</span> <span class="n">go_name</span> <span class="o">+</span> <span class="s1">&#39; (child)&#39;</span>

        <span class="n">go_collection</span><span class="o">.</span><span class="n">append</span><span class="p">({</span><span class="s1">&#39;id&#39;</span><span class="p">:</span> <span class="n">go_name</span><span class="p">,</span> <span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="n">go_name</span><span class="p">,</span> <span class="s1">&#39;level&#39;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span> <span class="s1">&#39;parents&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Other functions&#39;</span><span class="p">],</span> <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="p">[</span><span class="n">go_name_child</span><span class="p">]})</span>
        <span class="n">go_collection</span><span class="o">.</span><span class="n">append</span><span class="p">({</span><span class="s1">&#39;id&#39;</span><span class="p">:</span> <span class="n">go_name_child</span><span class="p">,</span> <span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="n">go_name_child</span><span class="p">,</span> <span class="s1">&#39;level&#39;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> <span class="s1">&#39;parents&#39;</span><span class="p">:</span> <span class="p">[</span><span class="n">go_name</span><span class="p">],</span> <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="p">[]})</span>

<span class="n">go_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">go_collection</span><span class="p">)</span>

<span class="n">parsed_pathway_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">parsed_pathway_df</span><span class="p">,</span> <span class="n">go_df</span><span class="p">],</span> <span class="n">ignore_index</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="c1"># parsed_pathway_df</span>


<span class="c1"># add membrane protein areas</span>
<span class="n">membrane_monomers</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/monomer_vs_area.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">membrane_complexes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/complex_vs_area.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="c1"># rename complex column in membrane_complexes to id</span>
<span class="n">membrane_complexes</span> <span class="o">=</span> <span class="n">membrane_complexes</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;complex&#39;</span><span class="p">:</span> <span class="s1">&#39;id&#39;</span><span class="p">})</span>

<span class="c1"># concat, pick out id and area_trans</span>
<span class="n">membrane_proteins</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">membrane_monomers</span><span class="p">,</span> <span class="n">membrane_complexes</span><span class="p">],</span> <span class="n">ignore_index</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">membrane_proteins</span> <span class="o">=</span> <span class="n">membrane_proteins</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">,</span> <span class="s1">&#39;area_trans&#39;</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</section>
</section>
<section id="Data-processing-into-final-tables">
<h1>Data processing into final tables<a class="headerlink" href="#Data-processing-into-final-tables" title="Link to this heading"></a></h1>
<section id="Specific-adjustments">
<h2>Specific adjustments<a class="headerlink" href="#Specific-adjustments" title="Link to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># removing specific troublesome interactions that have been discovered in the data</span>
<span class="c1"># should not be in model</span>


<span class="c1"># remove metal features from EG11415-MONOMER in parsed_protein_df\</span>
<span class="k">for</span> <span class="n">protein</span> <span class="ow">in</span> <span class="n">erroneous_monomer_metal_interactions</span><span class="p">:</span>
    <span class="n">prot_idx</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">protein</span><span class="p">)[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">prot_idx</span><span class="p">,</span> <span class="s1">&#39;metal_features&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[]</span>

<span class="c1"># classify folE gene use as cofactor production (THF)</span>
<span class="n">pathway_idx</span> <span class="o">=</span> <span class="n">parsed_pathway_df</span><span class="p">[</span><span class="n">parsed_pathway_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;6-HM-Dihydropterin-PP-Biosynthesis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">pathway_idx</span><span class="p">,</span> <span class="s1">&#39;parents&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Cofactor-Biosynthesis&#39;</span><span class="p">]</span>

<span class="n">protein_idx</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;GTP-CYCLOHYDRO-I-MONOMER&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">protein_idx</span><span class="p">,</span> <span class="s1">&#39;direct_annotations&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="nb">set</span><span class="p">([</span><span class="s1">&#39;PWY-6147&#39;</span><span class="p">])</span>
<br/><br/></pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">protein</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&#39;PPENTOMUT-MONOMER&#39;
</pre></div></div>
</div>
</section>
<section id="Add-membrane-protein-areas-to-parsed_protein_df">
<h2>Add membrane protein areas to parsed_protein_df<a class="headerlink" href="#Add-membrane-protein-areas-to-parsed_protein_df" title="Link to this heading"></a></h2>
</section>
<section id="Process-raw-EcoCyc-annotations-into-standard-EcoCyc-names">
<h2>Process raw EcoCyc annotations into standard EcoCyc names<a class="headerlink" href="#Process-raw-EcoCyc-annotations-into-standard-EcoCyc-names" title="Link to this heading"></a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><br/><span></span><span class="c1"># remove all \ characters from keys in ALLOWED_METAL_NAMES</span>
<span class="n">NON_REGEX_METAL</span> <span class="o">=</span> <span class="p">{</span><span class="n">key</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\\</span><span class="s1">&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">):</span> <span class="n">value</span> <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">ALLOWED_METAL_NAMES</span><span class="o">.</span><span class="n">items</span><span class="p">()}</span>

<span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
<span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">object</span><span class="p">)</span>

<span class="n">metal_pattern</span> <span class="o">=</span> <span class="s1">&#39;|&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">ALLOWED_METAL_NAMES</span><span class="o">.</span><span class="n">keys</span><span class="p">())</span>
<span class="n">metal_regex</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">compile</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;((</span><span class="si">{</span><span class="n">metal_pattern</span><span class="si">}</span><span class="s1">)(\s\d[\.,;]|[\.,;]|\s\())&#39;</span><span class="p">)</span>


<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>

    <span class="n">metal_binding</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;metal_features&#39;</span><span class="p">]</span>

    <span class="n">metal_count_dict</span> <span class="o">=</span> <span class="p">{}</span>
    <span class="n">existing_matches</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>

    <span class="k">for</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">metal_binding</span><span class="p">:</span>
        <span class="n">matches</span> <span class="o">=</span> <span class="n">metal_regex</span><span class="o">.</span><span class="n">search</span><span class="p">(</span><span class="n">feature</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">matches</span><span class="p">:</span>
            <span class="n">metal</span> <span class="o">=</span> <span class="n">matches</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="mi">0</span><span class="p">)[:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>

            <span class="c1"># eliminate duplicates</span>
            <span class="k">if</span> <span class="n">metal</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">existing_matches</span><span class="p">:</span>

                <span class="n">existing_matches</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">metal</span><span class="p">)</span>

                <span class="k">if</span> <span class="s1">&#39;heme&#39;</span> <span class="ow">in</span> <span class="n">feature</span><span class="p">:</span>
                    <span class="n">metal</span> <span class="o">=</span> <span class="n">metal</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;Iron&#39;</span><span class="p">,</span> <span class="s1">&#39;heme&#39;</span><span class="p">)</span>
                <span class="k">if</span> <span class="s1">&#39;alamin&#39;</span> <span class="ow">in</span> <span class="n">feature</span><span class="p">:</span>
                    <span class="n">metal</span> <span class="o">=</span> <span class="n">metal</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;Cobalt&#39;</span><span class="p">,</span> <span class="s1">&#39;Cobalamin&#39;</span><span class="p">)</span>

                <span class="c1"># check if last char of metal is a number, then crop</span>
                <span class="k">if</span> <span class="n">metal</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
                    <span class="n">metal</span> <span class="o">=</span> <span class="n">metal</span><span class="p">[:</span><span class="o">-</span><span class="mi">2</span><span class="p">]</span>

                <span class="n">metal</span> <span class="o">=</span> <span class="n">metal</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span>

                <span class="c1"># replace metal name with allowed metal name</span>
                <span class="n">metal</span> <span class="o">=</span> <span class="n">NON_REGEX_METAL</span><span class="p">[</span><span class="n">metal</span><span class="p">]</span>

                <span class="k">if</span> <span class="n">metal</span> <span class="ow">in</span> <span class="n">metal_count_dict</span><span class="p">:</span>
                    <span class="n">metal_count_dict</span><span class="p">[</span><span class="n">metal</span><span class="p">]</span> <span class="o">+=</span> <span class="mi">1</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">metal_count_dict</span><span class="p">[</span><span class="n">metal</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>

        <span class="k">else</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;No match for </span><span class="si">{</span><span class="n">feature</span><span class="si">}</span><span class="s1"> in </span><span class="si">{</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s2">&quot;id&quot;</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>



    <span class="c1"># EXCEPTIONS</span>
    <span class="c1"># if both magnesium and manganese are present, replace with magnesium</span>
    <span class="c1"># TODO remove when using UniProt data. Ecocyc data is not as reliable</span>
    <span class="k">if</span> <span class="s1">&#39;MG+2&#39;</span> <span class="ow">in</span> <span class="n">metal_count_dict</span> <span class="ow">and</span> <span class="s1">&#39;MN+2&#39;</span> <span class="ow">in</span> <span class="n">metal_count_dict</span> <span class="ow">and</span> <span class="n">metal_count_dict</span><span class="p">[</span><span class="s1">&#39;MG+2&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">metal_count_dict</span><span class="p">[</span><span class="s1">&#39;MN+2&#39;</span><span class="p">]:</span>
        <span class="k">del</span> <span class="n">metal_count_dict</span><span class="p">[</span><span class="s1">&#39;MN+2&#39;</span><span class="p">]</span>
    <span class="c1"># same with cobalt</span>
    <span class="k">if</span> <span class="s1">&#39;CO+2&#39;</span> <span class="ow">in</span> <span class="n">metal_count_dict</span> <span class="ow">and</span> <span class="s1">&#39;MG+2&#39;</span> <span class="ow">in</span> <span class="n">metal_count_dict</span> <span class="ow">and</span> <span class="n">metal_count_dict</span><span class="p">[</span><span class="s1">&#39;CO+2&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">metal_count_dict</span><span class="p">[</span><span class="s1">&#39;MG+2&#39;</span><span class="p">]:</span>
        <span class="k">del</span> <span class="n">metal_count_dict</span><span class="p">[</span><span class="s1">&#39;CO+2&#39;</span><span class="p">]</span>
    <span class="k">elif</span> <span class="s1">&#39;CO+2&#39;</span> <span class="ow">in</span> <span class="n">metal_count_dict</span><span class="p">:</span>
        <span class="n">metal_count_dict</span><span class="p">[</span><span class="s1">&#39;MG+2&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">metal_count_dict</span><span class="p">[</span><span class="s1">&#39;CO+2&#39;</span><span class="p">]</span>
        <span class="k">del</span> <span class="n">metal_count_dict</span><span class="p">[</span><span class="s1">&#39;CO+2&#39;</span><span class="p">]</span>

    <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">metal_count_dict</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
No match for UniProt: Magnesium or manganese. in 3-ISOPROPYLMALDEHYDROG-MONOMER
No match for conserved, Fe(III) binding motif in BASS-MONOMER
No match for predicted heme d ligand in CYDA-MONOMER
No match for UniProt: Zn(2+); catalytic. in CYTDEAM-MONOMER
No match for UniProt: Fe(2+); catalytic. in CYTDEAM-MONOMER
No match for The amino-terminus of ClpA contains a Zinc binding site. in EG10156-MONOMER
No match for The active-site magnesium ion is coordinated by three aspartate residues (401, 403, 555). Two of them form part of
the PDXD active-site motif. in EG10238-MONOMER
No match for Divalent magnesium ions are chelated by three aspartate residues, two in the conserved DPD sequence (345, 347)
and one in the conserved EGYMD sequence (269). in EG10239-MONOMER
No match for Based on crystal structures, Glu-265 and Asp-309 coordinate a divalent cation. in EG10239-MONOMER
No match for These residues are thought to coordinate the one or two divalent magnesium ions required for the
gyrase cleavage-religation reaction. in EG10424-MONOMER
No match for Mg&lt;sup&gt;2+&lt;/sup&gt; binding site in EG10599-MONOMER
No match for This metal-binding motif was identified in a crystal structure [&lt;a href=&#34;http://www.ncbi.nlm.nih.gov/pubmed/16885166&#34;&gt;Ito06&lt;/a&gt;]. in EG10696-MONOMER
No match for Asp-10 and Asp-134 coordinate a divalent manganese ion which, when present may inhibit enzymatic activity. in EG10860-MONOMER
No match for Asn-44 and Glu-48 coordinate magnesium when it is present. in EG10860-MONOMER
No match for Asp-10, Glu-48 and Asp-70 coordinate a divalent manganese ion which is required for catalytic activity. in EG10860-MONOMER
No match for binds Zn&lt;sup&gt;2+&lt;/sup&gt; and Cd&lt;sup&gt;2+&lt;/sup&gt; with high affinity and Fe&lt;sup&gt;2+&lt;/sup&gt; with lower affinity in EG11167-MONOMER
No match for binds Fe&lt;sup&gt;2+&lt;/sup&gt; with high affinity in EG11167-MONOMER
No match for Zn&lt;sup&gt;2+&lt;/sup&gt; binding in EG11918-MONOMER
No match for predicted zinc binding site in EG11955-MONOMER
No match for coordination of zinc in EG12045-MONOMER
No match for cluster of cysteines that may be involved in binding zinc; possible zinc finger domains in EG12045-MONOMER
No match for conserved histidines located in periplasmic regions on either side of the WWD region; ligands for heme in holoCcmE in EG12054-MONOMER
No match for Ligands to &lt;i&gt;b&lt;/i&gt; heme in EG12054-MONOMER
No match for UniProt: Zn(2+); catalytic. in EG12097-MONOMER
No match for Calcium in EG12244-MONOMER
No match for substrate binding oxidation site: Cu(I) &amp;rarr; Cu(II) in EG12318-MONOMER
No match for labile or regulatory copper binding site; functions as an electron transfer site not a substrate reaction site in EG12318-MONOMER
No match for type I, blue copper site in EG12318-MONOMER
No match for type III, coupled binuclear Cu site in EG12318-MONOMER
No match for a third Met ligand is likely located in the region 380-403; substrate binding oxidation site: Cu(I) &amp;rarr; Cu(II) in EG12318-MONOMER
No match for type II, non-blue copper site in EG12318-MONOMER
No match for conserved in members of the TEES family of bacterial GTPases; implicated in potassium binding in &lt;i&gt;S. thermophilus&lt;/i&gt; nFeoB in FEOB-MONOMER
No match for 4Fe:4S coordination site in FUM-FE-S
No match for 3Fe:4S coordination site in FUM-FE-S
No match for 2Fe:2S coordination site in FUM-FE-S
No match for UniProt: Copper(1+). in G6263-MONOMER
No match for Ag(I) binding, symmetrical binding of two Ag(I) ions at the CusS dimer interface; F43, H176 and H42&#39; co-ordinate one Ag(I) ion while F43&#39;, H176&#39; and H42 coordinate the second in G6318-MONOMER
No match for Ag(I) binding; involved in binding one Ag(I) ion per CusS monomer in G6318-MONOMER
No match for Each of the cysteines in this conserved iron-binding motif is individually required
for MiaB enzymatic activity. in G6364-MONOMER
No match for potential metal binding site in G6527-MONOMER
No match for potential Zn(2) binding site in G6527-MONOMER
No match for enzyme shows preference for Fe&lt;sup&gt;2+&lt;/sup&gt;, Ni&lt;sup&gt;2+&lt;/sup&gt; and Co&lt;sup&gt;2+&lt;/sup&gt; over Zn&lt;sup&gt;2+&lt;/sup&gt; in G6530-MONOMER
No match for Cu&lt;sup&gt;2+&lt;/sup&gt;-binding site in G7014-MONOMER
No match for potential, conserved copper binding site in G7139-MONOMER
No match for zinc binding site in G7430-MONOMER
No match for zinc binding in G7458-MONOMER
No match for Cys ligands for the nickel ion of the Ni-Fe active site in HYAB-MONOMER
No match for heme &lt;i&gt;b&lt;/i&gt; ligand in HYAC-MONOMER
No match for ligand single calcium ions bound to each monomer of the homodimer; R167 and Ser152 ligands are from monomer A, Ser172 ligand is from monomer B in MONOMER0-341
No match for predicted copper ligands in NADH-DHII-MONOMER
No match for UniProt: Magnesium or cobalt. in NAG1P-URIDYLTRANS-MONOMER
No match for coordinates a molybdenum hydroxo ligand in NARG-MONOMER
No match for [4Fe-4S] coordination; FS1 in NARY-MONOMER
No match for [3Fe-4S] coordination, FS4 in NARY-MONOMER
No match for [4Fe-4S] coordination, FS2 in NARY-MONOMER
No match for [4Fe-4S] corodination, FS3 in NARY-MONOMER
No match for implicated in Na&lt;sup&gt;+&lt;/sup&gt; binding above pH7 in NHAA-MONOMER
No match for These residues make up the zinc-binding site. Histidines 79 and 238 appear to directly coordinate zinc, with
additional coordination provided by one of the two other residues. All four are essential for catalytic
activity [&lt;a href=&#34;http://www.ncbi.nlm.nih.gov/pubmed/11148046&#34;&gt;Jackman01&lt;/a&gt;]. Extended X-ray absorption fine structure spectroscopy suggests that aspartate-246
coordinates zinc with the aid of a solvent molecule [&lt;a href=&#34;http://www.ncbi.nlm.nih.gov/pubmed/12620676&#34;&gt;McClure03&lt;/a&gt;]. in UDPACYLGLCNACDEACETYL-MONOMER
No match for methionine residues shown to coordinate Cu(I) in the periplasmic binding cleft in YBDE-MONOMER
No match for residues constituting a (Met&lt;sub&gt;2&lt;/sub&gt; O/N) metal binding motif in YBDE-MONOMER
No match for methionine pair located in transmembrane region of protein; suggested to form binding site as part of a methionine shuttle for movement of substrate in YBDE-MONOMER
No match for methionine pair located in periplasmic region of protein; suggested to form binding site as part of a methionine shuttle for movement of substrate in YBDE-MONOMER
No match for implicated in metal (Zn&lt;sup&gt;2+&lt;/sup&gt; and Cd&lt;sup&gt;2+&lt;/sup&gt;) coordination in YIIP-MONOMER
No match for zinc-coordinating residues in ZNUA-MONOMER
No match for implicated in a second zinc binding site in ZNUA-MONOMER
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># manual corrections in data</span>
<span class="n">correction_idx</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;3-OXOACYL-ACP-REDUCT-MONOMER&#39;</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>

<span class="c1"># remove cofactors</span>
<span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">correction_idx</span><span class="p">,</span> <span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">{}</span>

<span class="n">correction_idx</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;CARBPSYN-LARGE&#39;</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>

<span class="c1"># remove cofactors</span>
<span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">correction_idx</span><span class="p">,</span> <span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span>
<span class="n">mn_cofactor_count</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">correction_idx</span><span class="p">,</span> <span class="s1">&#39;metal_features_processed&#39;</span><span class="p">][</span><span class="s1">&#39;MN+2&#39;</span><span class="p">]</span>
<span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">correction_idx</span><span class="p">,</span> <span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;MG+2&#39;</span><span class="p">:</span> <span class="n">mn_cofactor_count</span><span class="p">}</span>

<span class="c1"># dps</span>
<span class="n">correction_idx</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;EG11415-MONOMER&#39;</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>
<span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">correction_idx</span><span class="p">,</span> <span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">{}</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_protein_df</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;metal_features&#39;</span><span class="p">])</span>
<span class="n">parsed_protein_df</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>id</th>
      <th>common_name</th>
      <th>seq</th>
      <th>enzyme_reaction</th>
      <th>cofactors</th>
      <th>other_features</th>
      <th>direct_annotations</th>
      <th>go_annotations</th>
      <th>uniprot_id</th>
      <th>metal_features_processed</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1-ACYLGLYCEROL-3-P-ACYLTRANSFER-MONOMER</td>
      <td>1-acylglycerol-3-phosphate &lt;i&gt;O&lt;/i&gt;-acyltransf...</td>
      <td>MLYIFRLIITVIYSILVCVFGSIYCLFSPRNPKHVATFGHMFGRLA...</td>
      <td>{ENZRXN0-7995, ENZRXN0-7462, ENZRXN0-7991, 1-A...</td>
      <td>{}</td>
      <td>[]</td>
      <td>{PWY-5667, PWY0-1319}</td>
      <td>{GO:0006629, GO:0016020, GO:0016024, GO:000384...</td>
      <td>P26647</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1-PFK-MONOMER</td>
      <td>1-phosphofructokinase</td>
      <td>MSRRVATITLNPAYDLVGFCPEIERGEVNLVKTTGLHAAGKGINVA...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[ATP]</td>
      <td>{PWY0-1314}</td>
      <td>{GO:0000166, GO:0042803, GO:0016301, GO:000844...</td>
      <td>P0AEW9</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2-DEHYDROPANTOATE-REDUCT-MONOMER</td>
      <td>2-dehydropantoate 2-reductase</td>
      <td>MKITVLGCGALGQLWLTALCKQGHEVQGWLRVPQPYCSVNLVETDG...</td>
      <td>{2-DEHYDROPANTOATE-REDUCT-ENZRXN}</td>
      <td>{}</td>
      <td>[NADP]</td>
      <td>{PANTO-PWY}</td>
      <td>{GO:0005737, GO:0016491, GO:0050661, GO:001661...</td>
      <td>P0A9J4</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2-ISOPROPYLMALATESYN-MONOMER</td>
      <td>2-isopropylmalate synthase</td>
      <td>MSQQVIIFDTTLRDGEQALQASLSVKEKLQIALALERMGVDVMEVG...</td>
      <td>{ENZRXN0-6250}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{LEUSYN-PWY}</td>
      <td>{GO:0005737, GO:0030145, GO:0003985, GO:000385...</td>
      <td>P09151</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>4</th>
      <td>2-OCTAPRENYL-METHOXY-BENZOQ-METH-MONOMER</td>
      <td>bifunctional 2-octaprenyl-6-methoxy-1,4-benzoq...</td>
      <td>MVDKSQETTHFGFQTVAKEQKADMVAHVFHSVASKYDVMNDLMSFG...</td>
      <td>{2-OCTAPRENYL-METHOXY-BENZOQ-METH-ENZRXN, ADOM...</td>
      <td>{}</td>
      <td>[]</td>
      <td>{PWY-6708, MENAQUINONESYN-PWY}</td>
      <td>{GO:0008757, GO:0005737, GO:0102094, GO:010202...</td>
      <td>P0A887</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>4429</th>
      <td>YTFR-MONOMER</td>
      <td>galactofuranose ABC transporter putative ATP b...</td>
      <td>MTTDQHQEILRTEGLSKFFPGVKALDNVDFSLRRGEIMALLGENGA...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[ATP]</td>
      <td>{}</td>
      <td>{GO:0140271, GO:0000166, GO:0008643, GO:004287...</td>
      <td>Q6BEX0</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>4430</th>
      <td>YTFT-MONOMER</td>
      <td>galactofuranose ABC transporter putative membr...</td>
      <td>MMPQSLPDTTTPKRRFRWPTGMPQLVALLLVLLVDSLVAPHFWQVV...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{}</td>
      <td>{GO:0140271, GO:0008643, GO:0005515, GO:004319...</td>
      <td>P39328</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>4431</th>
      <td>ZNUA-MONOMER</td>
      <td>Zn&lt;sup&gt;2+&lt;/sup&gt; ABC transporter periplasmic bi...</td>
      <td>MLHKKTLLFAALSAALWGGATQAADAAVVASLKPVGFIASAIADGV...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{}</td>
      <td>{GO:0008270, GO:0007155, GO:0030288, GO:001602...</td>
      <td>P39172</td>
      <td>{'ZN+2': 1}</td>
    </tr>
    <tr>
      <th>4432</th>
      <td>ZNUB-MONOMER</td>
      <td>Zn&lt;sup&gt;2+&lt;/sup&gt; ABC transporter membrane subunit</td>
      <td>MIELLFPGWLAGIMLACAAGPLGSFVVWRRMSYFGDTLAHASLLGV...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{}</td>
      <td>{GO:0043190, GO:0000006, GO:0016020, GO:000682...</td>
      <td>P39832</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>4433</th>
      <td>ZNUC-MONOMER</td>
      <td>Zn&lt;sup&gt;2+&lt;/sup&gt; ABC transporter ATP binding su...</td>
      <td>MTSLVSLENVSVSFGQRRVLSDVSLELKPGKILTLLGPNGAGKSTL...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[ATP]</td>
      <td>{}</td>
      <td>{GO:0000166, GO:0015633, GO:0043190, GO:001602...</td>
      <td>P0A9X1</td>
      <td>{}</td>
    </tr>
  </tbody>
</table>
<p>4434 rows × 10 columns</p>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;other_features_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
<span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;other_features_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;other_features_processed&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">object</span><span class="p">)</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>

    <span class="n">other_features</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;other_features&#39;</span><span class="p">]</span>

    <span class="n">other_feature_count_dict</span> <span class="o">=</span> <span class="p">{}</span>
    <span class="n">existing_matches</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>

    <span class="k">for</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">other_features</span><span class="p">:</span>

        <span class="c1"># eliminate duplicates</span>
        <span class="k">if</span> <span class="n">feature</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">existing_matches</span><span class="p">:</span>

            <span class="n">existing_matches</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">feature</span><span class="p">)</span>

            <span class="k">if</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">ACCEPTED_OTHER_FEATURES</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">other_feature_count_dict</span><span class="p">:</span>
                    <span class="n">other_feature_count_dict</span><span class="p">[</span><span class="n">feature</span><span class="p">]</span> <span class="o">+=</span> <span class="mi">1</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">other_feature_count_dict</span><span class="p">[</span><span class="n">feature</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>

    <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;other_features_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">other_feature_count_dict</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># processed go_annotations. if no pathways exist for monomer, use go annotations</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>

    <span class="n">cur_go_annotations</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;go_annotations&#39;</span><span class="p">]</span>
    <span class="n">cur_pathways</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;direct_annotations&#39;</span><span class="p">]</span>

    <span class="n">replacement_pathways</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>

    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">cur_pathways</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>

        <span class="k">for</span> <span class="n">go_term</span> <span class="ow">in</span> <span class="n">TEMP_GO_TERMS</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">go_annotation</span> <span class="ow">in</span> <span class="n">cur_go_annotations</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">go_term</span> <span class="o">==</span> <span class="n">go_annotation</span><span class="p">:</span>
                    <span class="n">replacement_pathways</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">TEMP_GO_TERMS</span><span class="p">[</span><span class="n">go_term</span><span class="p">]</span> <span class="o">+</span> <span class="s1">&#39; (child)&#39;</span><span class="p">)</span>

        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">replacement_pathways</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s2">&quot;id&quot;</span><span class="p">],</span> <span class="nb">set</span><span class="p">([</span><span class="n">replacement_pathways</span><span class="p">[</span><span class="mi">0</span><span class="p">]]))</span>
            <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;direct_annotations&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="nb">set</span><span class="p">([</span><span class="n">replacement_pathways</span><span class="p">[</span><span class="mi">0</span><span class="p">]])</span>


<span class="c1"># parsed_protein_df[parsed_protein_df[&#39;id&#39;] == &#39;PD00197&#39;]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CYDC-MONOMER {&#39;Redox homeostasis (child)&#39;}
CYDD-MONOMER {&#39;Redox homeostasis (child)&#39;}
DSBC-MONOMER {&#39;Response to copper ion (child)&#39;}
EG10013-MONOMER {&#39;Proteolysis (child)&#39;}
EG10037-MONOMER {&#39;DNA repair (child)&#39;}
EG10087-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10108-MONOMER {&#39;Redox homeostasis (child)&#39;}
EG10116-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10125-MONOMER {&#39;Transcription (child)&#39;}
EG10126-MONOMER {&#39;Porin (child)&#39;}
EG10143-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10156-MONOMER {&#39;Proteolysis (child)&#39;}
EG10158-MONOMER {&#39;Proteolysis (child)&#39;}
EG10159-MONOMER {&#39;Protein folding (child)&#39;}
EG10204-MONOMER {&#39;DNA replication (child)&#39;}
EG10212-MONOMER {&#39;Proteolysis (child)&#39;}
EG10218-MONOMER {&#39;Cell division (child)&#39;}
EG10226-MONOMER {&#39;Transcription (child)&#39;}
EG10227-MONOMER {&#39;Cell division (child)&#39;}
EG10228-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10229-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10230-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10236-MONOMER {&#39;DNA replication (child)&#39;}
EG10237-MONOMER {&#39;DNA replication (child)&#39;}
EG10238-MONOMER {&#39;DNA replication (child)&#39;}
EG10239-MONOMER {&#39;DNA replication (child)&#39;}
EG10240-MONOMER {&#39;DNA replication (child)&#39;}
EG10241-MONOMER {&#39;DNA replication (child)&#39;}
EG10242-MONOMER {&#39;DNA replication (child)&#39;}
EG10243-MONOMER {&#39;DNA replication (child)&#39;}
EG10244-MONOMER {&#39;DNA replication (child)&#39;}
EG10245-MONOMER {&#39;DNA replication (child)&#39;}
EG10286-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10320-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10329-MONOMER {&#39;DNA repair (child)&#39;}
EG10335-MONOMER {&#39;Translation (child)&#39;}
EG10339-MONOMER {&#39;Cell division (child)&#39;}
EG10342-MONOMER {&#39;Cell division (child)&#39;}
EG10347-MONOMER {&#39;Cell division (child)&#39;}
EG10360-MONOMER {&#39;Translation (child)&#39;}
EG10374-MONOMER {&#39;Proteolysis (child)&#39;}
EG10397-MONOMER {&#39;Proteolysis (child)&#39;}
EG10416-MONOMER {&#39;Protein folding (child)&#39;}
EG10423-MONOMER {&#39;Transcription (child)&#39;}
EG10424-MONOMER {&#39;Transcription (child)&#39;}
EG10438-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10439-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10442-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10443-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10455-MONOMER {&#39;Protein folding (child)&#39;}
EG10461-MONOMER {&#39;Protein folding (child)&#39;}
EG10462-MONOMER {&#39;Proteolysis (child)&#39;}
EG10463-MONOMER {&#39;Protein folding (child)&#39;}
EG10466-MONOMER {&#39;Transcription (child)&#39;}
EG10467-MONOMER {&#39;Transcription (child)&#39;}
EG10471-MONOMER {&#39;Proteolysis (child)&#39;}
EG10488-MONOMER {&#39;Proteolysis (child)&#39;}
EG10490-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10504-MONOMER {&#39;Translation (child)&#39;}
EG10505-MONOMER {&#39;Translation (child)&#39;}
EG10506-MONOMER {&#39;Translation (child)&#39;}
EG10528-MONOMER {&#39;Porin (child)&#39;}
EG10529-MONOMER {&#39;Translation (child)&#39;}
EG10530-MONOMER {&#39;Proteolysis (child)&#39;}
EG10534-MONOMER {&#39;DNA replication (child)&#39;}
EG10535-MONOMER {&#39;Proteolysis (child)&#39;}
EG10542-MONOMER {&#39;Proteolysis (child)&#39;}
EG10570-MONOMER {&#39;Proteolysis (child)&#39;}
EG10571-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10580-MONOMER {&#39;Proteolysis (child)&#39;}
EG10596-MONOMER {&#39;Cell division (child)&#39;}
EG10597-MONOMER {&#39;Cell division (child)&#39;}
EG10598-MONOMER {&#39;Cell division (child)&#39;}
EG10599-MONOMER {&#39;Protein folding (child)&#39;}
EG10600-MONOMER {&#39;Protein folding (child)&#39;}
EG10603-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10611-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
EG10618-MONOMER {&#39;DNA replication (child)&#39;}
EG10624-MONOMER {&#39;DNA repair (child)&#39;}
EG10625-MONOMER {&#39;DNA repair (child)&#39;}
EG10626-MONOMER {&#39;DNA replication (child)&#39;}
EG10627-MONOMER {&#39;DNA repair (child)&#39;}
EG10651-MONOMER {&#39;DNA repair (child)&#39;}
EG10656-MONOMER {&#39;Transcription (child)&#39;}
EG10662-MONOMER {&#39;DNA repair (child)&#39;}
EG10666-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10668-MONOMER {&#39;DNA repair (child)&#39;}
EG10669-MONOMER {&#39;Porin (child)&#39;}
EG10670-MONOMER {&#39;Porin (child)&#39;}
EG10671-MONOMER {&#39;Porin (child)&#39;}
EG10673-MONOMER {&#39;Proteolysis (child)&#39;}
EG10684-MONOMER {&#39;Cell division (child)&#39;}
EG10694-MONOMER {&#39;Transcription (child)&#39;}
EG10696-MONOMER {&#39;Proteolysis (child)&#39;}
EG10697-MONOMER {&#39;Proteolysis (child)&#39;}
EG10698-MONOMER {&#39;Proteolysis (child)&#39;}
EG10715-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10736-MONOMER {&#39;DNA repair (child)&#39;}
EG10741-MONOMER {&#39;Proteolysis (child)&#39;}
EG10746-MONOMER {&#39;DNA replication (child)&#39;}
EG10747-MONOMER {&#39;DNA repair (child)&#39;}
EG10757-MONOMER {&#39;Protein folding (child)&#39;}
EG10758-MONOMER {&#39;Protein folding (child)&#39;}
EG10760-MONOMER {&#39;Proteolysis (child)&#39;}
EG10761-MONOMER {&#39;Translation (child)&#39;}
EG10762-MONOMER {&#39;Translation (child)&#39;}
EG10763-MONOMER {&#39;DNA replication (child)&#39;}
EG10764-MONOMER {&#39;DNA replication (child)&#39;}
EG10776-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10777-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10785-MONOMER {&#39;Translation (child)&#39;}
EG10786-MONOMER {&#39;Proteolysis (child)&#39;}
EG10823-MONOMER {&#39;DNA repair (child)&#39;}
EG10824-MONOMER {&#39;DNA repair (child)&#39;}
EG10825-MONOMER {&#39;DNA repair (child)&#39;}
EG10826-MONOMER {&#39;DNA repair (child)&#39;}
EG10828-MONOMER {&#39;DNA replication (child)&#39;}
EG10829-MONOMER {&#39;DNA repair (child)&#39;}
EG10830-MONOMER {&#39;DNA repair (child)&#39;}
EG10831-MONOMER {&#39;DNA repair (child)&#39;}
EG10832-MONOMER {&#39;DNA repair (child)&#39;}
EG10833-MONOMER {&#39;DNA replication (child)&#39;}
EG10834-MONOMER {&#39;DNA repair (child)&#39;}
EG10836-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10837-MONOMER {&#39;DNA replication (child)&#39;}
EG10839-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10852-MONOMER {&#39;Translation (child)&#39;}
EG10857-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10864-MONOMER {&#39;Translation (child)&#39;}
EG10865-MONOMER {&#39;Translation (child)&#39;}
EG10866-MONOMER {&#39;Translation (child)&#39;}
EG10867-MONOMER {&#39;Translation (child)&#39;}
EG10868-MONOMER {&#39;Translation (child)&#39;}
EG10869-MONOMER {&#39;Translation (child)&#39;}
EG10870-MONOMER {&#39;Translation (child)&#39;}
EG10871-MONOMER {&#39;Translation (child)&#39;}
EG10872-MONOMER {&#39;Translation (child)&#39;}
EG10873-MONOMER {&#39;Translation (child)&#39;}
EG10874-MONOMER {&#39;Translation (child)&#39;}
EG10875-MONOMER {&#39;Translation (child)&#39;}
EG10876-MONOMER {&#39;Translation (child)&#39;}
EG10877-MONOMER {&#39;Translation (child)&#39;}
EG10878-MONOMER {&#39;Translation (child)&#39;}
EG10879-MONOMER {&#39;Translation (child)&#39;}
EG10880-MONOMER {&#39;Translation (child)&#39;}
EG10881-MONOMER {&#39;Translation (child)&#39;}
EG10882-MONOMER {&#39;Translation (child)&#39;}
EG10883-MONOMER {&#39;Translation (child)&#39;}
EG10884-MONOMER {&#39;Translation (child)&#39;}
EG10885-MONOMER {&#39;Translation (child)&#39;}
EG10886-MONOMER {&#39;Translation (child)&#39;}
EG10887-MONOMER {&#39;Translation (child)&#39;}
EG10888-MONOMER {&#39;Translation (child)&#39;}
EG10889-MONOMER {&#39;Translation (child)&#39;}
EG10890-MONOMER {&#39;Translation (child)&#39;}
EG10891-MONOMER {&#39;Translation (child)&#39;}
EG10892-MONOMER {&#39;Translation (child)&#39;}
EG10893-MONOMER {&#39;Transcription (child)&#39;}
EG10899-MONOMER {&#39;Transcription (child)&#39;}
EG10900-MONOMER {&#39;Translation (child)&#39;}
EG10901-MONOMER {&#39;Translation (child)&#39;}
EG10902-MONOMER {&#39;Translation (child)&#39;}
EG10903-MONOMER {&#39;Translation (child)&#39;}
EG10904-MONOMER {&#39;Translation (child)&#39;}
EG10905-MONOMER {&#39;Translation (child)&#39;}
EG10906-MONOMER {&#39;Translation (child)&#39;}
EG10907-MONOMER {&#39;Translation (child)&#39;}
EG10908-MONOMER {&#39;Translation (child)&#39;}
EG10909-MONOMER {&#39;Translation (child)&#39;}
EG10910-MONOMER {&#39;Translation (child)&#39;}
EG10911-MONOMER {&#39;Translation (child)&#39;}
EG10912-MONOMER {&#39;Translation (child)&#39;}
EG10913-MONOMER {&#39;Translation (child)&#39;}
EG10914-MONOMER {&#39;Translation (child)&#39;}
EG10915-MONOMER {&#39;Translation (child)&#39;}
EG10916-MONOMER {&#39;Translation (child)&#39;}
EG10917-MONOMER {&#39;Translation (child)&#39;}
EG10918-MONOMER {&#39;Translation (child)&#39;}
EG10919-MONOMER {&#39;Translation (child)&#39;}
EG10920-MONOMER {&#39;Translation (child)&#39;}
EG10923-MONOMER {&#39;DNA repair (child)&#39;}
EG10924-MONOMER {&#39;DNA repair (child)&#39;}
EG10925-MONOMER {&#39;DNA repair (child)&#39;}
EG10926-MONOMER {&#39;DNA repair (child)&#39;}
EG10927-MONOMER {&#39;DNA replication (child)&#39;}
EG10942-MONOMER {&#39;Translation (child)&#39;}
EG10955-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG10956-MONOMER {&#39;Proteolysis (child)&#39;}
EG10976-MONOMER {&#39;DNA replication (child)&#39;}
EG10984-MONOMER {&#39;Cell division (child)&#39;}
EG10985-MONOMER {&#39;Protein folding (child)&#39;}
EG10986-MONOMER {&#39;DNA repair (child)&#39;}
EG11003-MONOMER {&#39;Protein folding (child)&#39;}
EG11004-MONOMER {&#39;Proteolysis (child)&#39;}
EG11007-MONOMER {&#39;Cell division (child)&#39;}
EG11008-MONOMER {&#39;Cell division (child)&#39;}
EG11009-MONOMER {&#39;Porin (child)&#39;}
EG11010-MONOMER {&#39;Cell division (child)&#39;}
EG11011-MONOMER {&#39;Cell division (child)&#39;}
EG11033-MONOMER {&#39;Translation (child)&#39;}
EG11035-MONOMER {&#39;Porin (child)&#39;}
EG11036-MONOMER {&#39;Translation (child)&#39;}
EG11037-MONOMER {&#39;Translation (child)&#39;}
EG11038-MONOMER {&#39;DNA replication (child)&#39;}
EG11056-MONOMER {&#39;DNA repair (child)&#39;}
EG11057-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11058-MONOMER {&#39;DNA repair (child)&#39;}
EG11061-MONOMER {&#39;DNA repair (child)&#39;}
EG11062-MONOMER {&#39;DNA repair (child)&#39;}
EG11063-MONOMER {&#39;DNA repair (child)&#39;}
EG11064-MONOMER {&#39;DNA replication (child)&#39;}
EG11068-MONOMER {&#39;DNA repair (child)&#39;}
EG11069-MONOMER {&#39;Cell division (child)&#39;}
EG11071-MONOMER {&#39;Cell division (child)&#39;}
EG11073-MONOMER {&#39;DNA repair (child)&#39;}
EG11080-MONOMER {&#39;Protein folding (child)&#39;}
EG11083-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11084-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11086-MONOMER {&#39;Cell division (child)&#39;}
EG11088-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11094-MONOMER {&#39;DNA replication (child)&#39;}
EG11103-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11111-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11128-MONOMER {&#39;Translation (child)&#39;}
EG11131-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11133-MONOMER {&#39;Proteolysis (child)&#39;}
EG11137-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11146-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11149-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11179-MONOMER {&#39;Translation (child)&#39;}
EG11183-MONOMER {&#39;Cell division (child)&#39;}
EG11191-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11203-MONOMER {&#39;Cell division (child)&#39;}
EG11210-MONOMER {&#39;DNA replication (child)&#39;}
EG11212-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11222-MONOMER {&#39;DNA repair (child)&#39;}
EG11231-MONOMER {&#39;Translation (child)&#39;}
EG11232-MONOMER {&#39;Translation (child)&#39;}
EG11236-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11249-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11250-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11252-MONOMER {&#39;DNA replication (child)&#39;}
EG11268-MONOMER {&#39;Translation (child)&#39;}
EG11272-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11281-MONOMER {&#39;DNA repair (child)&#39;}
EG11291-MONOMER {&#39;Proteolysis (child)&#39;}
EG11296-MONOMER {&#39;DNA repair (child)&#39;}
EG11299-MONOMER {&#39;Cell division (child)&#39;}
EG11312-MONOMER {&#39;Proteolysis (child)&#39;}
EG11334-MONOMER {&#39;DNA replication (child)&#39;}
EG11355-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11357-MONOMER {&#39;DNA repair (child)&#39;}
EG11359-MONOMER {&#39;Proteolysis (child)&#39;}
EG11364-MONOMER {&#39;Porin (child)&#39;}
EG11376-MONOMER {&#39;Cell division (child)&#39;}
EG11378-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
EG11384-MONOMER {&#39;Redox homeostasis (child)&#39;}
EG11385-MONOMER {&#39;Redox homeostasis (child)&#39;}
EG11389-MONOMER {&#39;Protein folding (child)&#39;}
EG11394-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11412-MONOMER {&#39;DNA replication (child)&#39;}
EG11413-MONOMER {&#39;DNA replication (child)&#39;}
EG11414-MONOMER {&#39;DNA replication (child)&#39;}
EG11432-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11440-MONOMER {&#39;Translation (child)&#39;}
EG11441-MONOMER {&#39;DNA replication (child)&#39;}
EG11449-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11500-MONOMER {&#39;DNA replication (child)&#39;}
EG11505-MONOMER {&#39;DNA replication (child)&#39;}
EG11506-MONOMER {&#39;Proteolysis (child)&#39;}
EG11508-MONOMER {&#39;Translation (child)&#39;}
EG11519-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11529-MONOMER {&#39;Cell division (child)&#39;}
EG11544-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11549-MONOMER {&#39;Proteolysis (child)&#39;}
EG11554-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11594-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11619-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11649-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11652-MONOMER {&#39;Proteolysis (child)&#39;}
EG11653-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
EG11662-MONOMER {&#39;Proteolysis (child)&#39;}
EG11676-MONOMER {&#39;Proteolysis (child)&#39;}
EG11707-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11711-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11721-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11733-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11738-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11744-MONOMER {&#39;Proteolysis (child)&#39;}
EG11775-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11785-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11808-MONOMER {&#39;DNA repair (child)&#39;}
EG11837-MONOMER {&#39;Translation (child)&#39;}
EG11838-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11849-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11854-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11855-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11861-MONOMER {&#39;Transcription (child)&#39;}
EG11862-MONOMER {&#39;Proteolysis (child)&#39;}
EG11878-MONOMER {&#39;Cell division (child)&#39;}
EG11881-MONOMER {&#39;Proteolysis (child)&#39;}
EG11889-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11913-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11915-MONOMER {&#39;DNA repair (child)&#39;}
EG11920-MONOMER {&#39;Proteolysis (child)&#39;}
EG11929-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11962-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11966-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11969-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG11989-MONOMER {&#39;Protein folding (child)&#39;}
EG12008-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12015-MONOMER {&#39;Proteolysis (child)&#39;}
EG12031-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12045-MONOMER {&#39;Translation (child)&#39;}
EG12080-MONOMER {&#39;DNA repair (child)&#39;}
EG12096-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12099-MONOMER {&#39;Translation (child)&#39;}
EG12105-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12108-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12111-MONOMER {&#39;Cell division (child)&#39;}
EG12114-MONOMER {&#39;Translation (child)&#39;}
EG12116-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12121-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12123-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12130-MONOMER {&#39;Protein folding (child)&#39;}
EG12131-MONOMER {&#39;Protein folding (child)&#39;}
EG12132-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
EG12138-MONOMER {&#39;Proteolysis (child)&#39;}
EG12140-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12155-MONOMER {&#39;Cell division (child)&#39;}
EG12163-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12165-MONOMER {&#39;DNA replication (child)&#39;}
EG12166-MONOMER {&#39;Protein folding (child)&#39;}
EG12177-MONOMER {&#39;Response to copper ion (child)&#39;}
EG12179-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12181-MONOMER {&#39;Redox homeostasis (child)&#39;}
EG12190-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12193-MONOMER {&#39;Protein folding (child)&#39;}
EG12195-MONOMER {&#39;Protein folding (child)&#39;}
EG12197-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12202-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12203-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12204-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12205-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12235-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12242-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12243-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12246-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12247-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12254-MONOMER {&#39;Proteolysis (child)&#39;}
EG12278-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12289-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12297-MONOMER {&#39;Cell division (child)&#39;}
EG12308-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12310-MONOMER {&#39;Proteolysis (child)&#39;}
EG12311-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
EG12313-MONOMER {&#39;Cell division (child)&#39;}
EG12314-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12318-MONOMER {&#39;Response to copper ion (child)&#39;}
EG12332-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
EG12344-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12356-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12371-MONOMER {&#39;Cell division (child)&#39;}
EG12386-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12406-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12424-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12431-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12454-MONOMER {&#39;Translation (child)&#39;}
EG12690-MONOMER {&#39;DNA replication (child)&#39;}
EG12694-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12717-MONOMER {&#39;DNA repair (child)&#39;}
EG12844-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12850-MONOMER {&#39;Proteolysis (child)&#39;}
EG12854-MONOMER {&#39;Cell division (child)&#39;}
EG12855-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12860-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG12878-MONOMER {&#39;Cell division (child)&#39;}
EG12879-MONOMER {&#39;Translation (child)&#39;}
EG20249-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG20253-MONOMER {&#39;Regulation of transcription (child)&#39;}
EG50001-MONOMER {&#39;Translation (child)&#39;}
EG50002-MONOMER {&#39;Translation (child)&#39;}
FLAVONADPREDUCT-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
FTSE-MONOMER {&#39;Cell division (child)&#39;}
FTSX-MONOMER {&#39;Cell division (child)&#39;}
G6088-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6110-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6111-MONOMER {&#39;Proteolysis (child)&#39;}
G6115-MONOMER {&#39;DNA replication (child)&#39;}
G6116-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6117-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6118-MONOMER {&#39;DNA repair (child)&#39;}
G6122-MONOMER {&#39;Proteolysis (child)&#39;}
G6129-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6144-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6165-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6167-MONOMER {&#39;Translation (child)&#39;}
G6175-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6181-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6195-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6201-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6209-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6226-MONOMER {&#39;Transcription (child)&#39;}
G6227-MONOMER {&#39;Transcription (child)&#39;}
G6243-MONOMER {&#39;DNA repair (child)&#39;}
G6247-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6251-MONOMER {&#39;DNA repair (child)&#39;}
G6259-MONOMER {&#39;Transcription (child)&#39;}
G6263-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6268-MONOMER {&#39;Redox homeostasis (child)&#39;}
G6274-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6276-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6291-MONOMER {&#39;Protein folding (child)&#39;}
G6302-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6306-MONOMER {&#39;DNA repair (child)&#39;}
G6311-MONOMER {&#39;Proteolysis (child)&#39;}
G6320-MONOMER {&#39;Porin (child)&#39;}
G6321-MONOMER {&#39;Response to copper ion (child)&#39;}
G6322-MONOMER {&#39;Response to copper ion (child)&#39;}
G6332-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6362-MONOMER {&#39;Translation (child)&#39;}
G6370-MONOMER {&#39;Porin (child)&#39;}
G6379-MONOMER {&#39;DNA repair (child)&#39;}
G6383-MONOMER {&#39;DNA repair (child)&#39;}
G6384-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6386-MONOMER {&#39;Protein folding (child)&#39;}
G6398-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6420-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6436-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6444-MONOMER {&#39;Transcription (child)&#39;}
G6463-MONOMER {&#39;Proteolysis (child)&#39;}
G6464-MONOMER {&#39;Cell division (child)&#39;}
G6467-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6470-MONOMER {&#39;Proteolysis (child)&#39;}
G6481-MONOMER {&#39;Protein folding (child)&#39;}
G6486-MONOMER {&#39;Cell division (child)&#39;}
G6493-MONOMER {&#39;Proteolysis (child)&#39;}
G6504-MONOMER {&#39;Porin (child)&#39;}
G6510-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6511-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6558-MONOMER {&#39;DNA repair (child)&#39;}
G6569-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6570-MONOMER {&#39;Response to copper ion (child)&#39;}
G6602-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6628-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6638-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6657-MONOMER {&#39;Porin (child)&#39;}
G6658-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6663-MONOMER {&#39;Cell division (child)&#39;}
G6664-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6671-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6680-MONOMER {&#39;Transcription (child)&#39;}
G6684-MONOMER {&#39;DNA replication (child)&#39;}
G6700-MONOMER {&#39;Porin (child)&#39;}
G6706-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6720-MONOMER {&#39;Transcription (child)&#39;}
G6737-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6745-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6749-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6750-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6761-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6773-MONOMER {&#39;Porin (child)&#39;}
G6774-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6782-MONOMER {&#39;Proteolysis (child)&#39;}
G6789-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6799-MONOMER {&#39;Transcription (child)&#39;}
G6812-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6814-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6825-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6830-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6853-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6856-MONOMER {&#39;Proteolysis (child)&#39;}
G6866-MONOMER {&#39;Porin (child)&#39;}
G6867-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6882-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6905-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
G6906-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
G6907-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
G6908-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
G6909-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
G6919-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6936-MONOMER {&#39;Cell division (child)&#39;}
G6937-MONOMER {&#39;DNA repair (child)&#39;}
G6957-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6976-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6985-MONOMER {&#39;Regulation of transcription (child)&#39;}
G6992-MONOMER {&#39;DNA repair (child)&#39;}
G7003-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7014-MONOMER {&#39;Response to copper ion (child)&#39;}
G7016-MONOMER {&#39;DNA repair (child)&#39;}
G7051-MONOMER {&#39;Cell division (child)&#39;}
G7055-MONOMER {&#39;DNA repair (child)&#39;}
G7068-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7071-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7072-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7082-MONOMER {&#39;Proteolysis (child)&#39;}
G7088-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7107-MONOMER {&#39;Porin (child)&#39;}
G7118-MONOMER {&#39;Proteolysis (child)&#39;}
G7133-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7137-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7161-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7176-MONOMER {&#39;Proteolysis (child)&#39;}
G7178-MONOMER {&#39;Proteolysis (child)&#39;}
G7207-MONOMER {&#39;Protein folding (child)&#39;}
G7245-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7247-MONOMER {&#39;Proteolysis (child)&#39;}
G7248-MONOMER {&#39;Proteolysis (child)&#39;}
G7257-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7258-MONOMER {&#39;Cell division (child)&#39;}
G7262-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7308-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7311-MONOMER {&#39;Proteolysis (child)&#39;}
G7313-MONOMER {&#39;DNA replication (child)&#39;}
G7324-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
G7326-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7329-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7331-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7344-MONOMER {&#39;Transcription (child)&#39;}
G7365-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7366-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7369-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7379-MONOMER {&#39;Proteolysis (child)&#39;}
G7397-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7416-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7424-MONOMER {&#39;Cell division (child)&#39;}
G7425-MONOMER {&#39;DNA repair (child)&#39;}
G7434-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7454-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
G7458-MONOMER {&#39;Cell division (child)&#39;}
G7466-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7467-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7472-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7488-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7491-MONOMER {&#39;Proteolysis (child)&#39;}
G7511-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
G7518-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7546-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7547-MONOMER {&#39;Transcription (child)&#39;}
G7563-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7571-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7572-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7601-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7602-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7619-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7629-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7630-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7638-MONOMER {&#39;Protein folding (child)&#39;}
G7647-MONOMER {&#39;DNA repair (child)&#39;}
G7657-MONOMER {&#39;Redox homeostasis (child)&#39;}
G7678-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7680-MONOMER {&#39;Cell division (child)&#39;}
G7681-MONOMER {&#39;Cell division (child)&#39;}
G7682-MONOMER {&#39;Proteolysis (child)&#39;}
G7688-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7689-MONOMER {&#39;Proteolysis (child)&#39;}
G7716-MONOMER {&#39;Protein folding (child)&#39;}
G7727-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7734-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7744-MONOMER {&#39;Protein folding (child)&#39;}
G7746-MONOMER {&#39;Translation (child)&#39;}
G7748-MONOMER {&#39;Iron-sulfur cluster assembly (child)&#39;}
G7751-MONOMER {&#39;DNA repair (child)&#39;}
G7790-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7814-MONOMER {&#39;Porin (child)&#39;}
G7821-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7853-MONOMER {&#39;Proteolysis (child)&#39;}
G7854-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7864-MONOMER {&#39;Cell division (child)&#39;}
G7865-MONOMER {&#39;Protein folding (child)&#39;}
G7882-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7891-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7909-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7912-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7913-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7921-MONOMER {&#39;Porin (child)&#39;}
G7924-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7925-MONOMER {&#39;Proteolysis (child)&#39;}
G7936-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7944-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7947-MONOMER {&#39;Regulation of transcription (child)&#39;}
G7948-MONOMER {&#39;Regulation of transcription (child)&#39;}
GLUTATHIONE-REDUCT-NADPH-MONOMER {&#39;Redox homeostasis (child)&#39;}
GPH-MONOMER {&#39;DNA repair (child)&#39;}
HMP-KIN-MONOMER {&#39;DNA repair (child)&#39;}
MODF-MONOMER {&#39;DNA repair (child)&#39;}
MONOMER0-1041 {&#39;Regulation of transcription (child)&#39;}
MONOMER0-1561 {&#39;Cell division (child)&#39;}
MONOMER0-185 {&#39;Regulation of transcription (child)&#39;}
MONOMER0-2141 {&#39;Cell division (child)&#39;}
MONOMER0-2383 {&#39;DNA replication (child)&#39;}
MONOMER0-2488 {&#39;Regulation of transcription (child)&#39;}
MONOMER0-2656 {&#39;Translation (child)&#39;}
MONOMER0-2673 {&#39;Regulation of transcription (child)&#39;}
MONOMER0-282 {&#39;Porin (child)&#39;}
NACMURLALAAMI1-MONOMER {&#39;Cell division (child)&#39;}
NACMURLALAAMI2-MONOMER {&#39;Cell division (child)&#39;}
PD00194 {&#39;Regulation of transcription (child)&#39;}
PD00196 {&#39;Regulation of transcription (child)&#39;}
PD00197 {&#39;Regulation of transcription (child)&#39;}
PD00200 {&#39;Regulation of transcription (child)&#39;}
PD00205 {&#39;Regulation of transcription (child)&#39;}
PD00208 {&#39;Regulation of transcription (child)&#39;}
PD00214 {&#39;Regulation of transcription (child)&#39;}
PD00219 {&#39;Regulation of transcription (child)&#39;}
PD00221 {&#39;Regulation of transcription (child)&#39;}
PD00222 {&#39;Regulation of transcription (child)&#39;}
PD00230 {&#39;Regulation of transcription (child)&#39;}
PD00232 {&#39;Regulation of transcription (child)&#39;}
PD00237 {&#39;Regulation of transcription (child)&#39;}
PD00242 {&#39;Regulation of transcription (child)&#39;}
PD00250 {&#39;Regulation of transcription (child)&#39;}
PD00251 {&#39;Regulation of transcription (child)&#39;}
PD00257 {&#39;Regulation of transcription (child)&#39;}
PD00260 {&#39;Regulation of transcription (child)&#39;}
PD00261 {&#39;Regulation of transcription (child)&#39;}
PD00266 {&#39;Regulation of transcription (child)&#39;}
PD00268 {&#39;Regulation of transcription (child)&#39;}
PD00283 {&#39;Regulation of transcription (child)&#39;}
PD00288 {&#39;Regulation of transcription (child)&#39;}
PD00291 {&#39;Regulation of transcription (child)&#39;}
PD00297 {&#39;Regulation of transcription (child)&#39;}
PD00339 {&#39;Regulation of transcription (child)&#39;}
PD00345 {&#39;Regulation of transcription (child)&#39;}
PD00347 {&#39;Regulation of transcription (child)&#39;}
PD00348 {&#39;Regulation of transcription (child)&#39;}
PD00353 {&#39;Regulation of transcription (child)&#39;}
PD00360 {&#39;Regulation of transcription (child)&#39;}
PD00361 {&#39;Regulation of transcription (child)&#39;}
PD00364 {&#39;Regulation of transcription (child)&#39;}
PD00365 {&#39;Regulation of transcription (child)&#39;}
PD00406 {&#39;Regulation of transcription (child)&#39;}
PD00413 {&#39;Regulation of transcription (child)&#39;}
PD00423 {&#39;Regulation of transcription (child)&#39;}
PD00436 {&#39;Regulation of transcription (child)&#39;}
PD00440 {&#39;Regulation of transcription (child)&#39;}
PD00444 {&#39;Regulation of transcription (child)&#39;}
PD00474 {&#39;Regulation of transcription (child)&#39;}
PD00519 {&#39;Regulation of transcription (child)&#39;}
PD00521 {&#39;Regulation of transcription (child)&#39;}
PD00763 {&#39;Regulation of transcription (child)&#39;}
PD00967 {&#39;Regulation of transcription (child)&#39;}
PD00969 {&#39;Regulation of transcription (child)&#39;}
PD01103 {&#39;Regulation of transcription (child)&#39;}
PD01196 {&#39;Regulation of transcription (child)&#39;}
PD01352 {&#39;Regulation of transcription (child)&#39;}
PD01379 {&#39;Regulation of transcription (child)&#39;}
PD01520 {&#39;Regulation of transcription (child)&#39;}
PD01896 {&#39;Regulation of transcription (child)&#39;}
PD02198 {&#39;Regulation of transcription (child)&#39;}
PD02233 {&#39;Regulation of transcription (child)&#39;}
PD02936 {&#39;Regulation of transcription (child)&#39;}
PD03028 {&#39;Regulation of transcription (child)&#39;}
PD03254 {&#39;Regulation of transcription (child)&#39;}
PD03270 {&#39;Regulation of transcription (child)&#39;}
PD03585 {&#39;Regulation of transcription (child)&#39;}
PD03695 {&#39;Regulation of transcription (child)&#39;}
PD03831 {&#39;DNA replication (child)&#39;}
PD03867 {&#39;Regulation of transcription (child)&#39;}
PD03938 {&#39;Regulation of transcription (child)&#39;}
PD04028 {&#39;Regulation of transcription (child)&#39;}
PD04032 {&#39;Regulation of transcription (child)&#39;}
PD04099 {&#39;Regulation of transcription (child)&#39;}
PD04132 {&#39;Regulation of transcription (child)&#39;}
PD04418 {&#39;Regulation of transcription (child)&#39;}
RPOB-MONOMER {&#39;Transcription (child)&#39;}
RPOC-MONOMER {&#39;Transcription (child)&#39;}
RPOD-MONOMER {&#39;Regulation of transcription (child)&#39;}
RPOE-MONOMER {&#39;Regulation of transcription (child)&#39;}
RPOH-MONOMER {&#39;Regulation of transcription (child)&#39;}
RPON-MONOMER {&#39;Regulation of transcription (child)&#39;}
RPOS-MONOMER {&#39;Regulation of transcription (child)&#39;}
SECB {&#39;Protein folding (child)&#39;}
SGCB-MONOMER {&#39;Proteolysis (child)&#39;}
TORT-MONOMER {&#39;Regulation of transcription (child)&#39;}
UDHA-MONOMER {&#39;Redox homeostasis (child)&#39;}
UUP-MONOMER {&#39;DNA repair (child)&#39;}
YBDE-MONOMER {&#39;Response to copper ion (child)&#39;}
YJJK-MONOMER {&#39;Translation (child)&#39;}
YRAO-MONOMER {&#39;DNA replication (child)&#39;}
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;EG10230-MONOMER&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>id</th>
      <th>common_name</th>
      <th>seq</th>
      <th>enzyme_reaction</th>
      <th>cofactors</th>
      <th>other_features</th>
      <th>direct_annotations</th>
      <th>go_annotations</th>
      <th>uniprot_id</th>
      <th>metal_features_processed</th>
      <th>other_features_processed</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>425</th>
      <td>EG10230-MONOMER</td>
      <td>RNA polymerase-binding transcription factor DksA</td>
      <td>MQEGQNRKTSSLSILAIAGVEPYQEKPGEEYMNEAQLAHFRRILEA...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{Regulation of transcription (child)}</td>
      <td>{GO:0008270, GO:0005737, GO:0097216, GO:000582...</td>
      <td>P0ABS1</td>
      <td>{'ZN+2': 1}</td>
      <td>{}</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><br/><span></span><span class="c1"># Add uncategorized pathways</span>
<span class="n">new_rows</span> <span class="o">=</span> <span class="p">[</span>
    <span class="p">{</span><span class="s1">&#39;id&#39;</span><span class="p">:</span> <span class="s1">&#39;Uncategorized (I)&#39;</span><span class="p">,</span> <span class="s1">&#39;parents&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Uncategorized (II)&#39;</span><span class="p">],</span> <span class="s1">&#39;level&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;Uncategorized (I)&#39;</span><span class="p">},</span>
    <span class="p">{</span><span class="s1">&#39;id&#39;</span><span class="p">:</span> <span class="s1">&#39;Uncategorized (II)&#39;</span><span class="p">,</span> <span class="s1">&#39;parents&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Uncategorized (I)&#39;</span><span class="p">],</span> <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Uncategorized (III)&#39;</span><span class="p">],</span> <span class="s1">&#39;level&#39;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span> <span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;Uncategorized&#39;</span><span class="p">},</span>
    <span class="p">{</span><span class="s1">&#39;id&#39;</span><span class="p">:</span> <span class="s1">&#39;Uncategorized (III)&#39;</span><span class="p">,</span> <span class="s1">&#39;parents&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Uncategorized (II)&#39;</span><span class="p">],</span> <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;level&#39;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> <span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;Uncategorized (III)&#39;</span><span class="p">},</span>
<span class="p">]</span>

<span class="c1"># Convert the dictionaries to a DataFrame.</span>
<span class="n">new_rows_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">new_rows</span><span class="p">)</span>

<span class="c1"># Insert the new rows into the existing DataFrame.</span>
<span class="c1"># Assuming you want to append at the end:</span>
<span class="n">parsed_pathway_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">parsed_pathway_df</span><span class="p">,</span> <span class="n">new_rows_df</span><span class="p">],</span> <span class="n">ignore_index</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="c1"># for all proteins with empty direct_annotations, add &#39;Uncategorized (III)&#39;</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;direct_annotations&#39;</span><span class="p">])</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;direct_annotations&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="nb">set</span><span class="p">([</span><span class="s1">&#39;Uncategorized (III)&#39;</span><span class="p">])</span>

<span class="n">parsed_protein_df</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>id</th>
      <th>common_name</th>
      <th>seq</th>
      <th>enzyme_reaction</th>
      <th>cofactors</th>
      <th>other_features</th>
      <th>direct_annotations</th>
      <th>go_annotations</th>
      <th>uniprot_id</th>
      <th>metal_features_processed</th>
      <th>other_features_processed</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1-ACYLGLYCEROL-3-P-ACYLTRANSFER-MONOMER</td>
      <td>1-acylglycerol-3-phosphate &lt;i&gt;O&lt;/i&gt;-acyltransf...</td>
      <td>MLYIFRLIITVIYSILVCVFGSIYCLFSPRNPKHVATFGHMFGRLA...</td>
      <td>{ENZRXN0-7995, ENZRXN0-7462, ENZRXN0-7991, 1-A...</td>
      <td>{}</td>
      <td>[]</td>
      <td>{PWY-5667, PWY0-1319}</td>
      <td>{GO:0006629, GO:0016020, GO:0016024, GO:000384...</td>
      <td>P26647</td>
      <td>{}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1-PFK-MONOMER</td>
      <td>1-phosphofructokinase</td>
      <td>MSRRVATITLNPAYDLVGFCPEIERGEVNLVKTTGLHAAGKGINVA...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[ATP]</td>
      <td>{PWY0-1314}</td>
      <td>{GO:0000166, GO:0042803, GO:0016301, GO:000844...</td>
      <td>P0AEW9</td>
      <td>{}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2-DEHYDROPANTOATE-REDUCT-MONOMER</td>
      <td>2-dehydropantoate 2-reductase</td>
      <td>MKITVLGCGALGQLWLTALCKQGHEVQGWLRVPQPYCSVNLVETDG...</td>
      <td>{2-DEHYDROPANTOATE-REDUCT-ENZRXN}</td>
      <td>{}</td>
      <td>[NADP]</td>
      <td>{PANTO-PWY}</td>
      <td>{GO:0005737, GO:0016491, GO:0050661, GO:001661...</td>
      <td>P0A9J4</td>
      <td>{}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2-ISOPROPYLMALATESYN-MONOMER</td>
      <td>2-isopropylmalate synthase</td>
      <td>MSQQVIIFDTTLRDGEQALQASLSVKEKLQIALALERMGVDVMEVG...</td>
      <td>{ENZRXN0-6250}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{LEUSYN-PWY}</td>
      <td>{GO:0005737, GO:0030145, GO:0003985, GO:000385...</td>
      <td>P09151</td>
      <td>{}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>4</th>
      <td>2-OCTAPRENYL-METHOXY-BENZOQ-METH-MONOMER</td>
      <td>bifunctional 2-octaprenyl-6-methoxy-1,4-benzoq...</td>
      <td>MVDKSQETTHFGFQTVAKEQKADMVAHVFHSVASKYDVMNDLMSFG...</td>
      <td>{2-OCTAPRENYL-METHOXY-BENZOQ-METH-ENZRXN, ADOM...</td>
      <td>{}</td>
      <td>[]</td>
      <td>{PWY-6708, MENAQUINONESYN-PWY}</td>
      <td>{GO:0008757, GO:0005737, GO:0102094, GO:010202...</td>
      <td>P0A887</td>
      <td>{}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>4429</th>
      <td>YTFR-MONOMER</td>
      <td>galactofuranose ABC transporter putative ATP b...</td>
      <td>MTTDQHQEILRTEGLSKFFPGVKALDNVDFSLRRGEIMALLGENGA...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[ATP]</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0140271, GO:0000166, GO:0008643, GO:004287...</td>
      <td>Q6BEX0</td>
      <td>{}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>4430</th>
      <td>YTFT-MONOMER</td>
      <td>galactofuranose ABC transporter putative membr...</td>
      <td>MMPQSLPDTTTPKRRFRWPTGMPQLVALLLVLLVDSLVAPHFWQVV...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0140271, GO:0008643, GO:0005515, GO:004319...</td>
      <td>P39328</td>
      <td>{}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>4431</th>
      <td>ZNUA-MONOMER</td>
      <td>Zn&lt;sup&gt;2+&lt;/sup&gt; ABC transporter periplasmic bi...</td>
      <td>MLHKKTLLFAALSAALWGGATQAADAAVVASLKPVGFIASAIADGV...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0008270, GO:0007155, GO:0030288, GO:001602...</td>
      <td>P39172</td>
      <td>{'ZN+2': 1}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>4432</th>
      <td>ZNUB-MONOMER</td>
      <td>Zn&lt;sup&gt;2+&lt;/sup&gt; ABC transporter membrane subunit</td>
      <td>MIELLFPGWLAGIMLACAAGPLGSFVVWRRMSYFGDTLAHASLLGV...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0043190, GO:0000006, GO:0016020, GO:000682...</td>
      <td>P39832</td>
      <td>{}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>4433</th>
      <td>ZNUC-MONOMER</td>
      <td>Zn&lt;sup&gt;2+&lt;/sup&gt; ABC transporter ATP binding su...</td>
      <td>MTSLVSLENVSVSFGQRRVLSDVSLELKPGKILTLLGPNGAGKSTL...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[ATP]</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0000166, GO:0015633, GO:0043190, GO:001602...</td>
      <td>P0A9X1</td>
      <td>{}</td>
      <td>{}</td>
    </tr>
  </tbody>
</table>
<p>4434 rows × 11 columns</p>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;EG10230-MONOMER&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>id</th>
      <th>common_name</th>
      <th>seq</th>
      <th>enzyme_reaction</th>
      <th>cofactors</th>
      <th>other_features</th>
      <th>direct_annotations</th>
      <th>go_annotations</th>
      <th>uniprot_id</th>
      <th>metal_features_processed</th>
      <th>other_features_processed</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>425</th>
      <td>EG10230-MONOMER</td>
      <td>RNA polymerase-binding transcription factor DksA</td>
      <td>MQEGQNRKTSSLSILAIAGVEPYQEKPGEEYMNEAQLAHFRRILEA...</td>
      <td>{}</td>
      <td>{}</td>
      <td>[]</td>
      <td>{Regulation of transcription (child)}</td>
      <td>{GO:0008270, GO:0005737, GO:0097216, GO:000582...</td>
      <td>P0ABS1</td>
      <td>{'ZN+2': 1}</td>
      <td>{}</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># decompose sequence into dict of amino acid counts</span>
<span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;sequence_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
<span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;sequence_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;sequence_processed&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">object</span><span class="p">)</span>

<span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;sequence_mass&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>

<span class="n">unique_aa</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>

    <span class="n">sequence</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;seq&#39;</span><span class="p">]</span>

    <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">sequence</span><span class="p">)</span> <span class="o">!=</span> <span class="nb">str</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;No sequence for </span><span class="si">{</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s2">&quot;id&quot;</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="k">continue</span>

    <span class="n">aa_count_dict</span> <span class="o">=</span> <span class="p">{}</span>

    <span class="n">mass</span> <span class="o">=</span> <span class="mi">0</span>

    <span class="k">for</span> <span class="n">aa</span> <span class="ow">in</span> <span class="n">sequence</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">aa</span> <span class="ow">in</span> <span class="n">aa_count_dict</span><span class="p">:</span>
            <span class="n">aa_count_dict</span><span class="p">[</span><span class="n">aa</span><span class="p">]</span> <span class="o">+=</span> <span class="mi">1</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">aa_count_dict</span><span class="p">[</span><span class="n">aa</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>

        <span class="k">if</span> <span class="n">aa</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">unique_aa</span><span class="p">:</span>
            <span class="n">unique_aa</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">aa</span><span class="p">)</span>

        <span class="n">mass</span> <span class="o">+=</span> <span class="n">AMINO_ACID_RESIDUE_MASSES</span><span class="p">[</span><span class="n">AMINO_ACID_MAP</span><span class="p">[</span><span class="n">aa</span><span class="p">]]</span>

    <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;sequence_processed&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">aa_count_dict</span>
    <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;sequence_mass&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">mass</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/var/folders/dx/0239zgvj0tgf46b5h8l7v_fc0000gn/T/ipykernel_62505/3799389604.py:33: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value &#39;27417.49714999999&#39; has dtype incompatible with int64, please explicitly cast to a compatible dtype first.
  parsed_protein_df.at[i, &#39;sequence_mass&#39;] = mass
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
No sequence for MONOMER0-1241
No sequence for MONOMER0-4223
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_protein_df</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;other_features&#39;</span><span class="p">,</span><span class="s1">&#39;seq&#39;</span><span class="p">])</span>

<span class="n">parsed_protein_df</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>id</th>
      <th>common_name</th>
      <th>enzyme_reaction</th>
      <th>cofactors</th>
      <th>direct_annotations</th>
      <th>go_annotations</th>
      <th>uniprot_id</th>
      <th>metal_features_processed</th>
      <th>other_features_processed</th>
      <th>sequence_processed</th>
      <th>sequence_mass</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1-ACYLGLYCEROL-3-P-ACYLTRANSFER-MONOMER</td>
      <td>1-acylglycerol-3-phosphate &lt;i&gt;O&lt;/i&gt;-acyltransf...</td>
      <td>{ENZRXN0-7995, ENZRXN0-7462, ENZRXN0-7991, 1-A...</td>
      <td>{}</td>
      <td>{PWY-5667, PWY0-1319}</td>
      <td>{GO:0006629, GO:0016020, GO:0016024, GO:000384...</td>
      <td>P26647</td>
      <td>{}</td>
      <td>{}</td>
      <td>{'M': 6, 'L': 21, 'Y': 8, 'I': 22, 'F': 12, 'R...</td>
      <td>27417.49715</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1-PFK-MONOMER</td>
      <td>1-phosphofructokinase</td>
      <td>{}</td>
      <td>{}</td>
      <td>{PWY0-1314}</td>
      <td>{GO:0000166, GO:0042803, GO:0016301, GO:000844...</td>
      <td>P0AEW9</td>
      <td>{}</td>
      <td>{}</td>
      <td>{'M': 8, 'S': 21, 'R': 18, 'V': 32, 'A': 30, '...</td>
      <td>33716.33963</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2-DEHYDROPANTOATE-REDUCT-MONOMER</td>
      <td>2-dehydropantoate 2-reductase</td>
      <td>{2-DEHYDROPANTOATE-REDUCT-ENZRXN}</td>
      <td>{}</td>
      <td>{PANTO-PWY}</td>
      <td>{GO:0005737, GO:0016491, GO:0050661, GO:001661...</td>
      <td>P0A9J4</td>
      <td>{}</td>
      <td>{}</td>
      <td>{'M': 7, 'K': 7, 'I': 24, 'T': 21, 'V': 21, 'L...</td>
      <td>33831.33232</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2-ISOPROPYLMALATESYN-MONOMER</td>
      <td>2-isopropylmalate synthase</td>
      <td>{ENZRXN0-6250}</td>
      <td>{}</td>
      <td>{LEUSYN-PWY}</td>
      <td>{GO:0005737, GO:0030145, GO:0003985, GO:000385...</td>
      <td>P09151</td>
      <td>{}</td>
      <td>{}</td>
      <td>{'M': 13, 'S': 30, 'Q': 22, 'V': 46, 'I': 39, ...</td>
      <td>57244.09138</td>
    </tr>
    <tr>
      <th>4</th>
      <td>2-OCTAPRENYL-METHOXY-BENZOQ-METH-MONOMER</td>
      <td>bifunctional 2-octaprenyl-6-methoxy-1,4-benzoq...</td>
      <td>{2-OCTAPRENYL-METHOXY-BENZOQ-METH-ENZRXN, ADOM...</td>
      <td>{}</td>
      <td>{PWY-6708, MENAQUINONESYN-PWY}</td>
      <td>{GO:0008757, GO:0005737, GO:0102094, GO:010202...</td>
      <td>P0A887</td>
      <td>{}</td>
      <td>{}</td>
      <td>{'M': 10, 'V': 22, 'D': 19, 'K': 16, 'S': 16, ...</td>
      <td>28037.26502</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>4429</th>
      <td>YTFR-MONOMER</td>
      <td>galactofuranose ABC transporter putative ATP b...</td>
      <td>{}</td>
      <td>{}</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0140271, GO:0000166, GO:0008643, GO:004287...</td>
      <td>Q6BEX0</td>
      <td>{}</td>
      <td>{}</td>
      <td>{'M': 11, 'T': 26, 'D': 24, 'Q': 26, 'H': 7, '...</td>
      <td>55207.40745</td>
    </tr>
    <tr>
      <th>4430</th>
      <td>YTFT-MONOMER</td>
      <td>galactofuranose ABC transporter putative membr...</td>
      <td>{}</td>
      <td>{}</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0140271, GO:0008643, GO:0005515, GO:004319...</td>
      <td>P39328</td>
      <td>{}</td>
      <td>{}</td>
      <td>{'M': 12, 'P': 15, 'Q': 11, 'S': 17, 'L': 53, ...</td>
      <td>35617.89729</td>
    </tr>
    <tr>
      <th>4431</th>
      <td>ZNUA-MONOMER</td>
      <td>Zn&lt;sup&gt;2+&lt;/sup&gt; ABC transporter periplasmic bi...</td>
      <td>{}</td>
      <td>{}</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0008270, GO:0007155, GO:0030288, GO:001602...</td>
      <td>P39172</td>
      <td>{'ZN+2': 1}</td>
      <td>{}</td>
      <td>{'M': 7, 'L': 34, 'H': 13, 'K': 20, 'T': 16, '...</td>
      <td>33738.19213</td>
    </tr>
    <tr>
      <th>4432</th>
      <td>ZNUB-MONOMER</td>
      <td>Zn&lt;sup&gt;2+&lt;/sup&gt; ABC transporter membrane subunit</td>
      <td>{}</td>
      <td>{}</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0043190, GO:0000006, GO:0016020, GO:000682...</td>
      <td>P39832</td>
      <td>{}</td>
      <td>{}</td>
      <td>{'M': 13, 'I': 18, 'E': 4, 'L': 47, 'F': 12, '...</td>
      <td>27692.26700</td>
    </tr>
    <tr>
      <th>4433</th>
      <td>ZNUC-MONOMER</td>
      <td>Zn&lt;sup&gt;2+&lt;/sup&gt; ABC transporter ATP binding su...</td>
      <td>{}</td>
      <td>{}</td>
      <td>{Uncategorized (III)}</td>
      <td>{GO:0000166, GO:0015633, GO:0043190, GO:001602...</td>
      <td>P0A9X1</td>
      <td>{}</td>
      <td>{}</td>
      <td>{'M': 5, 'T': 12, 'S': 13, 'L': 42, 'V': 25, '...</td>
      <td>27832.00030</td>
    </tr>
  </tbody>
</table>
<p>4434 rows × 11 columns</p>
</div></div>
</div>
</section>
<section id="Create-new-column-for-monomer-component-stoichiometry">
<h2>Create new column for monomer component stoichiometry<a class="headerlink" href="#Create-new-column-for-monomer-component-stoichiometry" title="Link to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">complex_ids</span> <span class="o">=</span> <span class="n">parsed_complex_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="n">monomer_names</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">recursive_component_tree</span><span class="p">(</span><span class="n">current_component_name</span><span class="p">,</span> <span class="n">complex_table</span><span class="p">,</span> <span class="n">protein_table</span><span class="p">,</span>
                             <span class="n">current_multiplier</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">component_list</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">parent</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">return_cofactors</span><span class="o">=</span><span class="kc">False</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Recursively find all downstream components of a given complex.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">complex_names</span> <span class="o">=</span> <span class="n">complex_table</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
    <span class="n">monomer_names</span> <span class="o">=</span> <span class="n">protein_table</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>


    <span class="n">my_children</span> <span class="o">=</span> <span class="p">{}</span>

    <span class="k">if</span> <span class="n">component_list</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">component_list</span> <span class="o">=</span> <span class="p">[]</span>


    <span class="k">if</span> <span class="n">current_component_name</span> <span class="ow">in</span> <span class="n">complex_names</span><span class="p">:</span>


        <span class="n">cplx_idx</span> <span class="o">=</span> <span class="n">complex_table</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">complex_table</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">current_component_name</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>
        <span class="n">stoichiometry</span> <span class="o">=</span> <span class="n">complex_table</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">cplx_idx</span><span class="p">,</span> <span class="s1">&#39;stoichiometry&#39;</span><span class="p">]</span>

        <span class="n">direct_children</span> <span class="o">=</span> <span class="p">{</span><span class="n">k</span><span class="p">:</span> <span class="nb">abs</span><span class="p">(</span><span class="n">v</span><span class="p">)</span> <span class="k">for</span> <span class="n">k</span><span class="p">,</span> <span class="n">v</span> <span class="ow">in</span> <span class="n">stoichiometry</span><span class="o">.</span><span class="n">items</span><span class="p">()</span> <span class="k">if</span> <span class="n">v</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">}</span>

        <span class="k">for</span> <span class="n">component_name</span><span class="p">,</span> <span class="n">coefficient</span> <span class="ow">in</span> <span class="n">stoichiometry</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>

            <span class="k">if</span> <span class="n">coefficient</span> <span class="o">&lt;</span> <span class="mi">0</span> <span class="ow">and</span> <span class="n">component_name</span> <span class="o">!=</span> <span class="n">current_component_name</span><span class="p">:</span>

                <span class="n">child_multiplier</span> <span class="o">=</span> <span class="nb">abs</span><span class="p">(</span><span class="n">coefficient</span> <span class="o">*</span> <span class="n">current_multiplier</span><span class="p">)</span>

                <span class="n">new_child</span> <span class="o">=</span> <span class="n">recursive_component_tree</span><span class="p">(</span><span class="n">component_name</span><span class="p">,</span> <span class="n">complex_table</span><span class="p">,</span> <span class="n">protein_table</span><span class="p">,</span>
                                                     <span class="n">child_multiplier</span><span class="p">,</span> <span class="n">component_list</span><span class="p">,</span> <span class="n">current_component_name</span><span class="p">,</span> <span class="n">return_cofactors</span><span class="p">)</span>

                <span class="n">my_children</span> <span class="o">=</span> <span class="n">my_children</span> <span class="o">|</span> <span class="n">new_child</span>


            <span class="k">elif</span> <span class="n">coefficient</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="ow">and</span> <span class="n">component_name</span> <span class="o">==</span> <span class="n">current_component_name</span><span class="p">:</span>
                <span class="k">continue</span>

            <span class="k">else</span><span class="p">:</span>
                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;key </span><span class="si">{</span><span class="n">component_name</span><span class="si">}</span><span class="s2"> and value </span><span class="si">{</span><span class="n">coefficient</span><span class="si">}</span><span class="s2"> for complex </span><span class="si">{</span><span class="n">component_name</span><span class="si">}</span><span class="s2"> not processed properly.&quot;</span><span class="p">)</span>

        <span class="n">component_list</span><span class="o">.</span><span class="n">append</span><span class="p">({</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="n">current_component_name</span><span class="p">,</span><span class="s1">&#39;parent&#39;</span><span class="p">:</span> <span class="n">parent</span><span class="p">,</span> <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="n">direct_children</span><span class="p">,</span>
                               <span class="s1">&#39;multiplier&#39;</span><span class="p">:</span> <span class="nb">int</span><span class="p">(</span><span class="n">current_multiplier</span><span class="p">),</span> <span class="p">})</span>


    <span class="k">elif</span> <span class="n">current_component_name</span> <span class="ow">in</span> <span class="n">monomer_names</span><span class="p">:</span>

        <span class="c1"># TODO check if enzrxn</span>
        <span class="k">if</span> <span class="n">return_cofactors</span><span class="p">:</span>
            <span class="n">protein_idx</span> <span class="o">=</span> <span class="n">protein_table</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">protein_table</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">current_component_name</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>

            <span class="n">protein_metals</span> <span class="o">=</span> <span class="n">protein_table</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">protein_idx</span><span class="p">,</span> <span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span>
            <span class="n">protein_other</span> <span class="o">=</span> <span class="n">protein_table</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">protein_idx</span><span class="p">,</span> <span class="s1">&#39;other_features_processed&#39;</span><span class="p">]</span>

            <span class="n">table_cofactors</span> <span class="o">=</span> <span class="n">protein_metals</span> <span class="o">|</span> <span class="n">protein_other</span>

            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">table_cofactors</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
                <span class="c1"># TODO Add apo protein to component list</span>
                <span class="n">my_children</span> <span class="o">=</span> <span class="p">{}</span>

                <span class="k">for</span> <span class="n">cofactor</span><span class="p">,</span> <span class="n">cofactor_coefficient</span> <span class="ow">in</span> <span class="n">table_cofactors</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
                    <span class="k">if</span> <span class="n">table_cofactors</span><span class="p">[</span><span class="n">cofactor</span><span class="p">]</span> <span class="o">!=</span>  <span class="kc">None</span><span class="p">:</span>
                        <span class="n">my_children</span><span class="p">[</span><span class="n">cofactor</span><span class="p">]</span> <span class="o">=</span> <span class="n">cofactor_coefficient</span>
                        <span class="n">component_list</span><span class="o">.</span><span class="n">append</span><span class="p">({</span><span class="s1">&#39;parent&#39;</span><span class="p">:</span> <span class="n">current_component_name</span><span class="p">,</span>
                                               <span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="n">cofactor</span><span class="p">,</span>
                                               <span class="s1">&#39;multiplier&#39;</span><span class="p">:</span> <span class="nb">abs</span><span class="p">(</span><span class="nb">int</span><span class="p">(</span><span class="n">current_multiplier</span> <span class="o">*</span> <span class="n">cofactor_coefficient</span><span class="p">)),</span>
                                               <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="kc">None</span><span class="p">})</span>

            <span class="n">component_list</span><span class="o">.</span><span class="n">append</span><span class="p">({</span><span class="s1">&#39;parent&#39;</span><span class="p">:</span> <span class="n">parent</span><span class="p">,</span> <span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="n">current_component_name</span><span class="p">,</span> <span class="s1">&#39;multiplier&#39;</span><span class="p">:</span> <span class="n">current_multiplier</span><span class="p">,</span> <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="n">my_children</span><span class="p">})</span>

        <span class="k">else</span><span class="p">:</span>
            <span class="n">my_children</span> <span class="o">=</span> <span class="kc">None</span>
            <span class="n">component_list</span><span class="o">.</span><span class="n">append</span><span class="p">({</span><span class="s1">&#39;parent&#39;</span><span class="p">:</span> <span class="n">parent</span><span class="p">,</span> <span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="n">current_component_name</span><span class="p">,</span> <span class="s1">&#39;multiplier&#39;</span><span class="p">:</span> <span class="n">current_multiplier</span><span class="p">,</span> <span class="s1">&#39;children&#39;</span><span class="p">:</span> <span class="kc">None</span><span class="p">})</span>



    <span class="k">else</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;component </span><span class="si">{</span><span class="n">current_component_name</span><span class="si">}</span><span class="s2"> not found in complex or protein tables&quot;</span><span class="p">)</span>

        <span class="k">return</span> <span class="p">{}</span>


    <span class="k">if</span> <span class="n">parent</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">{</span><span class="n">current_component_name</span><span class="p">:</span> <span class="n">my_children</span><span class="p">},</span> <span class="n">component_list</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">{</span><span class="n">current_component_name</span><span class="p">:</span> <span class="n">my_children</span><span class="p">}</span>
<br/></pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">complex_tree_structure</span><span class="p">,</span> <span class="n">nodes</span> <span class="o">=</span> <span class="n">recursive_component_tree</span><span class="p">(</span><span class="s1">&#39;CPLX0-8167&#39;</span><span class="p">,</span> <span class="n">parsed_complex_df</span><span class="p">,</span> <span class="n">parsed_protein_df</span><span class="p">)</span>
<span class="n">pp</span><span class="o">.</span><span class="n">pprint</span><span class="p">(</span><span class="n">nodes</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
[{&#39;children&#39;: None,
  &#39;multiplier&#39;: 4,
  &#39;name&#39;: &#39;HYAA-MONOMER&#39;,
  &#39;parent&#39;: &#39;FORMHYDROGI-CPLX&#39;},
 {&#39;children&#39;: None,
  &#39;multiplier&#39;: 4,
  &#39;name&#39;: &#39;HYAB-MONOMER&#39;,
  &#39;parent&#39;: &#39;FORMHYDROGI-CPLX&#39;},
 {&#39;children&#39;: None,
  &#39;multiplier&#39;: 2,
  &#39;name&#39;: &#39;HYAC-MONOMER&#39;,
  &#39;parent&#39;: &#39;FORMHYDROGI-CPLX&#39;},
 {&#39;children&#39;: {&#39;HYAA-MONOMER&#39;: 2, &#39;HYAB-MONOMER&#39;: 2, &#39;HYAC-MONOMER&#39;: 1},
  &#39;multiplier&#39;: 2,
  &#39;name&#39;: &#39;FORMHYDROGI-CPLX&#39;,
  &#39;parent&#39;: &#39;CPLX0-8167&#39;},
 {&#39;children&#39;: {&#39;FORMHYDROGI-CPLX&#39;: 2},
  &#39;multiplier&#39;: 1,
  &#39;name&#39;: &#39;CPLX0-8167&#39;,
  &#39;parent&#39;: None}]
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_complex_df</span><span class="p">[</span><span class="s1">&#39;monomer_component_stoichiometry&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
<span class="n">parsed_complex_df</span><span class="p">[</span><span class="s1">&#39;monomer_component_stoichiometry&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">parsed_complex_df</span><span class="p">[</span><span class="s1">&#39;monomer_component_stoichiometry&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">object</span><span class="p">)</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_complex_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>
    <span class="n">complex_name</span> <span class="o">=</span> <span class="n">parsed_complex_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;id&#39;</span><span class="p">]</span>
    <span class="n">complex_tree_structure</span><span class="p">,</span> <span class="n">nodes</span> <span class="o">=</span> <span class="n">recursive_component_tree</span><span class="p">(</span><span class="n">complex_name</span><span class="p">,</span> <span class="n">parsed_complex_df</span><span class="p">,</span> <span class="n">parsed_protein_df</span><span class="p">)</span>

    <span class="n">monomer_components</span> <span class="o">=</span> <span class="p">{</span><span class="n">node</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]:</span> <span class="n">node</span><span class="p">[</span><span class="s1">&#39;multiplier&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">node</span> <span class="ow">in</span> <span class="n">nodes</span> <span class="k">if</span> <span class="n">node</span><span class="p">[</span><span class="s1">&#39;children&#39;</span><span class="p">]</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">}</span>

    <span class="n">parsed_complex_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;monomer_component_stoichiometry&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">monomer_components</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
component CPLX0-7701 not found in complex or protein tables
component CPLX0-7677 not found in complex or protein tables
component MONOMER0-1781 not found in complex or protein tables
component CPLX0-7702 not found in complex or protein tables
component CSRB-RNA not found in complex or protein tables
component RNPB-RNA not found in complex or protein tables
component RRSA-RRNA not found in complex or protein tables
component RRLA-RRNA not found in complex or protein tables
component RRFA-RRNA not found in complex or protein tables
component RRSA-RRNA not found in complex or protein tables
component RRLA-RRNA not found in complex or protein tables
component RRFA-RRNA not found in complex or protein tables
component IS061-RNA not found in complex or protein tables
component CSRC-RNA not found in complex or protein tables
component FFS-RNA not found in complex or protein tables
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_complex_df</span> <span class="o">=</span> <span class="n">parsed_complex_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="p">[</span><span class="s2">&quot;id&quot;</span><span class="p">,</span> <span class="s2">&quot;common_name&quot;</span><span class="p">,</span> <span class="s2">&quot;stoichiometry&quot;</span><span class="p">,</span> <span class="s2">&quot;monomer_component_stoichiometry&quot;</span><span class="p">,</span> <span class="s2">&quot;cofactors&quot;</span><span class="p">]]</span>
<span class="n">parsed_complex_df</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>id</th>
      <th>common_name</th>
      <th>stoichiometry</th>
      <th>monomer_component_stoichiometry</th>
      <th>cofactors</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1-PFK</td>
      <td>1-phosphofructokinase</td>
      <td>{'1-PFK': 1, '1-PFK-MONOMER': -2}</td>
      <td>{'1-PFK-MONOMER': 2}</td>
      <td>{MG+2}</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2OXOGLUTARATEDEH-CPLX</td>
      <td>2-oxoglutarate dehydrogenase complex</td>
      <td>{'2OXOGLUTARATEDEH-CPLX': 1, 'E1O': -1, 'E2O':...</td>
      <td>{'E1O-MONOMER': 12, 'E2O-MONOMER': 24, 'E3-MON...</td>
      <td>{LIPOIC-ACID, MG+2, THIAMINE-PYROPHOSPHATE, FAD}</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3-ISOPROPYLMALDEHYDROG-CPLX</td>
      <td>3-isopropylmalate dehydrogenase</td>
      <td>{'3-ISOPROPYLMALDEHYDROG-CPLX': 1, '3-ISOPROPY...</td>
      <td>{'3-ISOPROPYLMALDEHYDROG-MONOMER': 2}</td>
      <td>{MN+2, MG+2}</td>
    </tr>
    <tr>
      <th>3</th>
      <td>3-ISOPROPYLMALISOM-CPLX</td>
      <td>3-isopropylmalate dehydratase</td>
      <td>{'3-ISOPROPYLMALISOM-CPLX': 1, 'LEUC-MONOMER':...</td>
      <td>{'LEUC-MONOMER': 1, 'LEUD-MONOMER': 1}</td>
      <td>{CPD-7}</td>
    </tr>
    <tr>
      <th>4</th>
      <td>3-METHYL-2-OXOBUT-OHCH3XFER-CPLX</td>
      <td>3-methyl-2-oxobutanoate hydroxymethyltransferase</td>
      <td>{'3-METHYL-2-OXOBUT-OHCH3XFER-CPLX': 1, '3-CH3...</td>
      <td>{'3-CH3-2-OXOBUTANOATE-OH-CH3-XFER-MONOMER': 10}</td>
      <td>{MG+2}</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>1088</th>
      <td>CPLX0-3964</td>
      <td>ribosome</td>
      <td>{'CPLX0-3964': 1, 'CPLX0-3953': -1, 'CPLX0-396...</td>
      <td>{'EG10900-MONOMER': 1, 'EG10901-MONOMER': 1, '...</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>1089</th>
      <td>CPLX0-8028</td>
      <td>CsrA complex with McaS RNA</td>
      <td>{'CPLX0-8028': 1, 'IS061-RNA': -1, 'CPLX0-7956...</td>
      <td>{'EG11447-MONOMER': 2}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>1090</th>
      <td>CPLX0-8053</td>
      <td>SelB-L-selenocysteinyl-tRNA&lt;sup&gt;sec&lt;/sup&gt;</td>
      <td>{'CPLX0-8053': 1, 'EG10942-MONOMER': -1}</td>
      <td>{'EG10942-MONOMER': 1}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>1091</th>
      <td>CPLX0-8253</td>
      <td>CsrA complex with CsrC RNA</td>
      <td>{'CPLX0-8253': 1, 'CSRC-RNA': -1, 'EG11447-MON...</td>
      <td>{'EG11447-MONOMER': 9}</td>
      <td>{}</td>
    </tr>
    <tr>
      <th>1092</th>
      <td>SRP-CPLX</td>
      <td>Signal Recognition Particle</td>
      <td>{'SRP-CPLX': 1, 'EG10300-MONOMER': -1, 'FFS-RN...</td>
      <td>{'EG10300-MONOMER': 1}</td>
      <td>{}</td>
    </tr>
  </tbody>
</table>
<p>1093 rows × 5 columns</p>
</div></div>
</div>
</section>
<section id="Create-tree-matrix-(also-for-Julia)">
<h2>Create tree matrix (also for Julia)<a class="headerlink" href="#Create-tree-matrix-(also-for-Julia)" title="Link to this heading"></a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># save names</span>
<span class="n">complex_ids</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_complex_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">])</span>
<span class="n">protein_ids</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">])</span>
<span class="n">cofactor_ids</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_cofactor_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">])</span>
<span class="n">pathway_ids</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">])</span>


<span class="n">name_idx</span> <span class="o">=</span> <span class="n">complex_ids</span> <span class="o">+</span> <span class="n">protein_ids</span> <span class="o">+</span> <span class="n">cofactor_ids</span>
<span class="n">tree_matrix</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">([</span><span class="nb">len</span><span class="p">(</span><span class="n">complex_ids</span><span class="p">)</span> <span class="o">+</span> <span class="nb">len</span><span class="p">(</span><span class="n">protein_ids</span><span class="p">)</span> <span class="o">+</span> <span class="nb">len</span><span class="p">(</span><span class="n">cofactor_ids</span><span class="p">),</span> <span class="nb">len</span><span class="p">(</span><span class="n">complex_ids</span><span class="p">)</span> <span class="o">+</span> <span class="nb">len</span><span class="p">(</span><span class="n">protein_ids</span><span class="p">)</span> <span class="o">+</span> <span class="nb">len</span><span class="p">(</span><span class="n">cofactor_ids</span><span class="p">)],</span> <span class="n">dtype</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">)</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_complex_df</span><span class="p">)):</span>
    <span class="n">name</span> <span class="o">=</span> <span class="n">parsed_complex_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;id&#39;</span><span class="p">]</span>
    <span class="n">tree_structure</span><span class="p">,</span> <span class="n">nodes</span> <span class="o">=</span> <span class="n">recursive_component_tree</span><span class="p">(</span><span class="n">name</span><span class="p">,</span> <span class="n">parsed_complex_df</span><span class="p">,</span> <span class="n">parsed_protein_df</span><span class="p">,</span> <span class="n">return_cofactors</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">node</span> <span class="ow">in</span> <span class="n">nodes</span><span class="p">:</span>
        <span class="n">node_name</span> <span class="o">=</span> <span class="n">node</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]</span>
        <span class="n">node_children</span> <span class="o">=</span> <span class="n">node</span><span class="p">[</span><span class="s1">&#39;children&#39;</span><span class="p">]</span>

        <span class="k">if</span> <span class="n">node_children</span> <span class="o">!=</span> <span class="kc">None</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">child_name</span><span class="p">,</span> <span class="n">child_coefficient</span> <span class="ow">in</span> <span class="n">node_children</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
                <span class="k">if</span> <span class="n">child_name</span> <span class="ow">in</span> <span class="n">name_idx</span><span class="p">:</span>
                        <span class="n">tree_matrix</span><span class="p">[</span><span class="n">name_idx</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">node_name</span><span class="p">),</span> <span class="n">name_idx</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">child_name</span><span class="p">)]</span> <span class="o">=</span> <span class="n">child_coefficient</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
component CPLX0-7701 not found in complex or protein tables
component CPLX0-7677 not found in complex or protein tables
component MONOMER0-1781 not found in complex or protein tables
component CPLX0-7702 not found in complex or protein tables
component CSRB-RNA not found in complex or protein tables
component RNPB-RNA not found in complex or protein tables
component RRSA-RRNA not found in complex or protein tables
component RRLA-RRNA not found in complex or protein tables
component RRFA-RRNA not found in complex or protein tables
component RRSA-RRNA not found in complex or protein tables
component RRLA-RRNA not found in complex or protein tables
component RRFA-RRNA not found in complex or protein tables
component IS061-RNA not found in complex or protein tables
component CSRC-RNA not found in complex or protein tables
component FFS-RNA not found in complex or protein tables
</pre></div></div>
</div>
</section>
</section>
<section id="Create-matrices-to-get-cofactor-counts">
<h1>Create matrices to get cofactor counts<a class="headerlink" href="#Create-matrices-to-get-cofactor-counts" title="Link to this heading"></a></h1>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_pathway_df</span><span class="p">[</span><span class="n">parsed_pathway_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Regulation of transcription&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>id</th>
      <th>parents</th>
      <th>children</th>
      <th>level</th>
      <th>common_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1174</th>
      <td>Regulation of transcription</td>
      <td>[Other functions]</td>
      <td>[Regulation of transcription (child)]</td>
      <td>2</td>
      <td>Regulation of transcription</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">priority_list</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Activation-Inactivation-Interconversion&#39;</span><span class="p">,</span><span class="s1">&#39;Metabolic-Clusters&#39;</span><span class="p">,</span> <span class="s1">&#39;Macromolecule-Modification&#39;</span><span class="p">,</span> <span class="s1">&#39;Glycan-Pathways&#39;</span><span class="p">,</span><span class="s1">&#39;Detoxification&#39;</span><span class="p">,</span>  <span class="s1">&#39;Degradation&#39;</span><span class="p">]</span>

<span class="c1"># pathway matrix is necessary to traverse tree</span>
<span class="n">pathway_matrix</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">((</span><span class="nb">len</span><span class="p">(</span><span class="n">pathway_ids</span><span class="p">),</span> <span class="nb">len</span><span class="p">(</span><span class="n">pathway_ids</span><span class="p">)),</span> <span class="n">dtype</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">)</span>
<span class="n">level_vector</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">pathway_ids</span><span class="p">),</span> <span class="n">dtype</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">)</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="p">)):</span>

    <span class="n">cur_pathway</span> <span class="o">=</span> <span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;id&#39;</span><span class="p">]</span>
    <span class="n">level_vector</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;level&#39;</span><span class="p">]</span>

    <span class="n">pathway_parents</span> <span class="o">=</span> <span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;parents&#39;</span><span class="p">]</span>
    <span class="n">pathway_children</span> <span class="o">=</span> <span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;children&#39;</span><span class="p">]</span>

    <span class="k">for</span> <span class="n">parent</span> <span class="ow">in</span> <span class="n">pathway_parents</span><span class="p">:</span>
        <span class="n">j</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">parent</span><span class="p">)</span>
        <span class="n">pathway_matrix</span><span class="p">[</span><span class="n">j</span><span class="p">,</span> <span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>

    <span class="k">for</span> <span class="n">child</span> <span class="ow">in</span> <span class="n">pathway_children</span><span class="p">:</span>
        <span class="n">j</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">child</span><span class="p">)</span>
        <span class="n">pathway_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">j</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>

<span class="n">original_pathway_matrix</span> <span class="o">=</span> <span class="n">pathway_matrix</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>

<span class="c1"># get superpathway indices</span>
<span class="n">super_pathway_idx</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;Super-Pathways&#39;</span><span class="p">)</span>
<span class="n">super_pathway_children_idxs</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">pathway_matrix</span><span class="p">[</span><span class="n">super_pathway_idx</span><span class="p">,</span> <span class="p">:]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>

<span class="c1"># zero out all superpathway children</span>
<span class="n">pathway_matrix</span><span class="p">[:,</span> <span class="n">super_pathway_children_idxs</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>

<span class="c1"># for columns (children) with multiple parents, if one parent leads to degradation or glycans, remove it.</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">pathway_matrix</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="p">:])):</span>
    <span class="n">cur_pathway</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>

    <span class="k">if</span> <span class="n">pathway_matrix</span><span class="p">[:,</span> <span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>

        <span class="n">nz_idxs</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">pathway_matrix</span><span class="p">[:,</span> <span class="n">i</span><span class="p">]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
        <span class="n">top_level_classes</span> <span class="o">=</span> <span class="p">[</span><span class="nb">list</span><span class="p">(</span><span class="n">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">j</span><span class="p">,</span> <span class="n">original_pathway_matrix</span><span class="p">,</span> <span class="n">pathway_ids</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">keys</span><span class="p">())[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="n">nz_idxs</span><span class="p">]</span>
        <span class="c1"># print(f&quot;multiple parents {top_level_classes} for {cur_pathway}&quot;)</span>


        <span class="c1"># when there are multiple parents, remove them in the following order of priority:</span>
        <span class="k">for</span> <span class="n">priority</span> <span class="ow">in</span> <span class="n">priority_list</span><span class="p">:</span>
            <span class="k">while</span> <span class="n">priority</span> <span class="ow">in</span> <span class="n">top_level_classes</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="n">nz_idxs</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                <span class="n">priority_index</span> <span class="o">=</span> <span class="n">top_level_classes</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">priority</span><span class="p">)</span>
                <span class="n">pathway_matrix</span><span class="p">[</span><span class="n">nz_idxs</span><span class="p">[</span><span class="n">priority_index</span><span class="p">],</span> <span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
                <span class="n">nz_idxs</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">pathway_matrix</span><span class="p">[:,</span> <span class="n">i</span><span class="p">]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
                <span class="n">top_level_classes</span><span class="p">[</span><span class="n">priority_index</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;N/A&#39;</span>


        <span class="c1"># then, if there are still multiple parents, remove all but the first one</span>
        <span class="c1"># TODO Change to parent with most frequently occuring 2nd parent.</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">nz_idxs</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="c1"># top_two_level_classes = [list(get_pathway_ith_level_parents(j, original_pathway_matrix, pathway_name_list, level_vector, level=2).keys())[0] for j in nz_idxs]</span>
            <span class="c1"># print(f&quot;multiple parents with 2nd level categories {top_two_level_classes} for {cur_pathway}&quot;)</span>
            <span class="n">pathway_matrix</span><span class="p">[</span><span class="n">nz_idxs</span><span class="p">[</span><span class="mi">1</span><span class="p">:],</span> <span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>

        <span class="n">nz_idxs</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">pathway_matrix</span><span class="p">[:,</span> <span class="n">i</span><span class="p">]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
        <span class="n">top_level_classes</span> <span class="o">=</span> <span class="p">[</span><span class="nb">list</span><span class="p">(</span><span class="n">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">j</span><span class="p">,</span> <span class="n">original_pathway_matrix</span><span class="p">,</span> <span class="n">pathway_ids</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">keys</span><span class="p">())</span> <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="n">nz_idxs</span><span class="p">]</span>

        <span class="c1"># print(f&quot;pruned to {top_level_classes}&quot;)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># if choice can be made, don&#39;t pick these.</span>
<span class="n">priority_list_second</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;SECONDARY-METABOLITE-BIOSYNTHESIS&#39;</span><span class="p">,</span> <span class="s1">&#39;Respiration&#39;</span><span class="p">]</span>

<span class="c1"># create protein name to pathway mapping</span>
<span class="n">W</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">((</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">),</span> <span class="nb">len</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">index</span><span class="p">)))</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>

    <span class="n">cur_pathways</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;direct_annotations&#39;</span><span class="p">]</span>

    <span class="k">for</span> <span class="n">pathway</span> <span class="ow">in</span> <span class="n">cur_pathways</span><span class="p">:</span>
        <span class="n">pathway_idx</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">pathway</span><span class="p">)</span>
        <span class="n">W</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">pathway_idx</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>


<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>

    <span class="n">cur_protein_pathways_idxs</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">W</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="p">:]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">cur_protein</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;id&#39;</span><span class="p">]</span>

    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">cur_protein_pathways_idxs</span><span class="p">)</span> <span class="o">&lt;</span> <span class="mi">2</span><span class="p">:</span>
        <span class="k">continue</span>

    <span class="k">else</span><span class="p">:</span>

        <span class="c1"># for pathway_idx in cur_protein_pathways_idxs:</span>

        <span class="c1"># get top level class of every pathway, and remove all but the first one of each class.</span>

        <span class="n">cur_protein_pathway_parents</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="n">cur_protein_pathway_two_parents</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>

        <span class="k">for</span> <span class="n">pathway_idx</span> <span class="ow">in</span> <span class="n">cur_protein_pathways_idxs</span><span class="p">:</span>
            <span class="n">top_parents</span> <span class="o">=</span> <span class="n">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">pathway_idx</span><span class="p">,</span> <span class="n">pathway_matrix</span><span class="p">,</span> <span class="n">pathway_ids</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
            <span class="n">top_two_parents</span> <span class="o">=</span> <span class="n">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">pathway_idx</span><span class="p">,</span> <span class="n">pathway_matrix</span><span class="p">,</span> <span class="n">pathway_ids</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>

            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">cur_protein</span><span class="si">}</span><span class="s2"> has 2nd parents </span><span class="si">{</span><span class="n">top_two_parents</span><span class="si">}</span><span class="s2"> for </span><span class="si">{</span><span class="n">pathway_ids</span><span class="p">[</span><span class="n">pathway_idx</span><span class="p">]</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">top_parents</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;multiple parents </span><span class="si">{</span><span class="n">top_parents</span><span class="si">}</span><span class="s2"> for </span><span class="si">{</span><span class="n">pathway_ids</span><span class="p">[</span><span class="n">pathway_idx</span><span class="p">]</span><span class="si">}</span><span class="s2"> for </span><span class="si">{</span><span class="n">cur_protein</span><span class="si">}</span><span class="s2">, should not happen.&quot;</span><span class="p">)</span>

            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">top_parents</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
                <span class="n">cur_protein_pathway_parents</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">top_parents</span><span class="o">.</span><span class="n">keys</span><span class="p">())[</span><span class="mi">0</span><span class="p">])</span>
                <span class="n">cur_protein_pathway_two_parents</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">top_two_parents</span><span class="o">.</span><span class="n">keys</span><span class="p">())[</span><span class="mi">0</span><span class="p">])</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="c1"># remove pathway with no parents</span>
                <span class="n">W</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">pathway_idx</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
                <span class="n">cur_protein_pathway_parents</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;N/A&#39;</span><span class="p">)</span>
                <span class="n">cur_protein_pathway_two_parents</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;N/A&#39;</span><span class="p">)</span>


        <span class="c1"># TODO - remove direct annotations with deprioritized parents</span>
        <span class="k">for</span> <span class="n">priority</span> <span class="ow">in</span> <span class="n">priority_list</span><span class="p">:</span>
            <span class="k">while</span> <span class="n">priority</span> <span class="ow">in</span> <span class="n">cur_protein_pathway_parents</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">W</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="p">:]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                <span class="n">priority_index</span> <span class="o">=</span> <span class="n">cur_protein_pathway_parents</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">priority</span><span class="p">)</span>
                <span class="n">W</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">cur_protein_pathways_idxs</span><span class="p">[</span><span class="n">priority_index</span><span class="p">]]</span> <span class="o">=</span> <span class="mi">0</span>
                <span class="n">cur_protein_pathway_parents</span><span class="p">[</span><span class="n">priority_index</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;N/A&#39;</span>

        <span class="c1"># same for 2nd level</span>
        <span class="k">for</span> <span class="n">priority</span> <span class="ow">in</span> <span class="n">priority_list_second</span><span class="p">:</span>
            <span class="k">while</span> <span class="n">priority</span> <span class="ow">in</span> <span class="n">cur_protein_pathway_two_parents</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">W</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="p">:]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                <span class="n">priority_index</span> <span class="o">=</span> <span class="n">cur_protein_pathway_two_parents</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">priority</span><span class="p">)</span>
                <span class="n">W</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">cur_protein_pathways_idxs</span><span class="p">[</span><span class="n">priority_index</span><span class="p">]]</span> <span class="o">=</span> <span class="mi">0</span>
                <span class="n">cur_protein_pathway_two_parents</span><span class="p">[</span><span class="n">priority_index</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;N/A&#39;</span>

        <span class="n">new_protein_pathways_idxs</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">W</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="p">:]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>

        <span class="c1"># for pathway_idx in cur_protein_pathways_idxs:</span>

        <span class="c1"># get top level class of every pathway, and remove all but the first one of each class.</span>

        <span class="n">new_protein_pathway_parents</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="n">new_protein_pathway_two_parents</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>

        <span class="k">for</span> <span class="n">pathway_idx</span> <span class="ow">in</span> <span class="n">new_protein_pathways_idxs</span><span class="p">:</span>
            <span class="n">top_parents</span> <span class="o">=</span> <span class="n">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">pathway_idx</span><span class="p">,</span> <span class="n">pathway_matrix</span><span class="p">,</span> <span class="n">pathway_ids</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
            <span class="n">top_two_parents</span> <span class="o">=</span> <span class="n">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">pathway_idx</span><span class="p">,</span> <span class="n">pathway_matrix</span><span class="p">,</span> <span class="n">pathway_ids</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>

            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">top_parents</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
                <span class="n">new_protein_pathway_parents</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">top_parents</span><span class="o">.</span><span class="n">keys</span><span class="p">())[</span><span class="mi">0</span><span class="p">])</span>
                <span class="n">new_protein_pathway_two_parents</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">top_two_parents</span><span class="o">.</span><span class="n">keys</span><span class="p">())[</span><span class="mi">0</span><span class="p">])</span>

        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;pruned to </span><span class="si">{</span><span class="n">cur_protein_pathway_two_parents</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>


        <span class="c1"># remove all N/A</span>
        <span class="c1"># cur_protein_pathways_idxs = cur_protein_pathways_idxs[cur_protein_pathway_parents != &#39;N/A&#39;]</span>

        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">unique</span><span class="p">(</span><span class="n">cur_protein_pathway_parents</span><span class="p">))</span> <span class="o">&lt;</span> <span class="mi">2</span><span class="p">:</span>
            <span class="k">continue</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1-ACYLGLYCEROL-3-P-ACYLTRANSFER-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5667
1-ACYLGLYCEROL-3-P-ACYLTRANSFER-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1319
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
2-OCTAPRENYL-METHOXY-BENZOQ-METH-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6708
2-OCTAPRENYL-METHOXY-BENZOQ-METH-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for MENAQUINONESYN-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
3-OXOACYL-ACP-REDUCT-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-862
3-OXOACYL-ACP-REDUCT-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-6282
3-OXOACYL-ACP-REDUCT-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5973
3-OXOACYL-ACP-REDUCT-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5971
3-OXOACYL-ACP-REDUCT-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for FASYN-ELONG-PWY
3-OXOACYL-ACP-REDUCT-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for PWY-6519
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;]
3-OXOACYL-ACP-SYNTHII-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5966
3-OXOACYL-ACP-SYNTHII-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5965
3-OXOACYL-ACP-SYNTHII-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5973
3-OXOACYL-ACP-SYNTHII-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5971
3-OXOACYL-ACP-SYNTHII-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for FASYN-ELONG-PWY
3-OXOACYL-ACP-SYNTHII-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for PWY-6519
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;]
6PFK-1-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
6PFK-1-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
6PFK-1-MONOMER has 2nd parents {&#39;PWY0-1517&#39;: 2} for PWY0-1517
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;PWY0-1517&#39;]
6PFK-2-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
6PFK-2-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
6PFK-2-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for GALACTITOLCAT-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Alcohol-Degradation&#39;]
6PGLUCONOLACT-MONOMER has 2nd parents {&#39;Pentose-Phosphate-Cycle&#39;: 2} for OXIDATIVEPENT-PWY
6PGLUCONOLACT-MONOMER has 2nd parents {&#39;GLYCOLYSIS-E-D&#39;: 2} for GLYCOLYSIS-E-D
pruned to [&#39;Pentose-Phosphate-Cycle&#39;, &#39;N/A&#39;]
ACETATEKINA-MONOMER has 2nd parents {&#39;PWY-5485&#39;: 2} for PWY-5485
ACETATEKINA-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY0-1312
ACETATEKINA-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
ACETATEKINA-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5437
ACETATEKINA-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY0-1477
pruned to [&#39;N/A&#39;, &#39;Fermentation&#39;, &#39;N/A&#39;, &#39;Fermentation&#39;, &#39;AMINE-DEG&#39;]
ACETOLACTSYNIII-HCHAIN-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for VALSYN-PWY
ACETOLACTSYNIII-HCHAIN-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ILEUSYN-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
ACETOLACTSYNIII-ICHAIN-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for VALSYN-PWY
ACETOLACTSYNIII-ICHAIN-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ILEUSYN-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
ACETYL-COA-ACETYLTRANSFER-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for PWY0-1337
ACETYL-COA-ACETYLTRANSFER-MONOMER has 2nd parents {&#39;Other-Degradation&#39;: 2} for ACETOACETATE-DEG-PWY
pruned to [&#39;Fatty-Acid-and-Lipid-Degradation&#39;, &#39;Other-Degradation&#39;]
ACETYLORNTRANSAM-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLUTORN-PWY
ACETYLORNTRANSAM-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for DAPLYSINESYN-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
ACONITASE-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
ACONITASE-MONOMER has 2nd parents {&#39;GLYOXYLATE-BYPASS&#39;: 2} for GLYOXYLATE-BYPASS
pruned to [&#39;TCA-VARIANTS&#39;, &#39;GLYOXYLATE-BYPASS&#39;]
ACONITATEDEHYDRB-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
ACONITATEDEHYDRB-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
ACONITATEDEHYDRB-MONOMER has 2nd parents {&#39;GLYOXYLATE-BYPASS&#39;: 2} for GLYOXYLATE-BYPASS
ACONITATEDEHYDRB-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY0-42
pruned to [&#39;N/A&#39;, &#39;TCA-VARIANTS&#39;, &#39;GLYOXYLATE-BYPASS&#39;, &#39;CARBOXYLATES-DEG&#39;]
ACPSUB-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for P2-PWY
ACPSUB-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-6038
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;CARBOXYLATES-DEG&#39;]
ACS-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY0-42
ACS-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY0-1313
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
ACSERLYA-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for CYSTSYN-PWY
ACSERLYA-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for LCYSDEG-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;]
ACSERLYB-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-7870
ACSERLYB-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for CYSTSYN-PWY
ACSERLYB-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for LCYSDEG-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;]
ACYLCOADEHYDROG-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for PWY0-1337
ACYLCOADEHYDROG-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for FAO-PWY
pruned to [&#39;Fatty-Acid-and-Lipid-Degradation&#39;, &#39;Fatty-Acid-and-Lipid-Degradation&#39;]
ADENODEAMIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6611
ADENODEAMIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6609
ADENODEAMIN-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
ADENODEAMIN-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1296
ADENODEAMIN-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-7179
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
ADENPRIBOSYLTRAN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6605
ADENPRIBOSYLTRAN-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-7805
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Noncarbon-Nutrients&#39;]
ADENYL-KIN-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
ADENYL-KIN-MONOMER has 2nd parents {&#39;PWY-7197&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7197
ADENYL-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
ADENYL-KIN-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
ADENYL-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7187
ADENYL-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7219
ADENYL-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7221
ADENYL-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-163
ADENYL-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7205
ADENYL-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7176
ADENYL-KIN-MONOMER has 2nd parents {&#39;Metabolic-Regulators&#39;: 2} for PPGPPMET-PWY
pruned to [&#39;PWY-7184&#39;, &#39;PWY-7197&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Metabolic-Regulators&#39;]
ADHE-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
ADHE-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5480
ADHE-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for PWY-7180
ADHE-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY-5436
ADHE-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for ETOH-ACETYLCOA-ANA-PWY
ADHE-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY0-1477
pruned to [&#39;N/A&#39;, &#39;Fermentation&#39;, &#39;Carbohydrates-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Alcohol-Degradation&#39;, &#39;AMINE-DEG&#39;]
ADHP-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
ADHP-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5480
ADHP-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for ETOH-ACETYLCOA-ANA-PWY
ADHP-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY0-1477
pruned to [&#39;N/A&#39;, &#39;Fermentation&#39;, &#39;Alcohol-Degradation&#39;, &#39;AMINE-DEG&#39;]
AERGLYC3PDEHYDROG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1561
AERGLYC3PDEHYDROG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1584
AERGLYC3PDEHYDROG-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY-6952
AERGLYC3PDEHYDROG-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY-4261
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Alcohol-Degradation&#39;, &#39;Alcohol-Degradation&#39;]
AGMATIN-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for PWY-40
AGMATIN-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-823
pruned to [&#39;Polyamine-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;]
AIRS-MONOMER has 2nd parents {&#39;PWY-6277&#39;: 2} for PWY-6277
AIRS-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6122
AIRS-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6121
pruned to [&#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
ALARACEBIOSYN-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-8072
ALARACEBIOSYN-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for ALADEG-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;]
ALARACECAT-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-8072
ALARACECAT-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for ALADEG-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;]
ALAS-MONOMER has 2nd parents {&#39;TRNA-CHARGING-PWY&#39;: 2, &#39;Aminoacyl-tRNAs-Charging&#39;: 2} for TRNA-CHARGING-PWY
ALAS-MONOMER has 2nd parents {&#39;ENTBACSYN-PWY&#39;: 2} for ENTBACSYN-PWY
pruned to [&#39;TRNA-CHARGING-PWY&#39;, &#39;N/A&#39;]
ALDHDEHYDROG-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY0-1221
ALDHDEHYDROG-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PUTDEG-PWY
ALDHDEHYDROG-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY-6535
pruned to [&#39;AMINE-DEG&#39;, &#39;AMINE-DEG&#39;, &#39;AMINE-DEG&#39;]
ALKAPHOSPHA-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
ALKAPHOSPHA-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for TRESYN-PWY
ALKAPHOSPHA-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
ALKAPHOSPHA-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-7805
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;NUCLEO-DEG&#39;, &#39;Noncarbon-Nutrients&#39;]
AMINEOXID-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for THRDLCTCAT-PWY
AMINEOXID-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for 2PHENDEG-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;AMINE-DEG&#39;]
ANGLYC3PDEHYDROGSUBUNITA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1591
ANGLYC3PDEHYDROGSUBUNITA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1582
ANGLYC3PDEHYDROGSUBUNITA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1581
ANGLYC3PDEHYDROGSUBUNITA-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY-6952
ANGLYC3PDEHYDROGSUBUNITA-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY-4261
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Alcohol-Degradation&#39;, &#39;Alcohol-Degradation&#39;]
ANGLYC3PDEHYDROGSUBUNITB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1591
ANGLYC3PDEHYDROGSUBUNITB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1582
ANGLYC3PDEHYDROGSUBUNITB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1581
ANGLYC3PDEHYDROGSUBUNITB-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY-6952
ANGLYC3PDEHYDROGSUBUNITB-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY-4261
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Alcohol-Degradation&#39;, &#39;Alcohol-Degradation&#39;]
ANGLYC3PDEHYDROGSUBUNITC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1591
ANGLYC3PDEHYDROGSUBUNITC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1582
ANGLYC3PDEHYDROGSUBUNITC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1581
ANGLYC3PDEHYDROGSUBUNITC-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY-6952
ANGLYC3PDEHYDROGSUBUNITC-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY-4261
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Alcohol-Degradation&#39;, &#39;Alcohol-Degradation&#39;]
ANSA-MONOMER has 2nd parents {&#39;ASPASN-PWY&#39;: 2} for ASPASN-PWY
ANSA-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for ASPARAGINE-DEG1-PWY
pruned to [&#39;N/A&#39;, &#39;Amino-Acid-Degradation&#39;]
ANSB-MONOMER has 2nd parents {&#39;ASPASN-PWY&#39;: 2} for ASPASN-PWY
ANSB-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for ASPARAGINE-DEG1-PWY
pruned to [&#39;N/A&#39;, &#39;Amino-Acid-Degradation&#39;]
ANTHRANSYNCOMPI-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLUTSYN-PWY
ANTHRANSYNCOMPI-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for TRPSYN-PWY
ANTHRANSYNCOMPI-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
ANTHRANSYNCOMPI-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
ANTHRANSYNCOMPII-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLUTSYN-PWY
ANTHRANSYNCOMPII-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for TRPSYN-PWY
ANTHRANSYNCOMPII-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
ANTHRANSYNCOMPII-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
APHA-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
APHA-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-6608
pruned to [&#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
ARGDECARBOXBIO-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for PWY-40
ARGDECARBOXBIO-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-823
pruned to [&#39;Polyamine-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;]
ARGDECARBOXDEG-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-823
ARGDECARBOXDEG-MONOMER has 2nd parents {&#39;Acid-Resistance&#39;: 2} for PWY0-1299
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Acid-Resistance&#39;]
ASL-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6123
ASL-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7219
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
ASNSYNB-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ASPARAGINE-BIOSYNTHESIS
ASNSYNB-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ASPARAGINESYN-PWY
ASNSYNB-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
ASNSYNB-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
ASP-SEMIALDEHYDE-DEHYDROGENASE-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for HOMOSERSYN-PWY
ASP-SEMIALDEHYDE-DEHYDROGENASE-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for DAPLYSINESYN-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
ASPAMINOTRANS-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ASPARTATESYN-PWY
ASPAMINOTRANS-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PHESYN
ASPAMINOTRANS-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for TYRSYN
ASPAMINOTRANS-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-5329
ASPAMINOTRANS-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTDEG-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Noncarbon-Nutrients&#39;, &#39;Amino-Acid-Degradation&#39;]
ASPARTASE-MONOMER has 2nd parents {&#39;ASPASN-PWY&#39;: 2} for ASPASN-PWY
ASPARTASE-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTDEG-PWY
ASPARTASE-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY-8291
ASPARTASE-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY-8294
pruned to [&#39;N/A&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
ASPKINIII-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for HOMOSERSYN-PWY
ASPKINIII-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for DAPLYSINESYN-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
BRANCHED-CHAINAMINOTRANSFER-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for VALSYN-PWY
BRANCHED-CHAINAMINOTRANSFER-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ILEUSYN-PWY
BRANCHED-CHAINAMINOTRANSFER-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ALANINE-VALINESYN-PWY
BRANCHED-CHAINAMINOTRANSFER-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PHESYN
BRANCHED-CHAINAMINOTRANSFER-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for LEUSYN-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
CARBOXYL-TRANSFERASE-ALPHA-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-4381
CARBOXYL-TRANSFERASE-ALPHA-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1264
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
CARBOXYL-TRANSFERASE-BETA-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-4381
CARBOXYL-TRANSFERASE-BETA-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1264
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
CARBPSYN-LARGE has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-5686
CARBPSYN-LARGE has 2nd parents {&#39;ARGSYN-PWY&#39;: 2} for ARGSYN-PWY
CARBPSYN-LARGE has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
CARBPSYN-LARGE has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
CARBPSYN-SMALL has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-5686
CARBPSYN-SMALL has 2nd parents {&#39;ARGSYN-PWY&#39;: 2} for ARGSYN-PWY
CARBPSYN-SMALL has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
CARBPSYN-SMALL has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
CDPDIGLYSYN-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5667
CDPDIGLYSYN-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1319
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
CHEA-SMALL has 2nd parents {&#39;PWY0-1518&#39;: 2} for PWY0-1518
CHEA-SMALL has 2nd parents {&#39;PWY0-1519&#39;: 2} for PWY0-1519
pruned to [&#39;PWY0-1518&#39;, &#39;PWY0-1519&#39;]
CHEY-MONOMER has 2nd parents {&#39;PWY0-1518&#39;: 2} for PWY0-1518
CHEY-MONOMER has 2nd parents {&#39;PWY0-1519&#39;: 2} for PWY0-1519
pruned to [&#39;PWY0-1518&#39;, &#39;PWY0-1519&#39;]
CHORISMUTPREPHENDEHYDRAT-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PHESYN
CHORISMUTPREPHENDEHYDRAT-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for TYRSYN
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
CHORISMUTPREPHENDEHYDROG-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PHESYN
CHORISMUTPREPHENDEHYDROG-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for TYRSYN
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
CITRYLY-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for P2-PWY
CITRYLY-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-6038
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;CARBOXYLATES-DEG&#39;]
CITSYN-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
CITSYN-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
CITSYN-MONOMER has 2nd parents {&#39;GLYOXYLATE-BYPASS&#39;: 2} for GLYOXYLATE-BYPASS
pruned to [&#39;N/A&#39;, &#39;TCA-VARIANTS&#39;, &#39;GLYOXYLATE-BYPASS&#39;]
CITTRANS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for P2-PWY
CITTRANS-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-6038
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;CARBOXYLATES-DEG&#39;]
CMPKI-MONOMER has 2nd parents {&#39;PWY-7197&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7197
CMPKI-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-163
CMPKI-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7205
pruned to [&#39;PWY-7197&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
COBALADENOSYLTRANS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-7971
COBALADENOSYLTRANS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6268
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
COBS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-5509
COBS-MONOMER has 2nd parents {&#39;COBALSYN-PWY&#39;: 2} for COBALSYN-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;N/A&#39;]
COBU-MONOMER has 2nd parents {&#39;COBALSYN-PWY&#39;: 2} for COBALSYN-PWY
COBU-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-7971
pruned to [&#39;N/A&#39;, &#39;Cofactor-Biosynthesis&#39;]
CTPSYN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7176
CTPSYN-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
CTPSYN-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
CYCLASE-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for HISTSYN-PWY
CYCLASE-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
CYCLASE-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
CYDA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1353
CYDA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
CYDA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1568
CYDA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7545
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
CYDB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1353
CYDB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
CYDB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1568
CYDB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7545
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
CYOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
CYOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1565
CYOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1329
CYOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1567
CYOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7544
CYOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1544
CYOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1561
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
CYOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
CYOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1565
CYOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1329
CYOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1567
CYOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7544
CYOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1544
CYOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1561
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
CYOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
CYOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1565
CYOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1329
CYOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1567
CYOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7544
CYOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1544
CYOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1561
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
CYOD-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
CYOD-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1565
CYOD-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1329
CYOD-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1567
CYOD-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7544
CYOD-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1544
CYOD-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1561
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
CYSTATHIONINE-BETA-LYASE-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-8072
CYSTATHIONINE-BETA-LYASE-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for HOMOSER-METSYN-PWY
CYSTATHIONINE-BETA-LYASE-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for LCYSDEG-PWY
CYSTATHIONINE-BETA-LYASE-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for ALADEG-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
CYTDEAM-MONOMER has 2nd parents {&#39;PWY-7194&#39;: 2} for PWY-7194
CYTDEAM-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-163
CYTDEAM-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7195
pruned to [&#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
CYTIDEAM-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7193
CYTIDEAM-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-163
CYTIDEAM-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6556
CYTIDEAM-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-181
CYTIDEAM-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-7181
CYTIDEAM-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1295
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
CYTOCHROMEC552-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1585
CYTOCHROMEC552-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8147
pruned to [&#39;Electron-Transfer&#39;, &#39;Cofactor-Biosynthesis&#39;]
DCTP-DEAM-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
DCTP-DEAM-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7187
pruned to [&#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;]
DCUA-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY-8291
DCUA-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY-8294
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
DEOA-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-181
DEOA-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-7181
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;NUCLEO-DEG&#39;]
DEOD-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6611
DEOD-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6609
DEOD-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6620
DEOD-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
DEOD-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-6608
DEOD-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1296
DEOD-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-7179
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
DEOXYGLUCONOKIN-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GALACTUROCAT-PWY
DEOXYGLUCONOKIN-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-7242
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
DHPDIOXYGEN-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for PWY-6690
DHPDIOXYGEN-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for HCAMHPDEG-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;, &#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;]
DIHYDROFOLATEREDUCT-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for 1CMET2-PWY
DIHYDROFOLATEREDUCT-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6614
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
DIHYDROXYACIDDEHYDRAT-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for VALSYN-PWY
DIHYDROXYACIDDEHYDRAT-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ILEUSYN-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
DISULFOXRED-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8147
DISULFOXRED-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for GLUT-REDOX-PWY
DISULFOXRED-MONOMER has 2nd parents {&#39;Protein-Modification&#39;: 2} for PWY0-1599
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;, &#39;Protein-Modification&#39;]
DLACTDEHYDROGFAD-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1565
DLACTDEHYDROGFAD-MONOMER has 2nd parents {&#39;Methylglyoxal-Detoxification&#39;: 2, &#39;Aldehyde-Degradation&#39;: 2} for PWY-5386
pruned to [&#39;Electron-Transfer&#39;, &#39;Methylglyoxal-Detoxification&#39;]
DLACTDEHYDROGNAD-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-8274
DLACTDEHYDROGNAD-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
pruned to [&#39;Fermentation&#39;, &#39;N/A&#39;]
DMBPPRIBOSYLTRANS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-5509
DMBPPRIBOSYLTRANS-MONOMER has 2nd parents {&#39;COBALSYN-PWY&#39;: 2} for COBALSYN-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;N/A&#39;]
DMSA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1577
DMSA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1356
DMSA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
DMSB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1577
DMSB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1356
DMSB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
DMSC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1577
DMSC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1356
DMSC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
DSBD-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8147
DSBD-MONOMER has 2nd parents {&#39;Protein-Modification&#39;: 2} for PWY0-1600
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Protein-Modification&#39;]
DTDPGLUCDEHYDRAT-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for DTDPRHAMSYN-PWY
DTDPGLUCDEHYDRAT-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for PWY-7315
pruned to [&#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
DTDPGLUCOSEPP-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for DTDPRHAMSYN-PWY
DTDPGLUCOSEPP-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for PWY-7315
pruned to [&#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
DTDPGLUCOSEPP2-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for DTDPRHAMSYN-PWY
DTDPGLUCOSEPP2-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for PWY-7315
pruned to [&#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
DTMPKI-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
DTMPKI-MONOMER has 2nd parents {&#39;PWY-7197&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7197
DTMPKI-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
DTMPKI-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7187
pruned to [&#39;PWY-7184&#39;, &#39;PWY-7197&#39;, &#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;]
DUTP-PYROP-MONOMER has 2nd parents {&#39;PWY-7206&#39;: 2, &#39;NUCLEO-DEG&#39;: 2} for PWY-7206
DUTP-PYROP-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
DUTP-PYROP-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
DUTP-PYROP-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7187
pruned to [&#39;PWY-7206&#39;, &#39;PWY-7184&#39;, &#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;]
DXS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6892
DXS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PYRIDOXSYN-PWY
DXS-MONOMER has 2nd parents {&#39;SECONDARY-METABOLITE-BIOSYNTHESIS&#39;: 2} for NONMEVIPP-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;N/A&#39;]
E1O-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
E1O-MONOMER has 2nd parents {&#39;Respiration&#39;: 2} for PWY-5084
pruned to [&#39;TCA-VARIANTS&#39;, &#39;N/A&#39;]
E1P-MONOMER has 2nd parents {&#39;Acetyl-CoA-Biosynthesis&#39;: 2} for PYRUVDEHYD-PWY
E1P-MONOMER has 2nd parents {&#39;GLYCOLYSIS-TCA-GLYOX-BYPASS&#39;: 2} for GLYCOLYSIS-TCA-GLYOX-BYPASS
pruned to [&#39;Acetyl-CoA-Biosynthesis&#39;, &#39;N/A&#39;]
E2O-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
E2O-MONOMER has 2nd parents {&#39;Respiration&#39;: 2} for PWY-5084
pruned to [&#39;TCA-VARIANTS&#39;, &#39;N/A&#39;]
E2P-MONOMER has 2nd parents {&#39;Acetyl-CoA-Biosynthesis&#39;: 2} for PYRUVDEHYD-PWY
E2P-MONOMER has 2nd parents {&#39;GLYCOLYSIS-TCA-GLYOX-BYPASS&#39;: 2} for GLYCOLYSIS-TCA-GLYOX-BYPASS
pruned to [&#39;Acetyl-CoA-Biosynthesis&#39;, &#39;N/A&#39;]
E3-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
E3-MONOMER has 2nd parents {&#39;Acetyl-CoA-Biosynthesis&#39;: 2} for PYRUVDEHYD-PWY
E3-MONOMER has 2nd parents {&#39;Respiration&#39;: 2} for PWY-5084
E3-MONOMER has 2nd parents {&#39;GLYCOLYSIS-TCA-GLYOX-BYPASS&#39;: 2} for GLYCOLYSIS-TCA-GLYOX-BYPASS
E3-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for 1CMET2-PWY
E3-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLYCLEAV-PWY
E3-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1608
pruned to [&#39;TCA-VARIANTS&#39;, &#39;Acetyl-CoA-Biosynthesis&#39;, &#39;N/A&#39;, &#39;N/A&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
EG10344-MONOMER has 2nd parents {&#39;PEPTIDOGLYCANSYN-PWY&#39;: 2} for PEPTIDOGLYCANSYN-PWY
EG10344-MONOMER has 2nd parents {&#39;Cell-Structure-Biosynthesis&#39;: 2} for PWY0-1586
pruned to [&#39;N/A&#39;, &#39;Cell-Structure-Biosynthesis&#39;]
EG10355-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for FUCCAT-PWY
EG10355-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for RIBOKIN-PWY
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;Carbohydrates-Degradation&#39;]
EG10564-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-8072
EG10564-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for HOMOSER-METSYN-PWY
EG10564-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for LCYSDEG-PWY
EG10564-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for ALADEG-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
EG10572-MONOMER has 2nd parents {&#39;PWY-7206&#39;: 2, &#39;NUCLEO-DEG&#39;: 2} for PWY-7206
EG10572-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
EG10572-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
EG10572-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7187
pruned to [&#39;PWY-7206&#39;, &#39;PWY-7184&#39;, &#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;]
EG10605-MONOMER has 2nd parents {&#39;PEPTIDOGLYCANSYN-PWY&#39;: 2} for PEPTIDOGLYCANSYN-PWY
EG10605-MONOMER has 2nd parents {&#39;Cell-Structure-Biosynthesis&#39;: 2} for PWY0-1586
pruned to [&#39;N/A&#39;, &#39;Cell-Structure-Biosynthesis&#39;]
EG10607-MONOMER has 2nd parents {&#39;PEPTIDOGLYCANSYN-PWY&#39;: 2} for PEPTIDOGLYCANSYN-PWY
EG10607-MONOMER has 2nd parents {&#39;Cell-Structure-Biosynthesis&#39;: 2} for PWY0-1586
pruned to [&#39;N/A&#39;, &#39;Cell-Structure-Biosynthesis&#39;]
EG10634-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
EG10634-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-6608
pruned to [&#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
EG10718-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY0-1533
EG10718-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-7805
pruned to [&#39;Noncarbon-Nutrients&#39;, &#39;Noncarbon-Nutrients&#39;]
EG10719-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY0-1533
EG10719-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-7805
pruned to [&#39;Noncarbon-Nutrients&#39;, &#39;Noncarbon-Nutrients&#39;]
EG10722-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY0-1533
EG10722-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-7805
pruned to [&#39;Noncarbon-Nutrients&#39;, &#39;Noncarbon-Nutrients&#39;]
EG10723-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY0-1533
EG10723-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-7805
pruned to [&#39;Noncarbon-Nutrients&#39;, &#39;Noncarbon-Nutrients&#39;]
EG10725-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY0-1533
EG10725-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-7805
pruned to [&#39;Noncarbon-Nutrients&#39;, &#39;Noncarbon-Nutrients&#39;]
EG10748-MONOMER has 2nd parents {&#39;PEPTIDOGLYCANSYN-PWY&#39;: 2} for PEPTIDOGLYCANSYN-PWY
EG10748-MONOMER has 2nd parents {&#39;Cell-Structure-Biosynthesis&#39;: 2} for PWY0-1586
pruned to [&#39;N/A&#39;, &#39;Cell-Structure-Biosynthesis&#39;]
EG11082-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6605
EG11082-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6556
EG11082-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7195
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
EG11090-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6151
EG11090-MONOMER has 2nd parents {&#39;SECONDARY-METABOLITE-BIOSYNTHESIS&#39;: 2} for PWY-6153
EG11090-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1391
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;N/A&#39;, &#39;NUCLEO-DEG&#39;]
EG11162-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
EG11162-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
pruned to [&#39;N/A&#39;, &#39;TCA-VARIANTS&#39;]
EG11195-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
EG11195-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
EG11195-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
EG11195-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for GLUCOSE1PMETAB-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Degradation&#39;]
EG11239-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
EG11239-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
EG11239-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
EG11245-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
EG11245-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
EG11245-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
EG11306-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-1275
EG11306-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-501
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
EG11371-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1545
EG11371-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5668
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
EG11470-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
EG11470-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
EG11470-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
EG11470-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for GLUCOSE1PMETAB-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Degradation&#39;]
EG11517-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
EG11517-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
EG11517-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
EG11600-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-5329
EG11600-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-5350
pruned to [&#39;Noncarbon-Nutrients&#39;, &#39;Noncarbon-Nutrients&#39;]
EG11784-MONOMER has 2nd parents {&#39;PWY-5485&#39;: 2} for PWY-5485
EG11784-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
EG11784-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5480
EG11784-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5437
pruned to [&#39;N/A&#39;, &#39;N/A&#39;, &#39;Fermentation&#39;, &#39;Fermentation&#39;]
EG11796-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-1275
EG11796-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-522
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
EG11800-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1577
EG11800-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1578
EG11800-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1576
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
EG11815-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1578
EG11815-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1355
EG11815-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
EG11815-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8147
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Cofactor-Biosynthesis&#39;]
EG11817-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
EG11817-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-6608
pruned to [&#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
EG11829-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-7639
EG11829-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-5964
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
EG12026-MONOMER has 2nd parents {&#39;C1-COMPOUNDS&#39;: 2} for PWY-1801
EG12026-MONOMER has 2nd parents {&#39;Methylglyoxal-Detoxification&#39;: 2, &#39;Aldehyde-Degradation&#39;: 2} for PWY-5386
pruned to [&#39;C1-COMPOUNDS&#39;, &#39;Methylglyoxal-Detoxification&#39;]
EG12030-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6556
EG12030-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7195
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
EG12097-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6611
EG12097-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6609
EG12097-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
EG12097-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1296
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
EG12159-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6611
EG12159-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6609
EG12159-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6620
EG12159-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for SALVPURINE2-PWY
EG12159-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-163
EG12159-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-181
EG12159-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
EG12159-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-6608
EG12159-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1296
EG12159-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-7181
EG12159-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1295
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
EG12221-MONOMER has 2nd parents {&#39;ENTBACSYN-PWY&#39;: 2} for ENTBACSYN-PWY
EG12221-MONOMER has 2nd parents {&#39;Protein-Modification&#39;: 2} for PWY-6012
pruned to [&#39;N/A&#39;, &#39;Protein-Modification&#39;]
EG12244-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
EG12244-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1591
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
EG12286-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for LYXMET-PWY
EG12286-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-6961
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;CARBOXYLATES-DEG&#39;]
EG12287-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for LYXMET-PWY
EG12287-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-6961
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;CARBOXYLATES-DEG&#39;]
EG12293-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for THREONINE-DEG2-PWY
EG12293-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for THRDLCTCAT-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
EG12407-MONOMER has 2nd parents {&#39;ASPASN-PWY&#39;: 2} for ASPASN-PWY
EG12407-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for ASPARAGINE-DEG1-PWY
pruned to [&#39;N/A&#39;, &#39;Amino-Acid-Degradation&#39;]
EG12712-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6151
EG12712-MONOMER has 2nd parents {&#39;SECONDARY-METABOLITE-BIOSYNTHESIS&#39;: 2} for PWY-6153
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;N/A&#39;]
EG50003-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-501
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for NAGLIPASYN-PWY
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1319
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for KDO-LIPASYN-PWY
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5966
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-4381
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5965
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-862
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-6282
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5973
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5971
EG50003-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for FASYN-ELONG-PWY
EG50003-MONOMER has 2nd parents {&#39;KDO-NAGLIPASYN-PWY&#39;: 2} for KDO-NAGLIPASYN-PWY
EG50003-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for PWY-6519
EG50003-MONOMER has 2nd parents {&#39;Protein-Modification&#39;: 2} for PWY-6012
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Other-biosynthesis&#39;, &#39;Protein-Modification&#39;]
ENOLASE-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
ENOLASE-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
ENOLASE-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
ENOYL-ACP-REDUCT-NADH-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-862
ENOYL-ACP-REDUCT-NADH-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-6282
ENOYL-ACP-REDUCT-NADH-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5973
ENOYL-ACP-REDUCT-NADH-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5971
ENOYL-ACP-REDUCT-NADH-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for FASYN-ELONG-PWY
ENOYL-ACP-REDUCT-NADH-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for PWY-6519
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;]
ENTB-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-BIOSYN&#39;: 2} for PWY-5901
ENTB-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1319
ENTB-MONOMER has 2nd parents {&#39;ENTBACSYN-PWY&#39;: 2} for ENTBACSYN-PWY
ENTB-MONOMER has 2nd parents {&#39;Protein-Modification&#39;: 2} for PWY-6012
pruned to [&#39;AROMATIC-COMPOUNDS-BIOSYN&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Protein-Modification&#39;]
ENTF-MONOMER has 2nd parents {&#39;ENTBACSYN-PWY&#39;: 2} for ENTBACSYN-PWY
ENTF-MONOMER has 2nd parents {&#39;Protein-Modification&#39;: 2} for PWY-6012
pruned to [&#39;N/A&#39;, &#39;Protein-Modification&#39;]
F16B-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
F16B-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
F16B-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
FABA-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-862
FABA-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for FASYN-ELONG-PWY
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
FABB-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5966
FABB-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5965
FABB-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-862
FABB-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-6282
FABB-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5971
FABB-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for FASYN-ELONG-PWY
FABB-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for PWY-6519
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;]
FABH-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5966
FABH-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-4381
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
FABZ-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-862
FABZ-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-6282
FABZ-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5973
FABZ-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5971
FABZ-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for FASYN-ELONG-PWY
FABZ-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for PWY-6519
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;]
FADA-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for PWY0-1337
FADA-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for FAO-PWY
pruned to [&#39;Fatty-Acid-and-Lipid-Degradation&#39;, &#39;Fatty-Acid-and-Lipid-Degradation&#39;]
FADB-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for PWY0-1337
FADB-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for FAO-PWY
pruned to [&#39;Fatty-Acid-and-Lipid-Degradation&#39;, &#39;Fatty-Acid-and-Lipid-Degradation&#39;]
FDNG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1356
FDNG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1585
FDNG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1355
FDNG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FDNH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1356
FDNH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1585
FDNH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1355
FDNH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FDNI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1356
FDNI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1585
FDNI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1355
FDNI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FDOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1356
FDOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1585
FDOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1355
FDOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FDOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1356
FDOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1585
FDOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1355
FDOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FDOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1356
FDOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1585
FDOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1355
FDOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FERREDOXIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6892
FERREDOXIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-1275
FERREDOXIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-501
FERREDOXIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-1507
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
FGAMSYN-MONOMER has 2nd parents {&#39;PWY-6277&#39;: 2} for PWY-6277
FGAMSYN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6122
FGAMSYN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6121
pruned to [&#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
FOLC-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-2161
FOLC-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6614
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
FORMATEDEHYDROGH-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
FORMATEDEHYDROGH-MONOMER has 2nd parents {&#39;Hydrogen-Production&#39;: 2} for PWY-6772
pruned to [&#39;N/A&#39;, &#39;Hydrogen-Production&#39;]
FRUCBISALD-CLASSI-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
FRUCBISALD-CLASSI-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
FRUCBISALD-CLASSI-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
FRUCTBISALD-CLASSII-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
FRUCTBISALD-CLASSII-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
FRUCTBISALD-CLASSII-MONOMER has 2nd parents {&#39;PWY0-1517&#39;: 2} for PWY0-1517
FRUCTBISALD-CLASSII-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;PWY0-1517&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
FUCISOM-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for DARABCATK12-PWY
FUCISOM-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for FUCCAT-PWY
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;Carbohydrates-Degradation&#39;]
FUCPALDOL-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for DARABCATK12-PWY
FUCPALDOL-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for FUCCAT-PWY
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;Carbohydrates-Degradation&#39;]
FUCULOKIN-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for DARABCATK12-PWY
FUCULOKIN-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for FUCCAT-PWY
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;Carbohydrates-Degradation&#39;]
FUM-FE-S has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
FUM-FE-S has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1582
FUM-FE-S has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
FUM-FE-S has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1576
pruned to [&#39;N/A&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FUM-FLAVO has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
FUM-FLAVO has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1582
FUM-FLAVO has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
FUM-FLAVO has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1576
pruned to [&#39;N/A&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FUM-MEMB1 has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
FUM-MEMB1 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1582
FUM-MEMB1 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
FUM-MEMB1 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1576
pruned to [&#39;N/A&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FUM-MEMB2 has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
FUM-MEMB2 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1582
FUM-MEMB2 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
FUM-MEMB2 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1576
pruned to [&#39;N/A&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
FUMA-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
FUMA-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
pruned to [&#39;N/A&#39;, &#39;TCA-VARIANTS&#39;]
FUMB-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
FUMB-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
pruned to [&#39;N/A&#39;, &#39;TCA-VARIANTS&#39;]
FUMC-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
FUMC-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
pruned to [&#39;N/A&#39;, &#39;TCA-VARIANTS&#39;]
G6261-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
G6261-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
G6358-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6556
G6358-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7195
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
G6425-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
G6425-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
G6425-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
G6425-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for GLUCOSE1PMETAB-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Degradation&#39;]
G6621-MONOMER has 2nd parents {&#39;Polymer-Degradation&#39;: 2} for PWY0-1546
G6621-MONOMER has 2nd parents {&#39;Other-Degradation&#39;: 2} for PWY0-1261
pruned to [&#39;Polymer-Degradation&#39;, &#39;Other-Degradation&#39;]
G6646-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-461
G6646-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY-6535
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;AMINE-DEG&#39;]
G6755-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-461
G6755-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PUTDEG-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;AMINE-DEG&#39;]
G6810-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
G6810-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
G6811-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-461
G6811-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY-6535
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;AMINE-DEG&#39;]
G6862-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for 1CMET2-PWY
G6862-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6614
G6862-MONOMER has 2nd parents {&#39;HORMONE-SYN&#39;: 2} for PWY0-1433
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;HORMONE-SYN&#39;]
G6903-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
G6903-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
pruned to [&#39;N/A&#39;, &#39;TCA-VARIANTS&#39;]
G6986-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for LEUSYN-PWY
G6986-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY0-1465
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;CARBOXYLATES-DEG&#39;]
G7164-MONOMER has 2nd parents {&#39;PWY-7206&#39;: 2, &#39;NUCLEO-DEG&#39;: 2} for PWY-7206
G7164-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
G7164-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
G7164-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7187
pruned to [&#39;PWY-7206&#39;, &#39;PWY-7184&#39;, &#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;]
G7187-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
G7187-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
G7187-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
G7212-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for PWY0-1337
G7212-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for FAO-PWY
pruned to [&#39;Fatty-Acid-and-Lipid-Degradation&#39;, &#39;Fatty-Acid-and-Lipid-Degradation&#39;]
G7213-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for PWY0-1337
G7213-MONOMER has 2nd parents {&#39;Fatty-Acid-and-Lipid-Degradation&#39;: 2} for FAO-PWY
pruned to [&#39;Fatty-Acid-and-Lipid-Degradation&#39;, &#39;Fatty-Acid-and-Lipid-Degradation&#39;]
G7285-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
G7285-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5480
G7285-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for PWY-7180
G7285-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY-5436
G7285-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for ETOH-ACETYLCOA-ANA-PWY
G7285-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-5162
G7285-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY0-1477
pruned to [&#39;N/A&#39;, &#39;Fermentation&#39;, &#39;Carbohydrates-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Alcohol-Degradation&#39;, &#39;CARBOXYLATES-DEG&#39;, &#39;AMINE-DEG&#39;]
G7288-MONOMER has 2nd parents {&#39;PWY-5485&#39;: 2} for PWY-5485
G7288-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY0-1312
G7288-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
G7288-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY0-1477
pruned to [&#39;N/A&#39;, &#39;Fermentation&#39;, &#39;N/A&#39;, &#39;AMINE-DEG&#39;]
G7322-MONOMER has 2nd parents {&#39;PEPTIDOGLYCANSYN-PWY&#39;: 2} for PEPTIDOGLYCANSYN-PWY
G7322-MONOMER has 2nd parents {&#39;Cell-Structure-Biosynthesis&#39;: 2} for PWY0-1586
pruned to [&#39;N/A&#39;, &#39;Cell-Structure-Biosynthesis&#39;]
G7325-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6892
G7325-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8164
G7325-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY0-1021
G7325-MONOMER has 2nd parents {&#39;Nucleic-Acid-Processing&#39;: 2} for PWY-7892
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Nucleic-Acid-Processing&#39;]
G7401-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
G7401-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7220
G7401-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
G7401-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-7870
G7401-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for GLUT-REDOX-PWY
G7401-MONOMER has 2nd parents {&#39;Arsenic-Detoxification&#39;: 2} for PWY-4621-1
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;, &#39;Arsenic-Detoxification&#39;]
G7463-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
G7463-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
G7463-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for UDPNAGSYN-PWY
G7463-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for PWY-5659
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
G7485-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
G7485-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-6608
pruned to [&#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
G7486-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
G7486-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-6608
pruned to [&#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
G7487-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
G7487-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-6608
pruned to [&#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
G7544-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLYCOLATEMET-PWY
G7544-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLYOXDEG-PWY
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
G7545-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLYCOLATEMET-PWY
G7545-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLYOXDEG-PWY
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
G7596-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-461
G7596-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PUTDEG-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;AMINE-DEG&#39;]
G7668-MONOMER has 2nd parents {&#39;PEPTIDOGLYCANSYN-PWY&#39;: 2} for PEPTIDOGLYCANSYN-PWY
G7668-MONOMER has 2nd parents {&#39;Cell-Structure-Biosynthesis&#39;: 2} for PWY0-1586
pruned to [&#39;N/A&#39;, &#39;Cell-Structure-Biosynthesis&#39;]
G7877-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5437
G7877-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ILEUSYN-PWY
G7877-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1535
G7877-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for LCYSDEG-PWY
G7877-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for TRYPDEG-PWY
G7877-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for SERDEG-PWY
pruned to [&#39;Fermentation&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
GABATRANSAM-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLUTORN-PWY
GABATRANSAM-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-461
GABATRANSAM-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY-6537
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;AMINE-DEG&#39;]
GAPDH-A-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
GAPDH-A-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
GAPDH-A-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
GARTRANSFORMYL2-MONOMER has 2nd parents {&#39;PWY-6277&#39;: 2} for PWY-6277
GARTRANSFORMYL2-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6122
GARTRANSFORMYL2-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY0-1477
pruned to [&#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;AMINE-DEG&#39;]
GCVH-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for 1CMET2-PWY
GCVH-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLYCLEAV-PWY
GCVH-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1608
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
GCVP-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for 1CMET2-PWY
GCVP-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLYCLEAV-PWY
GCVP-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1608
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
GCVT-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for 1CMET2-PWY
GCVT-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLYCLEAV-PWY
GCVT-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1608
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
GKI-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for PWY0-1300
GKI-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLYCOLATEMET-PWY
GKI-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GALACTARDEG-PWY
GKI-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLUCARDEG-PWY
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
GLU6PDEHYDROG-MONOMER has 2nd parents {&#39;Pentose-Phosphate-Cycle&#39;: 2} for OXIDATIVEPENT-PWY
GLU6PDEHYDROG-MONOMER has 2nd parents {&#39;GLYCOLYSIS-E-D&#39;: 2} for GLYCOLYSIS-E-D
pruned to [&#39;Pentose-Phosphate-Cycle&#39;, &#39;N/A&#39;]
GLUC1PURIDYLTRANS-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for PWY-7343
GLUC1PURIDYLTRANS-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GALACTMETAB-PWY
pruned to [&#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
GLUCOKIN-MONOMER has 2nd parents {&#39;GLYCOCAT-PWY&#39;: 2, &#39;Polymer-Degradation&#39;: 2} for GLYCOCAT-PWY
GLUCOKIN-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for PWY0-1182
GLUCOKIN-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for TREDEGLOW-PWY
GLUCOKIN-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for GLUCOSE1PMETAB-PWY
pruned to [&#39;GLYCOCAT-PWY&#39;, &#39;Carbohydrates-Degradation&#39;, &#39;Carbohydrates-Degradation&#39;, &#39;Carbohydrates-Degradation&#39;]
GLUCONOKINI-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for IDNCAT-PWY
GLUCONOKINI-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLUCONSUPER-PWY
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
GLURS-MONOMER has 2nd parents {&#39;TRNA-CHARGING-PWY&#39;: 2, &#39;Aminoacyl-tRNAs-Charging&#39;: 2} for TRNA-CHARGING-PWY
GLURS-MONOMER has 2nd parents {&#39;Tetrapyrrole-Biosynthesis&#39;: 2} for PWY-5188
pruned to [&#39;TRNA-CHARGING-PWY&#39;, &#39;Tetrapyrrole-Biosynthesis&#39;]
GLUSYNLARGE-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLUTSYN-PWY
GLUSYNLARGE-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
GLUSYNLARGE-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
GLUSYNSMALL-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLUTSYN-PWY
GLUSYNSMALL-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
GLUSYNSMALL-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
GLUTAMIDOTRANS-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for HISTSYN-PWY
GLUTAMIDOTRANS-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
GLUTAMIDOTRANS-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
GLUTAMINESYN-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLNSYN-PWY
GLUTAMINESYN-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY-8291
GLUTAMINESYN-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY-8294
GLUTAMINESYN-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
GLY3KIN-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for PWY0-1300
GLY3KIN-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLYCOLATEMET-PWY
GLY3KIN-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GALACTARDEG-PWY
GLY3KIN-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLUCARDEG-PWY
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
GLYC3PDEHYDROGBIOSYN-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5667
GLYC3PDEHYDROGBIOSYN-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1319
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
GLYCDEH-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for GLYCEROLMETAB-PWY
GLYCDEH-MONOMER has 2nd parents {&#39;Methylglyoxal-Detoxification&#39;: 2, &#39;Aldehyde-Degradation&#39;: 2} for PWY-5453
pruned to [&#39;Alcohol-Degradation&#39;, &#39;Methylglyoxal-Detoxification&#39;]
GLYCEROL-3-P-ACYLTRANSFER-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5667
GLYCEROL-3-P-ACYLTRANSFER-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1319
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
GLYCRIBONUCSYN-MONOMER has 2nd parents {&#39;PWY-6277&#39;: 2} for PWY-6277
GLYCRIBONUCSYN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6122
GLYCRIBONUCSYN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6121
pruned to [&#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
GLYOHMETRANS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for 1CMET2-PWY
GLYOHMETRANS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-2161
GLYOHMETRANS-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLYSYN-PWY
GLYOHMETRANS-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1608
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;]
GMP-SYN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7221
GMP-SYN-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
GMP-SYN-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
GPMA-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
GPMA-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;]
GPT-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6609
GPT-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6620
GPT-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for SALVPURINE2-PWY
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
GRXB-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-7870
GRXB-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for GLUT-REDOX-PWY
GRXB-MONOMER has 2nd parents {&#39;Arsenic-Detoxification&#39;: 2} for PWY-4621-1
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;, &#39;Arsenic-Detoxification&#39;]
GRXC-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-7870
GRXC-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for GLUT-REDOX-PWY
GRXC-MONOMER has 2nd parents {&#39;Arsenic-Detoxification&#39;: 2} for PWY-4621-1
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;, &#39;Arsenic-Detoxification&#39;]
GSK-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6611
GSK-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6618
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
HCAA2-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for PWY-6690
HCAA2-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for HCAMHPDEG-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;, &#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;]
HCAC-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for PWY-6690
HCAC-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for HCAMHPDEG-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;, &#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;]
HCAD-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for PWY-6690
HCAD-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for HCAMHPDEG-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;, &#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;]
HMP-P-KIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6910
HMP-P-KIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6890
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
HOLO-ACP-SYNTH-MONOMER has 2nd parents {&#39;ENTBACSYN-PWY&#39;: 2} for ENTBACSYN-PWY
HOLO-ACP-SYNTH-MONOMER has 2nd parents {&#39;Protein-Modification&#39;: 2} for PWY-6012
pruned to [&#39;N/A&#39;, &#39;Protein-Modification&#39;]
HOMOCYSMET-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6151
HOMOCYSMET-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for HOMOSER-METSYN-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
HOMOCYSMETB12-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for 1CMET2-PWY
HOMOCYSMETB12-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6151
HOMOCYSMETB12-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for HOMOSER-METSYN-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
HYBA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1577
HYBA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1578
HYBA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1576
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
HYBC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1577
HYBC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1578
HYBC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1576
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
HYCBSMALL-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
HYCBSMALL-MONOMER has 2nd parents {&#39;Hydrogen-Production&#39;: 2} for PWY-6772
pruned to [&#39;N/A&#39;, &#39;Hydrogen-Production&#39;]
HYCC-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
HYCC-MONOMER has 2nd parents {&#39;Hydrogen-Production&#39;: 2} for PWY-6772
pruned to [&#39;N/A&#39;, &#39;Hydrogen-Production&#39;]
HYCD-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
HYCD-MONOMER has 2nd parents {&#39;Hydrogen-Production&#39;: 2} for PWY-6772
pruned to [&#39;N/A&#39;, &#39;Hydrogen-Production&#39;]
HYCELARGE-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
HYCELARGE-MONOMER has 2nd parents {&#39;Hydrogen-Production&#39;: 2} for PWY-6772
HYCELARGE-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8319
pruned to [&#39;N/A&#39;, &#39;Hydrogen-Production&#39;, &#39;Cofactor-Biosynthesis&#39;]
HYCF-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
HYCF-MONOMER has 2nd parents {&#39;Hydrogen-Production&#39;: 2} for PWY-6772
pruned to [&#39;N/A&#39;, &#39;Hydrogen-Production&#39;]
HYCG-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
HYCG-MONOMER has 2nd parents {&#39;Hydrogen-Production&#39;: 2} for PWY-6772
pruned to [&#39;N/A&#39;, &#39;Hydrogen-Production&#39;]
HYPOXANPRIBOSYLTRAN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6609
HYPOXANPRIBOSYLTRAN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6620
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
ISOCITDEH-SUBUNIT has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
ISOCITDEH-SUBUNIT has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
pruned to [&#39;N/A&#39;, &#39;TCA-VARIANTS&#39;]
KDGALDOL-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GALACTARDEG-PWY
KDGALDOL-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLUCARDEG-PWY
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
KDPGALDOL-4OH2OXOGLUTARALDOL-MONOMER has 2nd parents {&#39;Entner-Duodoroff-Pathways&#39;: 2} for ENTNER-DOUDOROFF-PWY
KDPGALDOL-4OH2OXOGLUTARALDOL-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GALACTUROCAT-PWY
KDPGALDOL-4OH2OXOGLUTARALDOL-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-7242
pruned to [&#39;Entner-Duodoroff-Pathways&#39;, &#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
KETOBUTFORMLY-INACT-MONOMER has 2nd parents {&#39;PWY-5485&#39;: 2} for PWY-5485
KETOBUTFORMLY-INACT-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
KETOBUTFORMLY-INACT-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5480
KETOBUTFORMLY-INACT-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5437
pruned to [&#39;N/A&#39;, &#39;N/A&#39;, &#39;Fermentation&#39;, &#39;Fermentation&#39;]
KETOLREDUCTOISOM-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PANTO-PWY
KETOLREDUCTOISOM-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for VALSYN-PWY
KETOLREDUCTOISOM-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ILEUSYN-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
LACTALDDEHYDROG-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for DARABCATK12-PWY
LACTALDDEHYDROG-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY0-1280
LACTALDDEHYDROG-MONOMER has 2nd parents {&#39;Aldehyde-Degradation&#39;: 2} for PWY0-1317
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;Alcohol-Degradation&#39;, &#39;Aldehyde-Degradation&#39;]
LACTALDREDUCT-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for PWY0-1280
LACTALDREDUCT-MONOMER has 2nd parents {&#39;Aldehyde-Degradation&#39;: 2} for PWY0-1315
pruned to [&#39;Alcohol-Degradation&#39;, &#39;Aldehyde-Degradation&#39;]
LAUROYLACYLTRAN-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for KDO-LIPASYN-PWY
LAUROYLACYLTRAN-MONOMER has 2nd parents {&#39;KDO-NAGLIPASYN-PWY&#39;: 2} for KDO-NAGLIPASYN-PWY
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;N/A&#39;]
LDC2-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for PWY0-1601
LDC2-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for PWY0-1303
LDC2-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-461
pruned to [&#39;Polyamine-Biosynthesis&#39;, &#39;Polyamine-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;]
LSERINEDEAM1-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1608
LSERINEDEAM1-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for SERDEG-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
LSERINEDEAM2-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1608
LSERINEDEAM2-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for SERDEG-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
LSERINEDEAM3-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1608
LSERINEDEAM3-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for SERDEG-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
LYSDECARBOX-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for PWY0-1601
LYSDECARBOX-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for PWY0-1303
LYSDECARBOX-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-461
pruned to [&#39;Polyamine-Biosynthesis&#39;, &#39;Polyamine-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;]
LYXK-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for LYXMET-PWY
LYXK-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-6961
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;CARBOXYLATES-DEG&#39;]
MALATE-DEHASE-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
MALATE-DEHASE-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
MALATE-DEHASE-MONOMER has 2nd parents {&#39;GLYOXYLATE-BYPASS&#39;: 2} for GLYOXYLATE-BYPASS
MALATE-DEHASE-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;N/A&#39;, &#39;TCA-VARIANTS&#39;, &#39;GLYOXYLATE-BYPASS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
MALIC-NAD-MONOMER has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
MALIC-NAD-MONOMER has 2nd parents {&#39;GLYOXYLATE-BYPASS&#39;: 2} for GLYOXYLATE-BYPASS
MALIC-NAD-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;TCA-VARIANTS&#39;, &#39;GLYOXYLATE-BYPASS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
MANNPISOM-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for PWY-5659
MANNPISOM-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for MANNCAT-PWY
pruned to [&#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Degradation&#39;]
MANX-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
MANX-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for MANNCAT-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Degradation&#39;]
MANY-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
MANY-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for MANNCAT-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Degradation&#39;]
MANZ-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
MANZ-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for MANNCAT-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Degradation&#39;]
METHYLMALONYL-COA-MUT-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY0-43
METHYLMALONYL-COA-MUT-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PROPIONMET-PWY
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
MHPCHYDROL-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for PWY-6690
MHPCHYDROL-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for HCAMHPDEG-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;, &#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;]
MHPF-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5480
MHPF-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for PWY-7180
MHPF-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY-5436
MHPF-MONOMER has 2nd parents {&#39;Alcohol-Degradation&#39;: 2} for ETOH-ACETYLCOA-ANA-PWY
MHPF-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-5162
MHPF-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY0-1477
pruned to [&#39;Fermentation&#39;, &#39;Carbohydrates-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Alcohol-Degradation&#39;, &#39;CARBOXYLATES-DEG&#39;, &#39;AMINE-DEG&#39;]
MHPHYDROXY-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for PWY-6690
MHPHYDROXY-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for HCAMHPDEG-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;, &#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;]
MONOMER0-145 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1577
MONOMER0-145 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1578
MONOMER0-145 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1576
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
MONOMER0-148 has 2nd parents {&#39;KETOGLUCONMET-PWY&#39;: 2} for KETOGLUCONMET-PWY
MONOMER0-148 has 2nd parents {&#39;Methylglyoxal-Detoxification&#39;: 2, &#39;Aldehyde-Degradation&#39;: 2} for PWY-5453
pruned to [&#39;N/A&#39;, &#39;Methylglyoxal-Detoxification&#39;]
MONOMER0-149 has 2nd parents {&#39;KETOGLUCONMET-PWY&#39;: 2} for KETOGLUCONMET-PWY
MONOMER0-149 has 2nd parents {&#39;Methylglyoxal-Detoxification&#39;: 2, &#39;Aldehyde-Degradation&#39;: 2} for PWY-5453
pruned to [&#39;N/A&#39;, &#39;Methylglyoxal-Detoxification&#39;]
MONOMER0-2663 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1353
MONOMER0-2663 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
MONOMER0-2663 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1568
MONOMER0-2663 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7545
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
MONOMER0-2822 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1353
MONOMER0-2822 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
MONOMER0-2822 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1568
MONOMER0-2822 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7545
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
MONOMER0-4521 has 2nd parents {&#39;PEPTIDOGLYCANSYN-PWY&#39;: 2} for PEPTIDOGLYCANSYN-PWY
MONOMER0-4521 has 2nd parents {&#39;Cell-Structure-Biosynthesis&#39;: 2} for PWY0-1586
pruned to [&#39;N/A&#39;, &#39;Cell-Structure-Biosynthesis&#39;]
MONOMER0-561 has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLYCOLATEMET-PWY
MONOMER0-561 has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLYOXDEG-PWY
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
MYRISTOYLACYLTRAN-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for KDO-LIPASYN-PWY
MYRISTOYLACYLTRAN-MONOMER has 2nd parents {&#39;KDO-NAGLIPASYN-PWY&#39;: 2} for KDO-NAGLIPASYN-PWY
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;N/A&#39;]
NAD-SYNTH-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PYRIDNUCSAL-PWY
NAD-SYNTH-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PYRIDNUCSYN-PWY
NAD-SYNTH-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-7761
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
NADH-DHII-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1567
NADH-DHII-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1568
NADH-DHII-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1573
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NAG1P-URIDYLTRANS-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for UDPNAGSYN-PWY
NAG1P-URIDYLTRANS-MONOMER has 2nd parents {&#39;Other-Degradation&#39;: 2} for PWY0-1261
pruned to [&#39;Carbohydrates-Biosynthesis&#39;, &#39;Other-Degradation&#39;]
NAG6PDEACET-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for GLUAMCAT-PWY
NAG6PDEACET-MONOMER has 2nd parents {&#39;Other-Degradation&#39;: 2} for PWY0-1261
pruned to [&#39;AMINE-DEG&#39;, &#39;Other-Degradation&#39;]
NARG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
NARG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1573
NARG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1581
NARG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NARH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
NARH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1573
NARH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1581
NARH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NARI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
NARI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1573
NARI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1581
NARI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NARL-MONOMER has 2nd parents {&#39;PWY0-1515&#39;: 2} for PWY0-1515
NARL-MONOMER has 2nd parents {&#39;PWY0-1514&#39;: 2} for PWY0-1514
pruned to [&#39;PWY0-1515&#39;, &#39;PWY0-1514&#39;]
NARP-MONOMER has 2nd parents {&#39;PWY0-1515&#39;: 2} for PWY0-1515
NARP-MONOMER has 2nd parents {&#39;PWY0-1514&#39;: 2} for PWY0-1514
pruned to [&#39;PWY0-1515&#39;, &#39;PWY0-1514&#39;]
NARV-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
NARV-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1581
NARV-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NARY-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
NARY-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1581
NARY-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NARZ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
NARZ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1581
NARZ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1321
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NICONUCADENYLYLTRAN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PYRIDNUCSAL-PWY
NICONUCADENYLYLTRAN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PYRIDNUCSYN-PWY
NICONUCADENYLYLTRAN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-7761
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
NRDA-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
NRDA-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
NRDA-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7220
NRDA-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
pruned to [&#39;PWY-7184&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;]
NRDB-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
NRDB-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
NRDB-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7220
NRDB-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
pruned to [&#39;PWY-7184&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;]
NRDE-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
NRDE-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
NRDE-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7220
NRDE-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
pruned to [&#39;PWY-7184&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;]
NRDF-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
NRDF-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
NRDF-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7220
NRDF-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
pruned to [&#39;PWY-7184&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;]
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;PWY-7197&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7197
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7220
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7187
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7221
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-163
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7205
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7176
NUCLEOSIDE-DIP-KIN-MONOMER has 2nd parents {&#39;Metabolic-Regulators&#39;: 2} for PPGPPMET-PWY
pruned to [&#39;PWY-7184&#39;, &#39;PWY-7197&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Metabolic-Regulators&#39;]
NUOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOB-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOC-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOE-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOE-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOE-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOE-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOE-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOE-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOE-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOF-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOF-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOF-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOF-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOF-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOF-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOF-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOG-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOH-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOI-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOJ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOJ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOJ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOJ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOJ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOJ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOJ-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOK-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOK-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOK-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOK-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOK-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOK-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOK-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOL-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOL-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOL-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOL-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOL-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOL-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOL-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUOM-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUOM-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUOM-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUOM-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUOM-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUOM-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUOM-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
NUON-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1335
NUON-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1590
NUON-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1334
NUON-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1348
NUON-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1336
NUON-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
NUON-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1352
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
PABASYN-COMPI-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-BIOSYN&#39;: 2} for PWY-6543
PABASYN-COMPI-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLUTSYN-PWY
PABASYN-COMPI-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
PABASYN-COMPI-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-BIOSYN&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
PABASYN-COMPII-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-BIOSYN&#39;: 2} for PWY-6543
PABASYN-COMPII-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for GLUTSYN-PWY
PABASYN-COMPII-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for GLUTAMINDEG-PWY
PABASYN-COMPII-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for AMMASSIM-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-BIOSYN&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Noncarbon-Nutrients&#39;]
PDXH-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PYRIDOXSYN-PWY
PDXH-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PLPSAL-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
PDXK-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6910
PDXK-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PLPSAL-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
PEPSYNTH-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
PEPSYNTH-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
PEPSYNTH-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
PGK has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
PGK has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
PGK has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
PGLUCISOM has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
PGLUCISOM has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
PGLUCISOM has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for UDPNAGSYN-PWY
PGLUCISOM has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for PWY-5659
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
PGMI-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
PGMI-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
PGMI-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
PGPPHOSPHAA-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1545
PGPPHOSPHAA-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5668
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
PGPPHOSPHAB-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-5785
PGPPHOSPHAB-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1545
PGPPHOSPHAB-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5668
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
PHENPRODIOLDEHYDROG-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for PWY-6690
PHENPRODIOLDEHYDROG-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for HCAMHPDEG-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;, &#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;]
PHENYLPRODIOXY-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for PWY-6690
PHENYLPRODIOXY-MONOMER has 2nd parents {&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;: 2} for HCAMHPDEG-PWY
pruned to [&#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;, &#39;AROMATIC-COMPOUNDS-DEGRADATION&#39;]
PHOSACETYLTRANS-MONOMER has 2nd parents {&#39;PWY-5485&#39;: 2} for PWY-5485
PHOSACETYLTRANS-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY0-1312
PHOSACETYLTRANS-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
PHOSACETYLTRANS-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5437
PHOSACETYLTRANS-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY0-1477
pruned to [&#39;N/A&#39;, &#39;Fermentation&#39;, &#39;N/A&#39;, &#39;Fermentation&#39;, &#39;AMINE-DEG&#39;]
PHOSGLUCOSAMINEMUT-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for UDPNAGSYN-PWY
PHOSGLUCOSAMINEMUT-MONOMER has 2nd parents {&#39;Other-Degradation&#39;: 2} for PWY0-1261
pruned to [&#39;Carbohydrates-Biosynthesis&#39;, &#39;Other-Degradation&#39;]
PHOSPHAGLYPSYN-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY0-1545
PHOSPHAGLYPSYN-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5668
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
PHOSPHOGLUCMUT-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for PWY-7343
PHOSPHOGLUCMUT-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for DTDPRHAMSYN-PWY
PHOSPHOGLUCMUT-MONOMER has 2nd parents {&#39;Polysaccharides-Biosynthesis&#39;: 2, &#39;Carbohydrates-Biosynthesis&#39;: 2} for GLYCOGENSYNTH-PWY
PHOSPHOGLUCMUT-MONOMER has 2nd parents {&#39;GLYCOCAT-PWY&#39;: 2, &#39;Polymer-Degradation&#39;: 2} for GLYCOCAT-PWY
PHOSPHOGLUCMUT-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for GLUCOSE1PMETAB-PWY
pruned to [&#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;Polysaccharides-Biosynthesis&#39;, &#39;GLYCOCAT-PWY&#39;, &#39;Carbohydrates-Degradation&#39;]
PKI-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
PKI-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
PKI-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
pruned to [&#39;N/A&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;]
PKII-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
PKII-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
PKII-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
pruned to [&#39;N/A&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;]
PPENTOMUT-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for PWY-7180
PPENTOMUT-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1296
PPENTOMUT-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1295
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
PROTEIN-CHEA has 2nd parents {&#39;PWY0-1518&#39;: 2} for PWY0-1518
PROTEIN-CHEA has 2nd parents {&#39;PWY0-1519&#39;: 2} for PWY0-1519
pruned to [&#39;PWY0-1518&#39;, &#39;PWY0-1519&#39;]
PROTOHEME-FERROCHELAT-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for HEMESYN2-PWY
PROTOHEME-FERROCHELAT-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for HEME-BIOSYNTHESIS-II-1
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
PROTOPORGENOXI-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for HEMESYN2-PWY
PROTOPORGENOXI-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for HEME-BIOSYNTHESIS-II-1
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
PRPPAMIDOTRANS-MONOMER has 2nd parents {&#39;PWY-6277&#39;: 2} for PWY-6277
PRPPAMIDOTRANS-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6122
PRPPAMIDOTRANS-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6121
pruned to [&#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
PSERTRANSAM-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PYRIDOXSYN-PWY
PSERTRANSAM-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for DAPLYSINESYN-PWY
PSERTRANSAM-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for SERSYN-PWY-1
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
PUTA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1544
PUTA-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PROUT-PWY-I
pruned to [&#39;Electron-Transfer&#39;, &#39;Amino-Acid-Degradation&#39;]
PYRUVFORMLY-MONOMER has 2nd parents {&#39;PWY-5485&#39;: 2} for PWY-5485
PYRUVFORMLY-MONOMER has 2nd parents {&#39;FERMENTATION-PWY&#39;: 2} for FERMENTATION-PWY
PYRUVFORMLY-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5480
PYRUVFORMLY-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5437
pruned to [&#39;N/A&#39;, &#39;N/A&#39;, &#39;Fermentation&#39;, &#39;Fermentation&#39;]
PYRUVOXID-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7544
PYRUVOXID-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY-7545
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
RED-GLUTAREDOXIN has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-7870
RED-GLUTAREDOXIN has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for GLUT-REDOX-PWY
RED-GLUTAREDOXIN has 2nd parents {&#39;Arsenic-Detoxification&#39;: 2} for PWY-4621-1
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Other-biosynthesis&#39;, &#39;Arsenic-Detoxification&#39;]
RED-THIOREDOXIN-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
RED-THIOREDOXIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8147
RED-THIOREDOXIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
RED-THIOREDOXIN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7220
RED-THIOREDOXIN-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
RED-THIOREDOXIN-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for THIOREDOX-PWY
RED-THIOREDOXIN-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-5329
RED-THIOREDOXIN-MONOMER has 2nd parents {&#39;SO4ASSIM-PWY&#39;: 2} for SO4ASSIM-PWY
RED-THIOREDOXIN-MONOMER has 2nd parents {&#39;Protein-Modification&#39;: 2} for PWY0-1600
pruned to [&#39;PWY-7184&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Other-biosynthesis&#39;, &#39;Noncarbon-Nutrients&#39;, &#39;N/A&#39;, &#39;Protein-Modification&#39;]
RED-THIOREDOXIN2-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
RED-THIOREDOXIN2-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8147
RED-THIOREDOXIN2-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
RED-THIOREDOXIN2-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7220
RED-THIOREDOXIN2-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
RED-THIOREDOXIN2-MONOMER has 2nd parents {&#39;Other-biosynthesis&#39;: 2} for THIOREDOX-PWY
RED-THIOREDOXIN2-MONOMER has 2nd parents {&#39;Noncarbon-Nutrients&#39;: 2} for PWY-5329
RED-THIOREDOXIN2-MONOMER has 2nd parents {&#39;SO4ASSIM-PWY&#39;: 2} for SO4ASSIM-PWY
pruned to [&#39;PWY-7184&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Other-biosynthesis&#39;, &#39;Noncarbon-Nutrients&#39;, &#39;N/A&#39;]
RHAMNISOM-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for RHAMCAT-PWY
RHAMNISOM-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for LYXMET-PWY
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;Carbohydrates-Degradation&#39;]
RIB5PISOMB-MONOMER has 2nd parents {&#39;Pentose-Phosphate-Cycle&#39;: 2} for NONOXIPENT-PWY
RIB5PISOMB-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for PWY0-44
pruned to [&#39;Pentose-Phosphate-Cycle&#39;, &#39;Carbohydrates-Degradation&#39;]
RIBAZOLEPHOSPHAT-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-5509
RIBAZOLEPHOSPHAT-MONOMER has 2nd parents {&#39;COBALSYN-PWY&#39;: 2} for COBALSYN-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;N/A&#39;]
RIBONUCLEOSIDE-TRIP-REDUCT-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7222
RIBONUCLEOSIDE-TRIP-REDUCT-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7220
RIBONUCLEOSIDE-TRIP-REDUCT-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
RIBONUCLEOSIDE-TRIP-REDUCT-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7187
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;]
RIBULPEPIM-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for ARABCAT-PWY
RIBULPEPIM-MONOMER has 2nd parents {&#39;Carbohydrates-Degradation&#39;: 2} for LYXMET-PWY
pruned to [&#39;Carbohydrates-Degradation&#39;, &#39;Carbohydrates-Degradation&#39;]
S-ADENMETSYN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6151
S-ADENMETSYN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for SAM-PWY
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
SDH-FE-S has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
SDH-FE-S has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1329
SDH-FE-S has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1353
pruned to [&#39;TCA-VARIANTS&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
SDH-FLAVO has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
SDH-FLAVO has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1329
SDH-FLAVO has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1353
pruned to [&#39;TCA-VARIANTS&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
SDH-MEMB1 has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
SDH-MEMB1 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1329
SDH-MEMB1 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1353
pruned to [&#39;TCA-VARIANTS&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
SDH-MEMB2 has 2nd parents {&#39;TCA-VARIANTS&#39;: 2} for TCA
SDH-MEMB2 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1329
SDH-MEMB2 has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1353
pruned to [&#39;TCA-VARIANTS&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
SERINE-O-ACETTRAN-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY-7870
SERINE-O-ACETTRAN-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for CYSTSYN-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
SERS-MONOMER has 2nd parents {&#39;TRNA-CHARGING-PWY&#39;: 2, &#39;Aminoacyl-tRNAs-Charging&#39;: 2} for TRNA-CHARGING-PWY
SERS-MONOMER has 2nd parents {&#39;ENTBACSYN-PWY&#39;: 2} for ENTBACSYN-PWY
SERS-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY0-901
pruned to [&#39;TRNA-CHARGING-PWY&#39;, &#39;N/A&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
SMALLILVM-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for VALSYN-PWY
SMALLILVM-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for ILEUSYN-PWY
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
SPED-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for BSUBPOLYAMSYN-PWY
SPED-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for PWY0-1303
pruned to [&#39;Polyamine-Biosynthesis&#39;, &#39;Polyamine-Biosynthesis&#39;]
SPERMIDINESYN-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for BSUBPOLYAMSYN-PWY
SPERMIDINESYN-MONOMER has 2nd parents {&#39;Polyamine-Biosynthesis&#39;: 2} for PWY0-1303
pruned to [&#39;Polyamine-Biosynthesis&#39;, &#39;Polyamine-Biosynthesis&#39;]
SUCCSEMIALDDEHYDROG-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-461
SUCCSEMIALDDEHYDROG-MONOMER has 2nd parents {&#39;AMINE-DEG&#39;: 2} for PWY-6537
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;AMINE-DEG&#39;]
THI-P-KIN-MONOMER has 2nd parents {&#39;PWY-6897&#39;: 2} for PWY-6897
THI-P-KIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8457
THI-P-KIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6894
pruned to [&#39;N/A&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
THIE-MONOMER has 2nd parents {&#39;PWY-6897&#39;: 2} for PWY-6897
THIE-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8457
THIE-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6894
pruned to [&#39;N/A&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
THIOESTERI-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-6282
THIOESTERI-MONOMER has 2nd parents {&#39;Lipid-Biosynthesis&#39;: 2} for PWY-5973
pruned to [&#39;Lipid-Biosynthesis&#39;, &#39;Lipid-Biosynthesis&#39;]
THIS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-6892
THIS-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-1507
THIS-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PWY0-1021
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
THREDEHYDCAT-MONOMER has 2nd parents {&#39;Fermentation&#39;: 2} for PWY-5437
THREDEHYDCAT-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for PWY0-1608
THREDEHYDCAT-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for SERDEG-PWY
pruned to [&#39;Fermentation&#39;, &#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
THREODEHYD-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for THREONINE-DEG2-PWY
THREODEHYD-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for THRDLCTCAT-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
THRS-MONOMER has 2nd parents {&#39;TRNA-CHARGING-PWY&#39;: 2, &#39;Aminoacyl-tRNAs-Charging&#39;: 2} for TRNA-CHARGING-PWY
THRS-MONOMER has 2nd parents {&#39;ENTBACSYN-PWY&#39;: 2} for ENTBACSYN-PWY
pruned to [&#39;TRNA-CHARGING-PWY&#39;, &#39;N/A&#39;]
THYMIDYLATESYN-MONOMER has 2nd parents {&#39;PWY-7184&#39;: 2, &#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7184
THYMIDYLATESYN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for 1CMET2-PWY
THYMIDYLATESYN-MONOMER has 2nd parents {&#39;PWY0-166&#39;: 2} for PWY0-166
THYMIDYLATESYN-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7187
pruned to [&#39;PWY-7184&#39;, &#39;Cofactor-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Nucleotide-Biosynthesis&#39;]
THZ-KIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY0-1597
THZ-KIN-MONOMER has 2nd parents {&#39;PWY-6897&#39;: 2} for PWY-6897
THZ-KIN-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for PWY-8457
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;N/A&#39;, &#39;Cofactor-Biosynthesis&#39;]
TORA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1578
TORA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1355
TORA-MONOMER has 2nd parents {&#39;Electron-Transfer&#39;: 2} for PWY0-1347
pruned to [&#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;, &#39;Electron-Transfer&#39;]
TPI-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for GLYCOLYSIS
TPI-MONOMER has 2nd parents {&#39;GLYCOLYSIS-VARIANTS&#39;: 2} for PWY-5484
TPI-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GLUCONEO-PWY
pruned to [&#39;GLYCOLYSIS-VARIANTS&#39;, &#39;GLYCOLYSIS-VARIANTS&#39;, &#39;Carbohydrates-Biosynthesis&#39;]
TRYPTOPHAN-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for LCYSDEG-PWY
TRYPTOPHAN-MONOMER has 2nd parents {&#39;Amino-Acid-Degradation&#39;: 2} for TRYPDEG-PWY
pruned to [&#39;Amino-Acid-Degradation&#39;, &#39;Amino-Acid-Degradation&#39;]
TSA-REDUCT-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLYCOLATEMET-PWY
TSA-REDUCT-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GALACTARDEG-PWY
TSA-REDUCT-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GLUCARDEG-PWY
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
TYRB-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for PHESYN
TYRB-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for LEUSYN-PWY
TYRB-MONOMER has 2nd parents {&#39;Amino-Acid-Biosynthesis&#39;: 2} for TYRSYN
pruned to [&#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;, &#39;Amino-Acid-Biosynthesis&#39;]
UDK-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7193
UDK-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-163
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
UDPGLUCEPIM-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for PWY-7344
UDPGLUCEPIM-MONOMER has 2nd parents {&#39;Carbohydrates-Biosynthesis&#39;: 2} for GALACTMETAB-PWY
UDPGLUCEPIM-MONOMER has 2nd parents {&#39;COLANSYN-PWY&#39;: 2} for COLANSYN-PWY
pruned to [&#39;Carbohydrates-Biosynthesis&#39;, &#39;Carbohydrates-Biosynthesis&#39;, &#39;N/A&#39;]
UDPNACETYLGLUCOSAMENOLPYRTRANS-MONOMER has 2nd parents {&#39;Cell-Structure-Biosynthesis&#39;: 2} for PWY-6387
UDPNACETYLGLUCOSAMENOLPYRTRANS-MONOMER has 2nd parents {&#39;Other-Degradation&#39;: 2} for PWY0-1261
pruned to [&#39;Cell-Structure-Biosynthesis&#39;, &#39;Other-Degradation&#39;]
UDPNACETYLMURAMATEDEHYDROG-MONOMER has 2nd parents {&#39;Cell-Structure-Biosynthesis&#39;: 2} for PWY-6387
UDPNACETYLMURAMATEDEHYDROG-MONOMER has 2nd parents {&#39;Other-Degradation&#39;: 2} for PWY0-1261
pruned to [&#39;Cell-Structure-Biosynthesis&#39;, &#39;Other-Degradation&#39;]
URACIL-PRIBOSYLTRANS-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-7183
URACIL-PRIBOSYLTRANS-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-163
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;]
UROGENDECARBOX-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for HEMESYN2-PWY
UROGENDECARBOX-MONOMER has 2nd parents {&#39;Cofactor-Biosynthesis&#39;: 2} for HEME-BIOSYNTHESIS-II-1
pruned to [&#39;Cofactor-Biosynthesis&#39;, &#39;Cofactor-Biosynthesis&#39;]
URPHOS-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY0-163
URPHOS-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1295
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;NUCLEO-DEG&#39;]
UXAC-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for GALACTUROCAT-PWY
UXAC-MONOMER has 2nd parents {&#39;CARBOXYLATES-DEG&#39;: 2} for PWY-7247
pruned to [&#39;CARBOXYLATES-DEG&#39;, &#39;CARBOXYLATES-DEG&#39;]
XANTHOSINEPHOSPHORY-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6609
XANTHOSINEPHOSPHORY-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for PWY-6620
XANTHOSINEPHOSPHORY-MONOMER has 2nd parents {&#39;Nucleotide-Biosynthesis&#39;: 2} for SALVPURINE2-PWY
XANTHOSINEPHOSPHORY-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for SALVADEHYPOX-PWY
XANTHOSINEPHOSPHORY-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY-6608
XANTHOSINEPHOSPHORY-MONOMER has 2nd parents {&#39;NUCLEO-DEG&#39;: 2} for PWY0-1296
pruned to [&#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;Nucleotide-Biosynthesis&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;, &#39;NUCLEO-DEG&#39;]
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># create pathway to 2nd layer pathway mapping</span>
<span class="n">W2</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">((</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">index</span><span class="p">),</span> <span class="nb">len</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">index</span><span class="p">)))</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>

    <span class="n">cur_pathway</span> <span class="o">=</span> <span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;id&#39;</span><span class="p">]</span>
    <span class="n">cur_pathway_idx</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">cur_pathway</span><span class="p">)</span>

    <span class="c1"># get 2nd level parents</span>
    <span class="n">cur_pathway_parents</span> <span class="o">=</span> <span class="n">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">cur_pathway_idx</span><span class="p">,</span> <span class="n">pathway_matrix</span><span class="p">,</span> <span class="n">pathway_ids</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>

    <span class="c1"># if len(cur_pathway_parents) &gt; 1:</span>
    <span class="c1">#     print(f&quot;cur pathway {cur_pathway} has parents {cur_pathway_parents}&quot;)</span>

    <span class="k">for</span> <span class="n">parent</span> <span class="ow">in</span> <span class="n">cur_pathway_parents</span><span class="p">:</span>
        <span class="n">parent_idx</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">parent</span><span class="p">)</span>
        <span class="n">W2</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">parent_idx</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>

<span class="c1"># zero diagonal (don&#39;t return self, since some pathways return themselves as level 2 parents)</span>
<span class="n">np</span><span class="o">.</span><span class="n">fill_diagonal</span><span class="p">(</span><span class="n">W2</span><span class="p">[</span><span class="mi">0</span><span class="p">:(</span><span class="n">W2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">-</span> <span class="nb">len</span><span class="p">(</span><span class="n">go_collection</span><span class="p">)),</span> <span class="mi">0</span><span class="p">:(</span><span class="n">W2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">-</span> <span class="nb">len</span><span class="p">(</span><span class="n">go_collection</span><span class="p">))],</span> <span class="mi">0</span><span class="p">)</span>


<span class="n">W1</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">((</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">index</span><span class="p">),</span> <span class="nb">len</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">index</span><span class="p">)))</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>

    <span class="k">if</span> <span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;level&#39;</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="mi">2</span><span class="p">:</span>
        <span class="n">cur_pathway</span> <span class="o">=</span> <span class="n">parsed_pathway_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;id&#39;</span><span class="p">]</span>
        <span class="n">cur_pathway_idx</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">cur_pathway</span><span class="p">)</span>

        <span class="c1"># get 2nd level parents</span>
        <span class="n">cur_pathway_parents</span> <span class="o">=</span> <span class="n">get_pathway_ith_level_parents</span><span class="p">(</span><span class="n">cur_pathway_idx</span><span class="p">,</span> <span class="n">pathway_matrix</span><span class="p">,</span> <span class="n">pathway_ids</span><span class="p">,</span> <span class="n">level_vector</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

        <span class="k">for</span> <span class="n">parent</span> <span class="ow">in</span> <span class="n">cur_pathway_parents</span><span class="p">:</span>
            <span class="n">parent_idx</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">parent</span><span class="p">)</span>
            <span class="n">W1</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">parent_idx</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>

<span class="n">np</span><span class="o">.</span><span class="n">fill_diagonal</span><span class="p">(</span><span class="n">W1</span><span class="p">,</span> <span class="mi">0</span><span class="p">)</span>
<br/><br/></pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">W</span><span class="p">[</span><span class="n">protein_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;EG10230-MONOMER&#39;</span><span class="p">),</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;Regulation of transcription&#39;</span><span class="p">)]</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
0.0
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># C matrix transforms complexes + monomers to just monomers.</span>

<span class="c1"># create protein name to index mapping</span>
<span class="n">protein_name_to_index</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>
    <span class="n">protein_name_to_index</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;id&#39;</span><span class="p">]]</span> <span class="o">=</span> <span class="n">i</span>

<span class="c1"># C matrix: complexes x proteins</span>
<span class="n">C</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">((</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_complex_df</span><span class="o">.</span><span class="n">index</span><span class="p">),</span> <span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)))</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_complex_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>

    <span class="n">complex_components</span> <span class="o">=</span> <span class="n">parsed_complex_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;monomer_component_stoichiometry&#39;</span><span class="p">]</span>

    <span class="c1"># TODO consider cofactors</span>
    <span class="c1"># complex_cofactors = filter_complex_df.loc[i, &#39;cofactors&#39;]</span>


    <span class="k">for</span> <span class="n">component_name</span><span class="p">,</span> <span class="n">component_count</span> <span class="ow">in</span> <span class="n">complex_components</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">component_count</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>             <span class="c1"># side effect of parquet</span>
            <span class="c1"># get index of component in filter_protein_df</span>
            <span class="n">component_index</span> <span class="o">=</span> <span class="n">protein_name_to_index</span><span class="p">[</span><span class="n">component_name</span><span class="p">]</span>

            <span class="k">if</span> <span class="n">parsed_complex_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;APORNAP-CPLX&#39;</span><span class="p">:</span>
                <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;component_name: </span><span class="si">{</span><span class="n">component_name</span><span class="si">}</span><span class="s1">, component_count: </span><span class="si">{</span><span class="n">component_count</span><span class="si">}</span><span class="s1">, component_index: </span><span class="si">{</span><span class="n">component_index</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>

            <span class="n">C</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">component_index</span><span class="p">]</span> <span class="o">=</span> <span class="n">component_count</span>

<span class="c1"># append an identity matrix to C</span>
<span class="n">C</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">concatenate</span><span class="p">((</span><span class="n">C</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">identity</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">))),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="n">C_names</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_complex_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">])</span> <span class="o">+</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
component_name: EG10893-MONOMER, component_count: 2, component_index: 697
component_name: RPOC-MONOMER, component_count: 1, component_index: 4122
component_name: RPOB-MONOMER, component_count: 1, component_index: 4121
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># P matrix transforms proteins to their respective cofactor counts.</span>

<span class="c1"># create cofactor name to index mapping</span>
<span class="n">cofactor_name_to_index</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_cofactor_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>
    <span class="n">cofactor_name_to_index</span><span class="p">[</span><span class="n">parsed_cofactor_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;id&#39;</span><span class="p">]]</span> <span class="o">=</span> <span class="n">i</span>

<span class="n">cofactor_ids</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_cofactor_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">])</span>

<span class="c1"># P matrix: proteins x cofactors</span>
<span class="n">P</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">((</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">),</span> <span class="nb">len</span><span class="p">(</span><span class="n">parsed_cofactor_df</span><span class="o">.</span><span class="n">index</span><span class="p">)))</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>
    <span class="n">protein_metals</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;metal_features_processed&#39;</span><span class="p">]</span>
    <span class="n">protein_other</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;other_features_processed&#39;</span><span class="p">]</span>

    <span class="k">for</span> <span class="n">metal</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">protein_metals</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">count</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>             <span class="c1"># side effect of parquet</span>
            <span class="n">cofactor_index</span> <span class="o">=</span> <span class="n">cofactor_name_to_index</span><span class="p">[</span><span class="n">metal</span><span class="p">]</span>
            <span class="n">P</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">cofactor_index</span><span class="p">]</span> <span class="o">=</span> <span class="n">count</span>

    <span class="k">for</span> <span class="n">other</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">protein_other</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">count</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>             <span class="c1"># side effect of parquet</span>
            <span class="n">cofactor_index</span> <span class="o">=</span> <span class="n">cofactor_name_to_index</span><span class="p">[</span><span class="n">other</span><span class="p">]</span>
            <span class="n">P</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">cofactor_index</span><span class="p">]</span> <span class="o">=</span> <span class="n">count</span>
<br/></pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># E matrix transforms cofactors to their respective elemental composition</span>

<span class="c1"># create list of unique elements</span>
<span class="n">unique_elements</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_cofactor_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>
    <span class="n">cofactor</span> <span class="o">=</span> <span class="n">parsed_cofactor_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;elemental_composition&#39;</span><span class="p">]</span>
    <span class="n">unique_elements</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">cofactor</span><span class="o">.</span><span class="n">keys</span><span class="p">())</span>

<span class="n">unique_elements</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">unique_elements</span><span class="p">)</span>

<span class="c1"># create E matrix: cofactors x elements</span>
<span class="n">E</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">((</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_cofactor_df</span><span class="o">.</span><span class="n">index</span><span class="p">),</span> <span class="nb">len</span><span class="p">(</span><span class="n">unique_elements</span><span class="p">)))</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_cofactor_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>
    <span class="n">cofactor</span> <span class="o">=</span> <span class="n">parsed_cofactor_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;elemental_composition&#39;</span><span class="p">]</span>

    <span class="k">for</span> <span class="n">element</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">cofactor</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">count</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>             <span class="c1"># side effect of parquet</span>
            <span class="n">element_index</span> <span class="o">=</span> <span class="n">unique_elements</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">element</span><span class="p">)</span>
            <span class="n">E</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">element_index</span><span class="p">]</span> <span class="o">=</span> <span class="n">count</span>


<span class="n">element_ids</span> <span class="o">=</span> <span class="n">unique_elements</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># A matrix transforms proteins to their amino acid composition</span>

<span class="n">A</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">((</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">),</span> <span class="nb">len</span><span class="p">(</span><span class="n">unique_aa</span><span class="p">)))</span>

<span class="n">amino_acid_single_letter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">unique_aa</span><span class="p">)</span>
<span class="n">amino_acid_ids</span> <span class="o">=</span> <span class="p">[</span><span class="n">AMINO_ACID_MAP</span><span class="p">[</span><span class="n">aa</span><span class="p">]</span> <span class="k">for</span> <span class="n">aa</span> <span class="ow">in</span> <span class="n">amino_acid_single_letter</span><span class="p">]</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">index</span><span class="p">)):</span>
    <span class="n">protein</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;sequence_processed&#39;</span><span class="p">]</span>

    <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">protein</span><span class="p">)</span> <span class="ow">is</span> <span class="ow">not</span> <span class="nb">dict</span><span class="p">:</span>
        <span class="k">continue</span>

    <span class="k">for</span> <span class="n">aa</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">protein</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">count</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>             <span class="c1"># side effect of parquet</span>
            <span class="n">aa_index</span> <span class="o">=</span> <span class="n">amino_acid_single_letter</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">aa</span><span class="p">)</span>
            <span class="n">A</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">aa_index</span><span class="p">]</span> <span class="o">=</span> <span class="n">count</span>
</pre></div>
</div>
</div>
</section>
<section id="Now-...-add-the-counts-&gt;:o">
<h1>Now … add the counts &gt;:o<a class="headerlink" href="#Now-...-add-the-counts->:o" title="Link to this heading"></a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[30]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">time</span> <span class="o">=</span> <span class="s1">&#39;2500&#39;</span>
<span class="n">date</span> <span class="o">=</span> <span class="s1">&#39;2024-02-17&#39;</span>
<span class="n">experiment</span> <span class="o">=</span> <span class="s1">&#39;metabolism-redux-classic-minimal&#39;</span>
<span class="n">entry</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">experiment</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">time</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">date</span><span class="si">}</span><span class="s1">&#39;</span>
<span class="n">folder</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;out/cofactors/</span><span class="si">{</span><span class="n">entry</span><span class="si">}</span><span class="s1">/&#39;</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">output_all</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">folder</span> <span class="o">+</span> <span class="s1">&#39;0_output.npy&#39;</span><span class="p">,</span><span class="n">allow_pickle</span><span class="o">=</span><span class="s1">&#39;TRUE&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">item</span><span class="p">()</span>
<span class="c1"># output = np.load(r&quot;out/geneRxnVerifData/output_glc.npy&quot;, allow_pickle=True, encoding=&#39;ASCII&#39;).tolist()</span>
<span class="n">output</span> <span class="o">=</span> <span class="n">output_all</span><span class="p">[</span><span class="s1">&#39;agents&#39;</span><span class="p">][</span><span class="s1">&#39;0&#39;</span><span class="p">]</span>
<span class="n">fba</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;fba_results&#39;</span><span class="p">]</span>
<span class="n">mass</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;mass&#39;</span><span class="p">]</span>
<span class="n">bulk</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">output</span><span class="p">[</span><span class="s1">&#39;bulk&#39;</span><span class="p">])</span>

<span class="n">fluxes</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_fluxes&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:])</span>
<span class="n">exchanges</span> <span class="o">=</span> <span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_exchange_dmdt&#39;</span><span class="p">]</span>

<span class="c1"># output[&#39;listeners&#39;][&#39;unique_molecule_counts&#39;][&#39;active_ribosome&#39;]</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[32]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">][</span><span class="s1">&#39;active_RNAP&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[32]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
[747,
 738,
 747,
 761,
 773,
 780,
 778,
 763,
 766,
 761,
 764,
 766,
 775,
 781,
 776,
 781,
 799,
 795,
 802,
 799,
 799,
 801,
 797,
 798,
 799,
 801,
 803,
 797,
 797,
 795,
 794,
 785,
 779,
 780,
 785,
 800,
 791,
 799,
 801,
 804,
 805,
 807,
 800,
 803,
 788,
 793,
 793,
 798,
 801,
 798,
 805,
 805,
 797,
 809,
 804,
 806,
 809,
 816,
 805,
 805,
 796,
 790,
 787,
 778,
 786,
 792,
 800,
 793,
 801,
 801,
 808,
 805,
 801,
 780,
 783,
 779,
 777,
 774,
 772,
 768,
 772,
 761,
 768,
 780,
 783,
 785,
 789,
 794,
 796,
 801,
 792,
 790,
 803,
 801,
 795,
 791,
 798,
 800,
 803,
 803,
 798,
 795,
 794,
 777,
 774,
 778,
 778,
 783,
 782,
 789,
 795,
 801,
 804,
 791,
 789,
 795,
 769,
 765,
 768,
 777,
 766,
 780,
 779,
 774,
 783,
 787,
 797,
 801,
 794,
 793,
 792,
 802,
 802,
 794,
 787,
 789,
 790,
 789,
 786,
 788,
 792,
 802,
 799,
 795,
 796,
 798,
 781,
 787,
 794,
 800,
 785,
 785,
 796,
 794,
 788,
 788,
 797,
 797,
 797,
 800,
 802,
 801,
 796,
 786,
 779,
 785,
 771,
 789,
 793,
 794,
 790,
 790,
 789,
 794,
 801,
 807,
 801,
 802,
 801,
 801,
 821,
 819,
 824,
 824,
 826,
 842,
 833,
 836,
 823,
 829,
 816,
 809,
 807,
 808,
 810,
 812,
 812,
 809,
 812,
 788,
 776,
 788,
 782,
 789,
 792,
 800,
 803,
 802,
 806,
 813,
 809,
 806,
 803,
 803,
 805,
 798,
 805,
 811,
 819,
 820,
 823,
 819,
 826,
 839,
 830,
 831,
 832,
 833,
 839,
 834,
 830,
 822,
 823,
 824,
 823,
 826,
 825,
 836,
 835,
 828,
 818,
 821,
 815,
 807,
 799,
 813,
 822,
 817,
 811,
 809,
 813,
 812,
 815,
 814,
 811,
 805,
 808,
 806,
 819,
 814,
 807,
 804,
 808,
 811,
 806,
 802,
 818,
 814,
 816,
 809,
 808,
 814,
 811,
 807,
 808,
 810,
 819,
 821,
 823,
 827,
 823,
 826,
 834,
 837,
 839,
 838,
 840,
 828,
 842,
 835,
 839,
 834,
 834,
 835,
 842,
 839,
 840,
 836,
 830,
 834,
 836,
 836,
 830,
 833,
 831,
 827,
 832,
 840,
 832,
 837,
 835,
 845,
 854,
 858,
 849,
 853,
 830,
 831,
 825,
 820,
 825,
 831,
 823,
 825,
 818,
 816,
 827,
 838,
 839,
 849,
 848,
 851,
 855,
 855,
 848,
 842,
 845,
 844,
 836,
 828,
 837,
 844,
 844,
 854,
 856,
 857,
 855,
 847,
 843,
 839,
 839,
 844,
 839,
 833,
 826,
 833,
 837,
 845,
 843,
 837,
 845,
 840,
 846,
 840,
 852,
 849,
 835,
 832,
 833,
 836,
 841,
 834,
 837,
 841,
 837,
 835,
 850,
 839,
 843,
 849,
 843,
 847,
 847,
 849,
 852,
 851,
 852,
 864,
 859,
 840,
 830,
 835,
 829,
 822,
 805,
 810,
 818,
 824,
 824,
 833,
 835,
 838,
 846,
 842,
 850,
 844,
 854,
 853,
 859,
 860,
 853,
 857,
 850,
 850,
 847,
 843,
 837,
 828,
 830,
 840,
 848,
 859,
 859,
 848,
 844,
 824,
 834,
 833,
 829,
 830,
 831,
 838,
 843,
 845,
 843,
 852,
 856,
 842,
 842,
 838,
 844,
 846,
 841,
 838,
 839,
 839,
 837,
 836,
 826,
 823,
 821,
 829,
 834,
 830,
 837,
 843,
 838,
 844,
 839,
 852,
 850,
 839,
 842,
 839,
 850,
 844,
 850,
 858,
 866,
 864,
 862,
 857,
 850,
 860,
 855,
 843,
 837,
 835,
 836,
 846,
 856,
 855,
 855,
 855,
 872,
 865,
 875,
 882,
 892,
 883,
 889,
 893,
 889,
 880,
 873,
 867,
 863,
 863,
 850,
 847,
 860,
 855,
 845,
 846,
 847,
 844,
 848,
 853,
 859,
 852,
 842,
 832,
 829,
 825,
 831,
 827,
 839,
 845,
 855,
 855,
 853,
 853,
 854,
 843,
 837,
 844,
 845,
 841,
 852,
 854,
 860,
 865,
 863,
 867,
 869,
 873,
 870,
 869,
 872,
 886,
 899,
 901,
 910,
 904,
 912,
 892,
 889,
 883,
 886,
 877,
 880,
 883,
 880,
 878,
 885,
 884,
 882,
 886,
 892,
 897,
 896,
 908,
 899,
 894,
 884,
 882,
 878,
 873,
 880,
 870,
 876,
 891,
 895,
 896,
 883,
 883,
 883,
 887,
 887,
 880,
 874,
 882,
 885,
 897,
 894,
 885,
 882,
 881,
 881,
 882,
 888,
 896,
 897,
 898,
 891,
 900,
 898,
 898,
 894,
 907,
 908,
 911,
 901,
 895,
 885,
 885,
 887,
 884,
 876,
 885,
 871,
 872,
 865,
 876,
 879,
 886,
 898,
 885,
 888,
 889,
 889,
 895,
 898,
 897,
 883,
 878,
 872,
 881,
 887,
 880,
 874,
 885,
 884,
 886,
 882,
 881,
 886,
 878,
 884,
 874,
 877,
 882,
 884,
 883,
 890,
 889,
 889,
 893,
 889,
 897,
 908,
 920,
 905,
 909,
 909,
 908,
 897,
 906,
 903,
 902,
 895,
 901,
 903,
 905,
 901,
 890,
 891,
 896,
 908,
 920,
 918,
 918,
 921,
 921,
 929,
 919,
 910,
 907,
 914,
 923,
 910,
 906,
 911,
 902,
 912,
 910,
 915,
 918,
 920,
 928,
 930,
 931,
 936,
 919,
 928,
 928,
 932,
 929,
 922,
 925,
 920,
 919,
 912,
 914,
 903,
 902,
 903,
 915,
 903,
 891,
 892,
 889,
 891,
 892,
 885,
 879,
 875,
 879,
 884,
 886,
 881,
 890,
 888,
 882,
 871,
 863,
 867,
 869,
 872,
 867,
 862,
 857,
 861,
 851,
 860,
 858,
 857,
 854,
 873,
 863,
 865,
 869,
 880,
 894,
 895,
 889,
 883,
 880,
 880,
 883,
 890,
 896,
 894,
 898,
 878,
 865,
 847,
 856,
 862,
 862,
 872,
 867,
 872,
 873,
 875,
 877,
 872,
 885,
 882,
 880,
 881,
 884,
 892,
 900,
 898,
 901,
 914,
 917,
 904,
 903,
 887,
 890,
 892,
 910,
 926,
 926,
 931,
 933,
 925,
 929,
 920,
 907,
 915,
 922,
 904,
 909,
 913,
 909,
 911,
 904,
 904,
 901,
 905,
 910,
 905,
 915,
 922,
 909,
 914,
 923,
 927,
 922,
 916,
 919,
 932,
 934,
 940,
 938,
 945,
 951,
 957,
 960,
 956,
 950,
 953,
 958,
 946,
 953,
 944,
 951,
 953,
 943,
 936,
 935,
 939,
 935,
 941,
 944,
 943,
 943,
 944,
 950,
 956,
 958,
 962,
 953,
 959,
 963,
 957,
 968,
 956,
 952,
 960,
 949,
 952,
 963,
 955,
 954,
 966,
 964,
 974,
 967,
 970,
 974,
 972,
 968,
 971,
 980,
 988,
 985,
 984,
 982,
 989,
 982,
 959,
 975,
 973,
 960,
 964,
 967,
 981,
 968,
 970,
 974,
 967,
 964,
 972,
 975,
 980,
 973,
 972,
 977,
 975,
 957,
 964,
 961,
 971,
 964,
 962,
 964,
 969,
 963,
 951,
 956,
 972,
 982,
 988,
 980,
 988,
 1006,
 1016,
 1005,
 1012,
 1004,
 1000,
 998,
 992,
 978,
 971,
 971,
 971,
 955,
 951,
 955,
 945,
 944,
 955,
 957,
 959,
 960,
 967,
 977,
 977,
 967,
 964,
 966,
 970,
 979,
 984,
 993,
 993,
 990,
 994,
 999,
 1000,
 1001,
 995,
 991,
 982,
 968,
 968,
 964,
 966,
 974,
 979,
 971,
 957,
 955,
 959,
 974,
 971,
 975,
 976,
 987,
 989,
 982,
 979,
 984,
 978,
 986,
 972,
 975,
 966,
 976,
 986,
 978,
 989,
 989,
 990,
 991,
 998,
 1000,
 1007,
 1004,
 1013,
 1003,
 1003,
 1011,
 1004,
 996,
 1000,
 1015,
 1026,
 1017,
 1010,
 988,
 994,
 993,
 983,
 976,
 ...]
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[33]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">f</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">folder</span> <span class="o">+</span> <span class="s1">&#39;agent_steps.pkl&#39;</span><span class="p">,</span> <span class="s1">&#39;rb&#39;</span><span class="p">)</span>
<span class="n">agent</span> <span class="o">=</span> <span class="n">dill</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
<span class="n">f</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>

<span class="n">metabolism</span> <span class="o">=</span> <span class="n">agent</span><span class="p">[</span><span class="s1">&#39;ecoli-metabolism-redux-classic&#39;</span><span class="p">]</span>
<span class="n">stoichiometry</span> <span class="o">=</span> <span class="n">metabolism</span><span class="o">.</span><span class="n">stoichiometry</span>
<br/></pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[34]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">initial_state</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="nb">open</span><span class="p">(</span><span class="s1">&#39;data/wcecoli_t0.json&#39;</span><span class="p">))</span>

<span class="n">bulk_ids</span> <span class="o">=</span> <span class="p">[</span><span class="n">item</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">initial_state</span><span class="p">[</span><span class="s1">&#39;bulk&#39;</span><span class="p">]]</span>

<span class="n">bulk</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="n">bulk_ids</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[35]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># in the bulk dataframe, update RNAP and 50S, 30S rib counts from unique molecules since they are sequestered as unique when in use.</span>
<span class="k">for</span> <span class="n">unique_key</span><span class="p">,</span> <span class="n">bulk_id</span> <span class="ow">in</span> <span class="p">[(</span><span class="s1">&#39;active_ribosome&#39;</span><span class="p">,</span> <span class="s1">&#39;CPLX0-3962&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;active_ribosome&#39;</span><span class="p">,</span> <span class="s1">&#39;CPLX0-3953&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;active_RNAP&#39;</span><span class="p">,</span> <span class="s1">&#39;APORNAP-CPLX&#39;</span><span class="p">)]:</span>
    <span class="k">if</span> <span class="n">unique_key</span> <span class="ow">in</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">]:</span>
        <span class="n">unique_count</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">][</span><span class="n">unique_key</span><span class="p">]</span>
        <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">bulk_id</span><span class="o">+</span><span class="s1">&#39;[c]&#39;</span><span class="p">]</span> <span class="o">+=</span> <span class="n">unique_count</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[36]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="s1">&#39;selC-tRNA[c]&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[36]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
0       2094
1       2094
2       2096
3       2097
4       2100
        ...
2496    4718
2497    4718
2498    4719
2499    4720
2500    4722
Name: selC-tRNA[c], Length: 2501, dtype: int64
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[37]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">ecocyc_to_wcm_map</span> <span class="o">=</span> <span class="p">{}</span>

<span class="c1"># combined complex and protein names</span>
<span class="n">complex_protein_names</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">])</span> <span class="o">+</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_complex_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">])</span>

<span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">complex_protein_names</span><span class="p">:</span>

    <span class="c1"># find complex name in bulk_ids</span>
    <span class="n">found</span> <span class="o">=</span> <span class="kc">False</span>

    <span class="k">try</span><span class="p">:</span>
        <span class="n">idx</span> <span class="o">=</span> <span class="n">bulk_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">name</span><span class="o">+</span><span class="s1">&#39;[c]&#39;</span><span class="p">)</span>
        <span class="n">ecocyc_to_wcm_map</span><span class="p">[</span><span class="n">name</span><span class="p">]</span> <span class="o">=</span> <span class="n">name</span><span class="o">+</span><span class="s1">&#39;[c]&#39;</span>
        <span class="n">found</span> <span class="o">=</span> <span class="kc">True</span>
        <span class="c1"># print(f&#39;found {complex_name} at {idx}&#39;)</span>

    <span class="k">except</span> <span class="ne">ValueError</span><span class="p">:</span>
        <span class="c1"># delete key</span>
        <span class="n">found</span> <span class="o">=</span> <span class="kc">False</span>


    <span class="k">if</span> <span class="n">found</span> <span class="o">==</span> <span class="kc">False</span><span class="p">:</span>

        <span class="k">for</span> <span class="nb">id</span> <span class="ow">in</span> <span class="n">bulk_ids</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">name</span><span class="o">+</span><span class="s1">&#39;[&#39;</span> <span class="ow">in</span> <span class="nb">id</span> <span class="ow">and</span> <span class="nb">id</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="n">name</span><span class="p">)</span> <span class="ow">and</span> <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="nb">id</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
                <span class="c1">#print(f&#39;found {name} in {id} with nonzero count&#39;)</span>
                <span class="n">ecocyc_to_wcm_map</span><span class="p">[</span><span class="n">name</span><span class="p">]</span> <span class="o">=</span> <span class="nb">id</span>
                <span class="n">found</span> <span class="o">=</span> <span class="kc">True</span>
                <span class="k">break</span>           <span class="c1"># ensures preferring nonzero counts</span>

            <span class="k">elif</span> <span class="n">name</span><span class="o">+</span><span class="s1">&#39;[&#39;</span> <span class="ow">in</span> <span class="nb">id</span> <span class="ow">and</span> <span class="nb">id</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="n">name</span><span class="p">):</span>
                <span class="c1"># print(f&#39;found {name} in {id} with zero count&#39;)</span>
                <span class="n">ecocyc_to_wcm_map</span><span class="p">[</span><span class="n">name</span><span class="p">]</span> <span class="o">=</span> <span class="nb">id</span>
                <span class="n">found</span> <span class="o">=</span> <span class="kc">True</span>

    <span class="k">if</span> <span class="n">found</span> <span class="o">==</span> <span class="kc">False</span><span class="p">:</span>
        <span class="n">ecocyc_to_wcm_map</span><span class="p">[</span><span class="n">name</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;--TRANS-ACENAPHTHENE-12-DIOL[c]&#39;</span> <span class="c1"># should be none</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;could not find </span><span class="si">{</span><span class="n">name</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
could not find MONOMER0-1241
could not find MONOMER0-4223
could not find CPLX0-3964
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[38]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">complex_wcm_names</span> <span class="o">=</span> <span class="p">[</span><span class="n">ecocyc_to_wcm_map</span><span class="p">[</span><span class="n">name</span><span class="p">]</span> <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">C_names</span><span class="p">]</span>

<span class="n">counts</span> <span class="o">=</span> <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">complex_wcm_names</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[39]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">time</span> <span class="o">=</span> <span class="s1">&#39;1300&#39;</span>
<span class="n">date</span> <span class="o">=</span> <span class="s1">&#39;2023-12-14&#39;</span>
<span class="n">experiment</span> <span class="o">=</span> <span class="s1">&#39;metabolism-redux-classic&#39;</span>
<span class="n">entry</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">experiment</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">time</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">date</span><span class="si">}</span><span class="s1">&#39;</span>
<span class="n">folder</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;out/cofactors/</span><span class="si">{</span><span class="n">entry</span><span class="si">}</span><span class="s1">/&#39;</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[40]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">output_all</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">folder</span> <span class="o">+</span> <span class="s1">&#39;0_output.npy&#39;</span><span class="p">,</span><span class="n">allow_pickle</span><span class="o">=</span><span class="s1">&#39;TRUE&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">item</span><span class="p">()</span>
<span class="c1"># output = np.load(r&quot;out/geneRxnVerifData/output_glc.npy&quot;, allow_pickle=True, encoding=&#39;ASCII&#39;).tolist()</span>
<span class="n">output</span> <span class="o">=</span> <span class="n">output_all</span><span class="p">[</span><span class="s1">&#39;agents&#39;</span><span class="p">][</span><span class="s1">&#39;0&#39;</span><span class="p">]</span>
<span class="n">fba</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;fba_results&#39;</span><span class="p">]</span>
<span class="n">mass</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;mass&#39;</span><span class="p">]</span>
<span class="n">bulk</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">output</span><span class="p">[</span><span class="s1">&#39;bulk&#39;</span><span class="p">])</span>

<span class="n">fluxes</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_fluxes&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:])</span>
<span class="n">exchanges</span> <span class="o">=</span> <span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_exchange_dmdt&#39;</span><span class="p">]</span>

<span class="n">bulk_ids</span> <span class="o">=</span> <span class="p">[</span><span class="n">item</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">initial_state</span><span class="p">[</span><span class="s1">&#39;bulk&#39;</span><span class="p">]]</span>

<span class="n">bulk</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="n">bulk_ids</span>
<span class="c1"># output[&#39;listeners&#39;][&#39;unique_molecule_counts&#39;][&#39;active_ribosome&#39;]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[41]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># in the bulk dataframe, update RNAP and 50S, 30S rib counts from unique molecules since they are sequestered as unique when in use.</span>
<span class="k">for</span> <span class="n">unique_key</span><span class="p">,</span> <span class="n">bulk_id</span> <span class="ow">in</span> <span class="p">[(</span><span class="s1">&#39;active_ribosome&#39;</span><span class="p">,</span> <span class="s1">&#39;CPLX0-3962&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;active_ribosome&#39;</span><span class="p">,</span> <span class="s1">&#39;CPLX0-3953&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;active_RNAP&#39;</span><span class="p">,</span> <span class="s1">&#39;APORNAP-CPLX&#39;</span><span class="p">)]:</span>
    <span class="k">if</span> <span class="n">unique_key</span> <span class="ow">in</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">]:</span>
        <span class="n">unique_count</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">][</span><span class="n">unique_key</span><span class="p">]</span>
        <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">bulk_id</span><span class="o">+</span><span class="s1">&#39;[c]&#39;</span><span class="p">]</span> <span class="o">+=</span> <span class="n">unique_count</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[42]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">rich_counts</span> <span class="o">=</span> <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">complex_wcm_names</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[43]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">time</span> <span class="o">=</span> <span class="s1">&#39;5500&#39;</span>
<span class="n">date</span> <span class="o">=</span> <span class="s1">&#39;2024-01-30&#39;</span>
<span class="n">experiment</span> <span class="o">=</span> <span class="s1">&#39;metabolism-redux-classic-anaerobic&#39;</span>
<span class="n">entry</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">experiment</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">time</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">date</span><span class="si">}</span><span class="s1">&#39;</span>
<span class="n">folder</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;out/cofactors/</span><span class="si">{</span><span class="n">entry</span><span class="si">}</span><span class="s1">/&#39;</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[44]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">output_all</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">folder</span> <span class="o">+</span> <span class="s1">&#39;0_output.npy&#39;</span><span class="p">,</span><span class="n">allow_pickle</span><span class="o">=</span><span class="s1">&#39;TRUE&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">item</span><span class="p">()</span>
<span class="c1"># output = np.load(r&quot;out/geneRxnVerifData/output_glc.npy&quot;, allow_pickle=True, encoding=&#39;ASCII&#39;).tolist()</span>
<span class="n">output</span> <span class="o">=</span> <span class="n">output_all</span><span class="p">[</span><span class="s1">&#39;agents&#39;</span><span class="p">][</span><span class="s1">&#39;0&#39;</span><span class="p">]</span>
<span class="n">fba</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;fba_results&#39;</span><span class="p">]</span>
<span class="n">mass</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;mass&#39;</span><span class="p">]</span>
<span class="n">bulk</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">output</span><span class="p">[</span><span class="s1">&#39;bulk&#39;</span><span class="p">])</span>

<span class="n">fluxes</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_fluxes&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:])</span>
<span class="n">exchanges</span> <span class="o">=</span> <span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_exchange_dmdt&#39;</span><span class="p">]</span>

<span class="n">bulk_ids</span> <span class="o">=</span> <span class="p">[</span><span class="n">item</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">initial_state</span><span class="p">[</span><span class="s1">&#39;bulk&#39;</span><span class="p">]]</span>

<span class="n">bulk</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="n">bulk_ids</span>
<span class="c1"># output[&#39;listeners&#39;][&#39;unique_molecule_counts&#39;][&#39;active_ribosome&#39;]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[45]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># in the bulk dataframe, update RNAP and 50S, 30S rib counts from unique molecules since they are sequestered as unique when in use.</span>
<span class="k">for</span> <span class="n">unique_key</span><span class="p">,</span> <span class="n">bulk_id</span> <span class="ow">in</span> <span class="p">[(</span><span class="s1">&#39;active_ribosome&#39;</span><span class="p">,</span> <span class="s1">&#39;CPLX0-3962&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;active_ribosome&#39;</span><span class="p">,</span> <span class="s1">&#39;CPLX0-3953&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;active_RNAP&#39;</span><span class="p">,</span> <span class="s1">&#39;APORNAP-CPLX&#39;</span><span class="p">)]:</span>
    <span class="k">if</span> <span class="n">unique_key</span> <span class="ow">in</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">]:</span>
        <span class="n">unique_count</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">][</span><span class="n">unique_key</span><span class="p">]</span>
        <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">bulk_id</span><span class="o">+</span><span class="s1">&#39;[c]&#39;</span><span class="p">]</span> <span class="o">+=</span> <span class="n">unique_count</span>

<span class="n">anaerobic_counts</span> <span class="o">=</span> <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">complex_wcm_names</span><span class="p">]</span>
</pre></div>
</div>
</div>
</section>
<section id="External-data-sets">
<h1>External data sets<a class="headerlink" href="#External-data-sets" title="Link to this heading"></a></h1>
</section>
<section id="Saving-outputs-to-files-compatible-with-Julia">
<h1>Saving outputs to files compatible with Julia<a class="headerlink" href="#Saving-outputs-to-files-compatible-with-Julia" title="Link to this heading"></a></h1>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[46]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="nb">sum</span><span class="p">(</span><span class="n">W2</span><span class="p">[:,</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;Regulation of transcription&#39;</span><span class="p">)])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[46]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1.0
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[47]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="n">protein_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;EG10230-MONOMER&#39;</span><span class="p">),</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;Regulation of transcription&#39;</span><span class="p">)]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[47]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
[425, 1174]
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[48]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">W</span><span class="p">[</span><span class="n">protein_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;EG10230-MONOMER&#39;</span><span class="p">),</span> <span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;Regulation of transcription&#39;</span><span class="p">)]</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[48]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
0.0
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[49]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">complex_names</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_complex_df</span><span class="p">[</span><span class="s1">&#39;common_name&#39;</span><span class="p">])</span>
<span class="n">protein_names</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;common_name&#39;</span><span class="p">])</span>
<span class="n">cofactor_names</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_cofactor_df</span><span class="p">[</span><span class="s1">&#39;common_name&#39;</span><span class="p">])</span>
<span class="n">pathway_names</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">parsed_pathway_df</span><span class="p">[</span><span class="s1">&#39;common_name&#39;</span><span class="p">])</span>

<span class="c1"># if cplx name is nan then use id</span>
<span class="n">complex_names</span> <span class="o">=</span> <span class="p">[</span><span class="n">complex_names</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">complex_names</span><span class="p">[</span><span class="n">i</span><span class="p">])</span> <span class="o">==</span> <span class="nb">str</span> <span class="k">else</span> <span class="n">parsed_complex_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">complex_names</span><span class="p">))]</span>
<span class="n">protein_names</span> <span class="o">=</span> <span class="p">[</span><span class="n">protein_names</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">protein_names</span><span class="p">[</span><span class="n">i</span><span class="p">])</span> <span class="o">==</span> <span class="nb">str</span> <span class="k">else</span> <span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">protein_names</span><span class="p">))]</span>
<span class="n">cofactor_names</span> <span class="o">=</span> <span class="p">[</span><span class="n">cofactor_names</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">cofactor_names</span><span class="p">[</span><span class="n">i</span><span class="p">])</span> <span class="o">==</span> <span class="nb">str</span> <span class="k">else</span> <span class="n">parsed_cofactor_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">cofactor_names</span><span class="p">))]</span>
<span class="n">pathway_names</span> <span class="o">=</span> <span class="p">[</span><span class="n">pathway_names</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">pathway_names</span><span class="p">[</span><span class="n">i</span><span class="p">])</span> <span class="o">==</span> <span class="nb">str</span> <span class="k">else</span> <span class="n">parsed_pathway_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">pathway_names</span><span class="p">))]</span>

<span class="c1"># save C, P and E to julia-compatible formats</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/C_matrix.csv&#39;</span><span class="p">,</span> <span class="n">C</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%i</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/P_matrix.csv&#39;</span><span class="p">,</span> <span class="n">P</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%i</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/E_matrix.csv&#39;</span><span class="p">,</span> <span class="n">E</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%i</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/W_matrix.csv&#39;</span><span class="p">,</span> <span class="n">W</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">float64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">)</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/W1_matrix.csv&#39;</span><span class="p">,</span> <span class="n">W1</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">float64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">)</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/W2_matrix.csv&#39;</span><span class="p">,</span> <span class="n">W2</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">float64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">)</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/A_matrix.csv&#39;</span><span class="p">,</span> <span class="n">A</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">float64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">)</span>


<span class="c1"># write all ids to single file with each list on a new line</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/complex_ids.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">complex_ids</span><span class="p">))</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/protein_ids.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">protein_ids</span><span class="p">))</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/cofactor_ids.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cofactor_ids</span><span class="p">))</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/element_ids.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">element_ids</span><span class="p">))</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/pathway_ids.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">pathway_ids</span><span class="p">))</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/amino_acid_ids.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">amino_acid_ids</span><span class="p">))</span>

<span class="c1"># write all names to single file with each list on a new line</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/complex_names.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">complex_names</span><span class="p">))</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/protein_names.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">protein_names</span><span class="p">))</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/cofactor_names.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">cofactor_names</span><span class="p">))</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/pathway_names.txt&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">pathway_names</span><span class="p">))</span>


<span class="c1"># save counts of proteins and complexes</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/counts.csv&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">counts</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%i</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="c1"># save counts of proteins and complexes</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/rich_counts.csv&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">rich_counts</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%i</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="c1"># save counts of proteins and complexes</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/anaerobic_counts.csv&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">anaerobic_counts</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%i</span><span class="s1">&#39;</span><span class="p">)</span>



<span class="c1"># save masses</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/monomer_masses.csv&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;sequence_mass&#39;</span><span class="p">]),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%f</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="c1"># list comprehension, if complex name in membrane_areas[&#39;id&#39;], add membrane_ids[&#39;area_trans&#39;], otherwise add 0</span>
<span class="n">membrane_ids</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">membrane_proteins</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">],</span> <span class="n">membrane_proteins</span><span class="p">[</span><span class="s1">&#39;area_trans&#39;</span><span class="p">]))</span>
<span class="n">membrane_complex_areas</span> <span class="o">=</span> <span class="p">[</span><span class="n">membrane_ids</span><span class="p">[</span><span class="nb">complex</span><span class="p">]</span> <span class="k">if</span> <span class="nb">complex</span> <span class="ow">in</span> <span class="n">membrane_ids</span> <span class="k">else</span> <span class="mi">0</span> <span class="k">for</span> <span class="nb">complex</span> <span class="ow">in</span> <span class="n">complex_ids</span><span class="p">]</span>
<span class="n">membrane_monomer_areas</span> <span class="o">=</span> <span class="p">[</span><span class="n">membrane_ids</span><span class="p">[</span><span class="n">monomer</span><span class="p">]</span> <span class="k">if</span> <span class="n">monomer</span> <span class="ow">in</span> <span class="n">membrane_ids</span> <span class="k">else</span> <span class="mi">0</span> <span class="k">for</span> <span class="n">monomer</span> <span class="ow">in</span> <span class="n">protein_ids</span><span class="p">]</span>

<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/complex_areas.csv&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">membrane_complex_areas</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%f</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/monomer_areas.csv&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">membrane_monomer_areas</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%f</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="c1"># save tree_matrix</span>
<span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/tree_matrix.csv&#39;</span><span class="p">,</span> <span class="n">tree_matrix</span><span class="p">,</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%i</span><span class="s1">&#39;</span><span class="p">)</span>
<br/><br/></pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[62]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">entry</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="sa">f</span><span class="s2">&quot;rich-</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">19</span><span class="p">)]):</span>
    <span class="n">folder</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;out/cofactors/</span><span class="si">{</span><span class="n">entry</span><span class="si">}</span><span class="s1">/&#39;</span>


    <span class="n">output_all</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">folder</span> <span class="o">+</span> <span class="s1">&#39;0_output.npy&#39;</span><span class="p">,</span> <span class="n">allow_pickle</span><span class="o">=</span><span class="s1">&#39;TRUE&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">item</span><span class="p">()</span>
    <span class="c1"># output = np.load(r&quot;out/geneRxnVerifData/output_glc.npy&quot;, allow_pickle=True, encoding=&#39;ASCII&#39;).tolist()</span>
    <span class="n">output</span> <span class="o">=</span> <span class="n">output_all</span><span class="p">[</span><span class="s1">&#39;agents&#39;</span><span class="p">][</span><span class="s1">&#39;0&#39;</span><span class="p">]</span>

    <span class="n">fba</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;fba_results&#39;</span><span class="p">]</span>
    <span class="n">mass</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;mass&#39;</span><span class="p">]</span>
    <span class="n">bulk</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">output</span><span class="p">[</span><span class="s1">&#39;bulk&#39;</span><span class="p">])</span>

    <span class="n">fluxes</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_fluxes&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:])</span>
    <span class="n">exchanges</span> <span class="o">=</span> <span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_exchange_dmdt&#39;</span><span class="p">]</span>

    <span class="n">bulk_ids</span> <span class="o">=</span> <span class="p">[</span><span class="n">item</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">initial_state</span><span class="p">[</span><span class="s1">&#39;bulk&#39;</span><span class="p">]]</span>

    <span class="n">bulk</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="n">bulk_ids</span>
    <span class="c1"># output[&#39;listeners&#39;][&#39;unique_molecule_counts&#39;][&#39;active_ribosome&#39;]</span>
    <span class="c1"># in the bulk dataframe, update RNAP and 50S, 30S rib counts from unique molecules since they are sequestered as unique when in use.</span>
    <span class="k">for</span> <span class="n">unique_key</span><span class="p">,</span> <span class="n">bulk_id</span> <span class="ow">in</span> <span class="p">[(</span><span class="s1">&#39;active_ribosome&#39;</span><span class="p">,</span> <span class="s1">&#39;CPLX0-3962&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;active_ribosome&#39;</span><span class="p">,</span> <span class="s1">&#39;CPLX0-3953&#39;</span><span class="p">),</span>
                                <span class="p">(</span><span class="s1">&#39;active_RNAP&#39;</span><span class="p">,</span> <span class="s1">&#39;APORNAP-CPLX&#39;</span><span class="p">)]:</span>
        <span class="k">if</span> <span class="n">unique_key</span> <span class="ow">in</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">]:</span>
            <span class="n">unique_count</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">][</span><span class="n">unique_key</span><span class="p">]</span>
            <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">bulk_id</span> <span class="o">+</span> <span class="s1">&#39;[c]&#39;</span><span class="p">]</span> <span class="o">+=</span> <span class="n">unique_count</span>

    <span class="k">if</span> <span class="n">i</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">rich_counts</span> <span class="o">=</span> <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">complex_wcm_names</span><span class="p">]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">names</span><span class="o">=</span><span class="s1">&#39;timestep&#39;</span><span class="p">)</span>
        <span class="n">rich_counts</span><span class="p">[</span><span class="s2">&quot;cell&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">i</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">temp</span> <span class="o">=</span> <span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">complex_wcm_names</span><span class="p">]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">names</span><span class="o">=</span><span class="s1">&#39;timestep&#39;</span><span class="p">)</span>
        <span class="n">temp</span><span class="p">[</span><span class="s2">&quot;cell&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">i</span>
        <span class="n">rich_counts</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">rich_counts</span><span class="p">,</span> <span class="n">temp</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[63]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/rich_counts_big.csv&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">rich_counts</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">int64</span><span class="p">),</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;</span><span class="si">%i</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[59]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">rich_counts</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[59]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>timestep</th>
      <th>1-PFK[c]</th>
      <th>2OXOGLUTARATEDEH-CPLX[c]</th>
      <th>3-ISOPROPYLMALDEHYDROG-CPLX[c]</th>
      <th>3-ISOPROPYLMALISOM-CPLX[c]</th>
      <th>3-METHYL-2-OXOBUT-OHCH3XFER-CPLX[c]</th>
      <th>3-OXOACYL-ACP-SYNTHII-CPLX[c]</th>
      <th>6PFK-1-CPX[c]</th>
      <th>6PFK-2-CPX[c]</th>
      <th>6PGLUCONDEHYDROG-CPLX[c]</th>
      <th>...</th>
      <th>YRBE-MONOMER[i]</th>
      <th>YRBF-MONOMER[m]</th>
      <th>YRBG-MONOMER[i]</th>
      <th>YTFQ-MONOMER[p]</th>
      <th>YTFR-MONOMER[m]</th>
      <th>YTFT-MONOMER[i]</th>
      <th>ZNUA-MONOMER[p]</th>
      <th>ZNUB-MONOMER[i]</th>
      <th>ZNUC-MONOMER[i]</th>
      <th>cell</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>75</td>
      <td>167</td>
      <td>2348</td>
      <td>4545</td>
      <td>510</td>
      <td>1015</td>
      <td>1013</td>
      <td>115</td>
      <td>4061</td>
      <td>...</td>
      <td>99</td>
      <td>180</td>
      <td>53</td>
      <td>1</td>
      <td>15</td>
      <td>0</td>
      <td>989</td>
      <td>0</td>
      <td>20</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>75</td>
      <td>167</td>
      <td>2348</td>
      <td>4545</td>
      <td>510</td>
      <td>1015</td>
      <td>1013</td>
      <td>115</td>
      <td>4061</td>
      <td>...</td>
      <td>100</td>
      <td>181</td>
      <td>53</td>
      <td>1</td>
      <td>15</td>
      <td>0</td>
      <td>990</td>
      <td>0</td>
      <td>20</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>75</td>
      <td>167</td>
      <td>2348</td>
      <td>4547</td>
      <td>510</td>
      <td>1016</td>
      <td>1014</td>
      <td>115</td>
      <td>4061</td>
      <td>...</td>
      <td>100</td>
      <td>183</td>
      <td>53</td>
      <td>1</td>
      <td>15</td>
      <td>0</td>
      <td>990</td>
      <td>0</td>
      <td>20</td>
      <td>0</td>
    </tr>
    <tr>
      <th>3</th>
      <td>3</td>
      <td>75</td>
      <td>167</td>
      <td>2348</td>
      <td>4548</td>
      <td>510</td>
      <td>1016</td>
      <td>1014</td>
      <td>115</td>
      <td>4064</td>
      <td>...</td>
      <td>100</td>
      <td>183</td>
      <td>53</td>
      <td>1</td>
      <td>15</td>
      <td>0</td>
      <td>991</td>
      <td>0</td>
      <td>20</td>
      <td>0</td>
    </tr>
    <tr>
      <th>4</th>
      <td>4</td>
      <td>75</td>
      <td>167</td>
      <td>2349</td>
      <td>4549</td>
      <td>510</td>
      <td>1017</td>
      <td>1014</td>
      <td>115</td>
      <td>4067</td>
      <td>...</td>
      <td>100</td>
      <td>184</td>
      <td>53</td>
      <td>1</td>
      <td>15</td>
      <td>0</td>
      <td>992</td>
      <td>0</td>
      <td>20</td>
      <td>0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>1296</th>
      <td>1296</td>
      <td>174</td>
      <td>260</td>
      <td>3190</td>
      <td>6683</td>
      <td>617</td>
      <td>987</td>
      <td>1905</td>
      <td>107</td>
      <td>5466</td>
      <td>...</td>
      <td>98</td>
      <td>315</td>
      <td>99</td>
      <td>0</td>
      <td>6</td>
      <td>13</td>
      <td>1583</td>
      <td>1</td>
      <td>52</td>
      <td>6</td>
    </tr>
    <tr>
      <th>1297</th>
      <td>1297</td>
      <td>174</td>
      <td>260</td>
      <td>3191</td>
      <td>6689</td>
      <td>617</td>
      <td>988</td>
      <td>1906</td>
      <td>107</td>
      <td>5467</td>
      <td>...</td>
      <td>99</td>
      <td>318</td>
      <td>99</td>
      <td>0</td>
      <td>6</td>
      <td>13</td>
      <td>1583</td>
      <td>1</td>
      <td>52</td>
      <td>6</td>
    </tr>
    <tr>
      <th>1298</th>
      <td>1298</td>
      <td>174</td>
      <td>260</td>
      <td>3193</td>
      <td>6695</td>
      <td>618</td>
      <td>988</td>
      <td>1908</td>
      <td>107</td>
      <td>5468</td>
      <td>...</td>
      <td>100</td>
      <td>319</td>
      <td>99</td>
      <td>0</td>
      <td>6</td>
      <td>13</td>
      <td>1583</td>
      <td>1</td>
      <td>52</td>
      <td>6</td>
    </tr>
    <tr>
      <th>1299</th>
      <td>1299</td>
      <td>174</td>
      <td>260</td>
      <td>3197</td>
      <td>6700</td>
      <td>618</td>
      <td>988</td>
      <td>1909</td>
      <td>107</td>
      <td>5474</td>
      <td>...</td>
      <td>98</td>
      <td>317</td>
      <td>99</td>
      <td>0</td>
      <td>6</td>
      <td>13</td>
      <td>1583</td>
      <td>1</td>
      <td>52</td>
      <td>6</td>
    </tr>
    <tr>
      <th>1300</th>
      <td>1300</td>
      <td>174</td>
      <td>260</td>
      <td>3199</td>
      <td>6710</td>
      <td>619</td>
      <td>988</td>
      <td>1911</td>
      <td>107</td>
      <td>5480</td>
      <td>...</td>
      <td>99</td>
      <td>317</td>
      <td>99</td>
      <td>0</td>
      <td>6</td>
      <td>13</td>
      <td>1584</td>
      <td>1</td>
      <td>52</td>
      <td>6</td>
    </tr>
  </tbody>
</table>
<p>9084 rows × 5529 columns</p>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># save counts as csv with header for Khoa</span>
<span class="n">counts</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/counts_header.csv&#39;</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/protein_table.csv&#39;</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</section>
<section id="Tests">
<h1>Tests<a class="headerlink" href="#Tests" title="Link to this heading"></a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;EG12332-MONOMER&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<p>First, let’s test that the cofactor counts are correct for SULFITE-REDUCT-CPLX and its constituent BETACOMP-MONOMER</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">parsed_protein_df</span><span class="p">)):</span>
    <span class="n">sequence</span> <span class="o">=</span> <span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="s1">&#39;sequence_processed&#39;</span><span class="p">]</span>

    <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">sequence</span><span class="p">)</span> <span class="ow">is</span> <span class="nb">dict</span> <span class="ow">and</span> <span class="s1">&#39;U&#39;</span> <span class="ow">in</span> <span class="n">sequence</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;found U in </span><span class="si">{</span><span class="n">parsed_protein_df</span><span class="o">.</span><span class="n">at</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s2">&quot;id&quot;</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Sulfite reductase complex count: </span><span class="si">{</span><span class="n">counts</span><span class="p">[</span><span class="mi">1020</span><span class="p">]</span><span class="si">}</span><span class="s1">, hemoprotein subunit count: </span><span class="si">{</span><span class="n">counts</span><span class="p">[</span><span class="mi">1271</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Expected iron count: </span><span class="si">{</span><span class="n">counts</span><span class="p">[</span><span class="mi">1020</span><span class="p">]</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="mi">5</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="mi">4</span><span class="si">}</span><span class="s1"> for the first, </span><span class="si">{</span><span class="n">counts</span><span class="p">[</span><span class="mi">1271</span><span class="p">]</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="mi">5</span><span class="w"> </span><span class="si">}</span><span class="s1"> for the second&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>Let’s check the pathway counts for the related pathway.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">cplx_pathways</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="p">[</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">((</span><span class="n">C</span> <span class="o">@</span> <span class="n">W</span><span class="p">)[</span><span class="mi">1020</span><span class="p">,</span> <span class="p">:])[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="p">]</span>
<span class="n">monomer_pathways</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="p">[</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">((</span><span class="n">C</span> <span class="o">@</span> <span class="n">W</span><span class="p">)[</span><span class="mi">1271</span><span class="p">,</span> <span class="p">:])[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="p">]</span>


<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Sulfite reductase is in the following pathways: </span><span class="si">{</span><span class="n">cplx_pathways</span><span class="si">}</span><span class="s1">, while the hemoprotein is in </span><span class="si">{</span><span class="n">monomer_pathways</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>That means the pathway counts for iron should be the same as the cplx counts (or larger)</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="c1"># pathway_counts = (iron_counts @ C_n @ W)[  np.where((C @ W)[1020, :])[0][0]  ]</span>

<span class="c1"># print(f&#39;Pathway counts for iron: {pathway_counts}&#39;)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;&quot;&quot;Monomer index: </span><span class="si">{</span><span class="n">C_names</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;SUPEROX-DISMUTFE-MONOMER&#39;</span><span class="p">)</span><span class="si">}</span><span class="s2">, complex index: </span><span class="si">{</span><span class="n">C_names</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;SUPEROX-DISMUTFE-CPLX&#39;</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;&quot;&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;&quot;&quot;Monomer count: </span><span class="si">{</span><span class="n">counts</span><span class="p">[</span><span class="mi">5232</span><span class="p">]</span><span class="si">}</span><span class="s2">, complex count: </span><span class="si">{</span><span class="n">counts</span><span class="p">[</span><span class="mi">1021</span><span class="p">]</span><span class="si">}</span><span class="s2">&quot;&quot;&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;&quot;&quot;Monomer index: </span><span class="si">{</span><span class="n">C_names</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;SUPEROX-DISMUTMN-MONOMER&#39;</span><span class="p">)</span><span class="si">}</span><span class="s2">, complex index: </span><span class="si">{</span><span class="n">C_names</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;SUPEROX-DISMUTMN-CPLX&#39;</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;&quot;&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;&quot;&quot;Monomer count: </span><span class="si">{</span><span class="n">counts</span><span class="p">[</span><span class="mi">5233</span><span class="p">]</span><span class="si">}</span><span class="s2">, complex count: </span><span class="si">{</span><span class="n">counts</span><span class="p">[</span><span class="mi">1022</span><span class="p">]</span><span class="si">}</span><span class="s2">&quot;&quot;&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">mn_idx</span> <span class="o">=</span> <span class="n">element_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;MN&#39;</span><span class="p">)</span>
<span class="n">mn_counts</span> <span class="o">=</span> <span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">counts</span><span class="p">)</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="o">-</span><span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span> <span class="o">*</span> <span class="n">C</span> <span class="o">@</span> <span class="n">P</span> <span class="o">@</span> <span class="n">E</span><span class="p">)[:,</span> <span class="n">mn_idx</span><span class="p">]</span>
<span class="c1"># each complex count of 1 counts as 1 total annotation, so need to normalize</span>
<span class="n">C_n</span> <span class="o">=</span> <span class="n">C</span> <span class="o">/</span> <span class="p">(</span><span class="n">C</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">keepdims</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span> <span class="o">+</span> <span class="mf">1e-10</span><span class="p">)</span>

<span class="n">nz_idx</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">((</span><span class="n">C_n</span> <span class="o">@</span> <span class="n">W</span> <span class="o">@</span> <span class="n">W2</span><span class="p">)</span><span class="o">.</span><span class="n">T</span> <span class="o">@</span> <span class="n">mn_counts</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
<span class="n">mn_counts_group</span> <span class="o">=</span> <span class="p">(</span><span class="n">C_n</span> <span class="o">@</span> <span class="n">W</span> <span class="o">@</span> <span class="n">W2</span><span class="p">)</span><span class="o">.</span><span class="n">T</span> <span class="o">@</span> <span class="n">mn_counts</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">cplx_pathways</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="p">[</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">((</span><span class="n">C</span> <span class="o">@</span> <span class="n">W</span><span class="p">)[</span><span class="mi">5233</span><span class="p">,</span> <span class="p">:])[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="p">]</span>
<span class="n">monomer_pathways</span> <span class="o">=</span> <span class="n">pathway_ids</span><span class="p">[</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">((</span><span class="n">C</span> <span class="o">@</span> <span class="n">W</span><span class="p">)[</span><span class="mi">1022</span><span class="p">,</span> <span class="p">:])[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="p">]</span>


<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Superoxide dismutase is in the following pathways: </span><span class="si">{</span><span class="n">cplx_pathways</span><span class="si">}</span><span class="s1">, while the monomer is in </span><span class="si">{</span><span class="n">monomer_pathways</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">pathway_counts</span> <span class="o">=</span> <span class="p">(</span><span class="n">mn_counts</span> <span class="o">@</span> <span class="n">C_n</span> <span class="o">@</span> <span class="n">W</span> <span class="o">@</span> <span class="n">W2</span><span class="p">)[</span>  <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">((</span><span class="n">C</span> <span class="o">@</span> <span class="n">W</span> <span class="o">@</span> <span class="n">W2</span><span class="p">)[</span><span class="mi">5233</span><span class="p">,</span> <span class="p">:])[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>  <span class="p">]</span>
<span class="n">pathway_counts</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">mn_counts</span><span class="p">[</span><span class="mi">1022</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">counts</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_pathway_df</span><span class="p">[</span><span class="n">parsed_pathway_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;PWY0-1507&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">pathway_ids</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;PWY0-1507&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">W2</span><span class="p">[</span><span class="mi">753</span><span class="p">,:])</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">pathway_ids</span><span class="p">[</span><span class="mi">718</span><span class="p">]</span>
</pre></div>
</div>
</div>
<section id="go-annotation-for-proteins">
<h2>go annotation for proteins<a class="headerlink" href="#go-annotation-for-proteins" title="Link to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;BCCP-MONOMER&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="n">parsed_protein_df</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;EG10889-MONOMER&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">parsed_protein_df</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="s1">&#39;RNA0-301[c]&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="s1">&#39;NARH-MONOMER[j]&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">ecocyc_</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">bulk</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="s1">&#39;NITRATREDUCTA-CPLX[i]&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">element</span> <span class="ow">in</span> <span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_fluxes&#39;</span><span class="p">]:</span>
    <span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">element</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">fluxes</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">stoichiometry</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="nb">vars</span><span class="p">(</span><span class="n">metabolism</span><span class="p">)</span><span class="o">.</span><span class="n">keys</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">metabolism</span><span class="o">.</span><span class="n">reaction_names</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;RXN0-7077&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">fluxes</span><span class="p">[:,</span> <span class="mi">3470</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">time</span> <span class="o">=</span> <span class="s1">&#39;2600&#39;</span>
<span class="n">date</span> <span class="o">=</span> <span class="s1">&#39;2023-11-28&#39;</span>
<span class="n">experiment</span> <span class="o">=</span> <span class="s1">&#39;metabolism-redux-classic&#39;</span>
<span class="n">entry</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">experiment</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">time</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">date</span><span class="si">}</span><span class="s1">&#39;</span>
<span class="n">folder</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;out/cofactors/</span><span class="si">{</span><span class="n">entry</span><span class="si">}</span><span class="s1">/&#39;</span>

<span class="n">f</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">folder</span> <span class="o">+</span> <span class="s1">&#39;agent_steps.pkl&#39;</span><span class="p">,</span> <span class="s1">&#39;rb&#39;</span><span class="p">)</span>
<span class="n">agent</span> <span class="o">=</span> <span class="n">dill</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
<span class="n">f</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>

<span class="n">metabolism</span> <span class="o">=</span> <span class="n">agent</span><span class="p">[</span><span class="s1">&#39;ecoli-metabolism-redux-classic&#39;</span><span class="p">]</span>
<span class="n">stoichiometry</span> <span class="o">=</span> <span class="n">metabolism</span><span class="o">.</span><span class="n">stoichiometry</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">output_all</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">folder</span> <span class="o">+</span> <span class="s1">&#39;0_output.npy&#39;</span><span class="p">,</span><span class="n">allow_pickle</span><span class="o">=</span><span class="s1">&#39;TRUE&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">item</span><span class="p">()</span>
<span class="c1"># output = np.load(r&quot;out/geneRxnVerifData/output_glc.npy&quot;, allow_pickle=True, encoding=&#39;ASCII&#39;).tolist()</span>
<span class="n">output</span> <span class="o">=</span> <span class="n">output_all</span><span class="p">[</span><span class="s1">&#39;agents&#39;</span><span class="p">][</span><span class="s1">&#39;0&#39;</span><span class="p">]</span>
<span class="n">fba</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;fba_results&#39;</span><span class="p">]</span>
<span class="n">mass</span> <span class="o">=</span> <span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;mass&#39;</span><span class="p">]</span>
<span class="n">bulk</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">output</span><span class="p">[</span><span class="s1">&#39;bulk&#39;</span><span class="p">])</span>

<span class="n">fluxes</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_fluxes&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:])</span>
<span class="n">exchanges</span> <span class="o">=</span> <span class="n">fba</span><span class="p">[</span><span class="s1">&#39;estimated_exchange_dmdt&#39;</span><span class="p">]</span>

<span class="n">output</span><span class="p">[</span><span class="s1">&#39;listeners&#39;</span><span class="p">][</span><span class="s1">&#39;unique_molecule_counts&#39;</span><span class="p">][</span><span class="s1">&#39;active_ribosome&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">flux_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">fluxes</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="n">metabolism</span><span class="o">.</span><span class="n">reaction_names</span><span class="p">)</span>
<span class="n">flux_df</span>

<span class="c1"># find top 10 fluxes for timestep 50</span>
<span class="n">flux_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">50</span><span class="p">,</span> <span class="p">:]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">ascending</span><span class="o">=</span><span class="kc">False</span><span class="p">)[</span><span class="mi">0</span><span class="p">:</span><span class="mi">50</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">exchanges</span><span class="o">.</span><span class="n">keys</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">exch_df_exp</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">([</span><span class="n">exchanges</span><span class="p">[</span><span class="s1">&#39;GLC[p]&#39;</span><span class="p">],</span> <span class="n">exchanges</span><span class="p">[</span><span class="s1">&#39;OXYGEN-MOLECULE[p]&#39;</span><span class="p">]])</span><span class="o">.</span><span class="n">T</span>
<span class="n">exch_df_exp</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;GLC[p]&#39;</span><span class="p">,</span> <span class="s1">&#39;OXYGEN-MOLECULE[p]&#39;</span><span class="p">]</span>
<span class="n">exch_df_exp</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;notebooks/cofactors/data/exchanges.csv&#39;</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="n">flux_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">50</span><span class="p">,</span> <span class="s1">&#39;NADH-DEHYDROG-A-RXN-NADH/UBIQUINONE-8/PROTON//NAD/CPD-9956/PROTON.46.&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">metabolism</span><span class="o">.</span><span class="n">reaction_names</span><span class="p">:</span>
    <span class="k">if</span> <span class="s1">&#39;RXN0-5388&#39;</span> <span class="ow">in</span> <span class="n">name</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2 notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021-2024, The Vivarium E. coli Authors.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>